// tatiye/Modal/index.js
import tatiyeNet,{setDecode,readmore}  from "../es6.js";  
import {profil} from "./profil.js";
import {myFrom} from "./myFrom.js";
import {myUpload} from "./myUpload.js";
import {myDelete} from "./myDelete.js";
import {myModal} from "./myModal.js";
import {myStandar} from "./myStandar.js";
import {myImport} from "./myImport.js";
import {myPdf}    from "./myPdf.js";
import {myFromAction}    from "./myFromAction.js";
import {myFormUpload}    from "./myFormUpload.js";
import {myRecycle}    from "./myRecycle.js";
import {myFromPdf}    from "./myFromPdf.js";
import {myFromImport}    from "./myFromImport.js";
import {myPridRaw}    from "./myPridRaw.js";
import {myImages}    from "./myImages.js";
import {myURLPdf}    from "./myURLPdf.js";
import {myRestore}    from "./myRestore.js";
import {myCrawler}    from "./myCrawler.js";
import {myconsole}    from "./myconsole.js";
import {myTerminal}    from "./myTerminal.js";
import {myPostleb}    from "./myPostleb.js";
import {myConsol}    from "./myConsol.js";
import {myInstal}    from "./myInstal.js";
import {myElement}    from "./myElement.js";
import {myMultiple}    from "./myMultiple.js";
import {privacy2,privacy3} from "../From/privacy.js";
export class Modal {
    constructor(titel, width, key,model,route,tabel='',content='',count='',page='',myrow='') {
        this.titel = readmore(titel,25)
        this.width = width
        this.key   = key
        this.model = model
        this.route = route
        this.tabel = tabel
        this.content = content
        this.count   = count
        this.page   = page
        this.myrow   = myrow
    }
    profil(newLastNAme,resheader,model=''){
       let result=profil(resheader,this.titel,this.width,this.route,this.key)
        mainRouteModal(result,resheader);
        $(".modal-content").draggable({
          handle: ".card-img-top",
          scroll: false
         });
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');
        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        }); 
    }
    restore(newLastNAme,resheader,model='',page=1){
        var typeContent=this.myrow.type;
        var reoadContent=this.myrow.reload;
        let result=myRestore(resheader,this.titel,this.width)
        mainRouteModal(result,resheader,this.titel);
        // console.log(this.myrow)
         privacy2(this.myrow,resheader)
         // DELETE FILE
         const buttonDeleteHandler = event => {
             // modal.remove();
           var settings = {
                  "url": tatiyeNet.api+"/v1/select/restore/"+this.tabel+"/"+this.key,
                  "method": "POST",
                  "timeout": 0,
                  "headers": {
                    "Content-Type": "application/json",
                    "Cookie": "PHPSESSID=nfni6jo1mrpn16kc3rq0515nvv"
                  },
                  "data": JSON.stringify({
                    "userid":tatiyeNet.userid
                  }),
                };

                $.ajax(settings).done(function (response) {
                    $("#"+resheader).html('');
                    if (typeContent=='datatables') {
                        $(reoadContent).DataTable().ajax.reload( null, false);
                    } else {
                        reload(); 
                    }
                });
         };
    let uidDelete = document.getElementById("uidDelete"+resheader);
        uidDelete.addEventListener('click', buttonDeleteHandler);
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');
        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }
    
    recycle(newLastNAme,resheader,model='',page=1){
        let result=myRecycle(resheader,this.titel,this.width)
        mainRouteModal(result,resheader,this.titel);
         privacy2(this.myrow,resheader)
         // DELETE FILE
         const buttonDeleteHandler = event => {
             // modal.remove();
           var settings = {
                  "url": tatiyeNet.api+"/v1/select/recycle/"+this.tabel+"/"+this.key,
                  "method": "POST",
                  "timeout": 0,
                  "headers": {
                    "Content-Type": "application/json",
                    "Cookie": "PHPSESSID=nfni6jo1mrpn16kc3rq0515nvv"
                  },
                  "data": JSON.stringify({
                    "userid":tatiyeNet.userid,
                    "package":this.myrow.package
                  }),
                };

                $.ajax(settings).done(function (response) {
                    $("#"+resheader).html('');
                    if (model=='datatables') {
                        $(".table").DataTable().ajax.reload( null, false);
                    } else if (model=='html') {
                        
                    } else {
                         // reload(page);

                        // window.app.storage();
                    }
                     
                });
          
          
         };
          let uidDelete = document.getElementById("uidDelete"+resheader);
          uidDelete.addEventListener('click', buttonDeleteHandler);
    
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');

        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }
    delete(newLastNAme,resheader,model='',page=1){
         let result=myDelete(resheader,this.titel,this.width)
        mainRouteModal(result,resheader,this.titel);
         privacy2(this.myrow,resheader)
         // DELETE FILE
         const buttonDeleteHandler = event => {
             // modal.remove();
           var settings = {
                  "url": tatiyeNet.api+"/v1/select/delete/"+this.tabel+"/"+this.key,
                  "method": "POST",
                  "timeout": 0,
                  "headers": {
                    "Content-Type": "application/json",
                    "Cookie": "PHPSESSID=nfni6jo1mrpn16kc3rq0515nvv"
                  },
                  "data": JSON.stringify({
                    "userid":tatiyeNet.userid
                  }),
                };

                $.ajax(settings).done(function (response) {
                    $("#"+resheader).html('');
                    if (model=='datatables') {
                        $(".table").DataTable().ajax.reload( null, false);
                    } else if (model=='html') {
                        
                    } else {
                         reload(page);

                        // window.app.storage();
                    }
                     
                });
          
          
         };
          let uidDelete = document.getElementById("uidDelete"+resheader);
          uidDelete.addEventListener('click', buttonDeleteHandler);
    
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');

        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }
    delete2(newLastNAme,resheader,model='',page=1){
        let result=myDelete(resheader,this.titel,this.width)
        mainRouteModal(result,resheader);
         privacy3(this.myrow,resheader)
       // console.log(this.myrow)
         // DELETE FILE
         const buttonDeleteHandler = event => {
            var mydelete = {
              "url": tatiyeNet.api+"/v2/"+model,
              "method": "DELETE",
              "timeout": 0,
              "headers": {
                "Authorization":tatiyeNet.token,
                "key":this.key,
                "userid": tatiyeNet.userid,
                "Cookie": "PHPSESSID=128g92bivnlb50155tkgnnkjoo"
              },
            };

            $.ajax(mydelete).done(function (response) {
                $("#"+resheader).html('');
                reload(page);

              
            });

          
         };
          let uidDelete = document.getElementById("uidDelete"+resheader);
          uidDelete.addEventListener('click', buttonDeleteHandler);
    
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');

        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }


    upload(newLastNAme,resheader,model=''){
        let result=myUpload(resheader,this.titel,this.width,this.route,this.key,this.page)
        mainRouteModal(result,resheader);
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');
        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }


    office(newLastNAme,resheader,model='') {
        let result=myImport(resheader,this.titel,this.width,this.route,this.key,this.page)
        mainRouteModal(result,resheader);
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');
        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }
    fromAction(newLastNAme,resheader,model='',content='') {
        if (content.element=='upload') {
            let result=myFormUpload(resheader,this.titel,'400px',this.route,this.key,content)
            mainRouteModal(result,resheader);
            setDraggable(resheader);
            let myclose = document.querySelectorAll("#myclose"+resheader);
            const headerClose = event => {
              $("#"+resheader).html('');
            };
            myclose.forEach(btn => {
              btn.addEventListener('click', headerClose);
            });
        } else if (content.element=='uploadFile') {
            let result=myFormUpload(resheader,this.titel,'400px',this.route,this.key,content)
            mainRouteModal(result,resheader);
            setDraggable(resheader);
            let myclose = document.querySelectorAll("#myclose"+resheader);
            const headerClose = event => {
              $("#"+resheader).html('');
            };
            myclose.forEach(btn => {
              btn.addEventListener('click', headerClose);
            });

        } else if (content.element=='office') {
                let result=myFromImport(resheader,this.titel,this.width,this.route,this.key,content)
                mainRouteModal(result,resheader);
                setDraggable(resheader);
                let myclose = document.querySelectorAll("#myclose"+resheader);
                const headerClose = event => {
                  $("#"+resheader).html('');
                };
                myclose.forEach(btn => {
                  btn.addEventListener('click', headerClose);
                });



        } else if (content.element=='crawler') {
            var result=myCrawler(resheader,this.titel,this.width,this.route,this.key)
                mainRouteModal(result,resheader);
                setDraggable(resheader);
                let myclose = document.querySelectorAll("#myclose"+resheader);
                const headerClose = event => {
                  $("#"+resheader).html('');
                };
                myclose.forEach(btn => {
                  btn.addEventListener('click', headerClose);
                });
        } else if (content.element=='get') {
            var result=myStandar(resheader,this.titel,this.width,this.route,this.key)
                mainRouteModal(result,resheader);
                setDraggable(resheader);
                let myclose = document.querySelectorAll("#myclose"+resheader);
                const headerClose = event => {
                  $("#"+resheader).html('');
                };
                myclose.forEach(btn => {
                  btn.addEventListener('click', headerClose);
                });

        } else if (content.element=='terminal') {
            var result=myTerminal(resheader,this.titel,this.width,this.myrow,this.key)
           
                mainRouteModal(result,resheader);
                setDraggable(resheader);
                let myclose = document.querySelectorAll("#myclose"+resheader);
                const headerClose = event => {
                  $("#"+resheader).html('');
                };
                myclose.forEach(btn => {
                  btn.addEventListener('click', headerClose);
                });
        } else if (content.element=='postleb') {
            var result=myPostleb(resheader,this.titel,this.width,this.myrow,this.key)
                mainRouteModal(result,resheader);
                setDraggable(resheader);
                let myclose = document.querySelectorAll("#myclose"+resheader);
                const headerClose = event => {
                  $("#"+resheader).html('');
                };
                myclose.forEach(btn => {
                  btn.addEventListener('click', headerClose);
                });

        } else if (content.element=='images') {
                var result=myImages(resheader,this.titel,this.width,this.route,this.key,content.filename)
                mainRouteModal(result,resheader);
                setDraggable(resheader);
                let myclose = document.querySelectorAll("#myclose"+resheader);
                const headerClose = event => {
                  $("#"+resheader).html('');
                };
                myclose.forEach(btn => {
                  btn.addEventListener('click', headerClose);
                });

        } else if (content.element=='myURLPdf') {
                var result=myURLPdf(resheader,this.titel,this.width,this.route,this.key,content.filename)
                mainRouteModal(result,resheader);
                setDraggable(resheader);
                let myclose = document.querySelectorAll("#myclose"+resheader);
                const headerClose = event => {
                  $("#"+resheader).html('');
                };
                myclose.forEach(btn => {
                  btn.addEventListener('click', headerClose);
                });



        } else if (content.element=='pridRaw') {
        
              let result=myPridRaw(resheader,
                this.titel,
                this.width,
                content,this.key,this.page)
              mainRouteModal(result,resheader);
              setDraggable(resheader);
              let myclose = document.querySelectorAll("#myclose"+resheader);
              const headerClose = event => {
                $("#"+resheader).html('');
              };
              myclose.forEach(btn => {
                btn.addEventListener('click', headerClose);
              });
        } else if (content.element=='dompdf') {
            let result=myFromPdf(resheader,this.titel,this.width,this.route,this.key,content)
            mainRouteModal(result,resheader);
            setDraggable(resheader);
            let myclose = document.querySelectorAll("#myclose"+resheader);
            const headerClose = event => {
              $("#"+resheader).html('');
            };
            myclose.forEach(btn => {
              btn.addEventListener('click', headerClose);
            });
    } else if (content.element=='console') {
            let result=myconsole(resheader,this.titel,this.width,this.route,this.key,content)
            mainRouteModal(result,resheader);
            setDraggable(resheader);
            let myclose = document.querySelectorAll("#myclose"+resheader);
            const headerClose = event => {
              $("#"+resheader).html('');
            };
            myclose.forEach(btn => {
              btn.addEventListener('click', headerClose);
            });
    } else if (content.element=='myconsole') {
            let result=myConsol(resheader,this.titel,this.width,this.route,this.key,content)
            mainRouteModal(result,resheader);
            setDraggable(resheader);
            let myclose = document.querySelectorAll("#myclose"+resheader);
            const headerClose = event => {
              $("#"+resheader).html('');
            };
            myclose.forEach(btn => {
              btn.addEventListener('click', headerClose);
            });
    } else if (content.element=='instal') {
            let result=myInstal(resheader,this.titel,this.width,this.route,this.key,content)
            mainRouteModal(result,resheader);
            setDraggable(resheader);
            let myclose = document.querySelectorAll("#myclose"+resheader);
            const headerClose = event => {
              $("#"+resheader).html('');
            };
            myclose.forEach(btn => {
              btn.addEventListener('click', headerClose);
            });
  
        } else {
            let result=myFromAction(resheader,this.titel,this.width,this.route,this.key,content)
            mainRouteModal(result,resheader);
            setDraggable(resheader);
            let myclose = document.querySelectorAll("#myclose"+resheader);
            const headerClose = event => {
              $("#"+resheader).html('');
            };
            myclose.forEach(btn => {
              btn.addEventListener('click', headerClose);
            });
       }
    }

    // pridRaw(newLastNAme,resheader,model='') {

    // }
    from(newLastNAme,resheader,model='') {
        
        let result=myFrom(resheader,this.titel,this.width,this.route,this.key,this.page)
        mainRouteModal(result,resheader);
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');
        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }
      dompdf(newLastNAme,resheader,model='') {
        let result=myPdf(resheader,this.titel,this.width,this.route,this.key)
        mainRouteModal(result,resheader);
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');
        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }  

    modalElement(newLastNAme) {
        if (newLastNAme.model=='multiple') {
          var result=myMultiple(newLastNAme.id,newLastNAme)
          mainRouteModal(result,newLastNAme.id);
        } else {
          var result=myElement(newLastNAme.data.id,newLastNAme)
          mainRouteModal(result,newLastNAme.data.id);
        }
  
    }
// 

    modal(newLastNAme,resheader,model='',type='') {
        console.log(model)
        if (type=='standar') {
          var result=myStandar(resheader,this.titel,this.width,this.route,this.key)
        } else if (type=='element') {
          var result=myElement(resheader,model)
        } else {
          var result=myModal(resheader,this.titel,this.width,this.route,this.key)
        }
        mainRouteModal(result,resheader);
        setDraggable(resheader);
        let myclose = document.querySelectorAll("#myclose"+resheader);
        const headerClose = event => {
          $("#"+resheader).html('');
        };
        myclose.forEach(btn => {
          btn.addEventListener('click', headerClose);
        });
    }

}


function mainRouteModal(result,resheader) {
        $("#mainRouteModal").append(result);
        $("#"+resheader).show();
}

function setDraggable(resheader) {
     $(".modal-contentNg").draggable({
      handle: "#modalHeader"+resheader,
      scroll: false
     });
}
 window.minimize=function (id,title,tx='',txClass='',myStack='',attrApp='') {
    var resHeaderx  =readmore(title,15);
       var template=`
        <div 
          id="minimizeOpen${id}" 
          titel="${title}" 
          textHeader="${tx}"
          txClass="${txClass}"
          stack="${myStack}"
          appnm="${attrApp}"
        style="width:200px; z-index: 31000;" class="demo-static-toast pull-left">
          <div class="toast p-4px ${tx}">
            <div style="border-bottom-color: rgb(205 212 224 / 0%);" class="toast-header ${tx}">
              <h6 class="${tx} tx-14 mg-b-0 mg-r-auto">${resHeaderx}</h6>
               <a onclick="minimizeOpen('${id}','${title}');" class="ml-2 mb-1 close tx-normal ">
               <i class="ms-Icon ms-Icon--ChromeRestore ${tx}" id="welcomeMinimize"></i>
              </a>
            </div>
          </div>
        </div>
       `;
       $("#"+id).hide();
       $("#footerMd").append(template);
 }
 window.minimizeOpen=function (id,title) {
  $("#"+id).show();
  $("#minimizeOpen"+id).remove();
 }
  window.minimizeLage=function (id,width) {
   if($("#wzeLage"+id).hasClass("largeitem")) {
      $('#wzeLage'+id).css({"width":'700px'});
      $("#icon"+id).html(' <i class="ms-Icon ms-Icon--Checkbox"></i>');
   }
   else {
     $('#wzeLage'+id).css({"width":width});
      $("#icon"+id).html('<i class="ms-Icon ms-Icon--ChromeRestore"></i>');
   }
   $("#wzeLage"+id).toggleClass('largeitem');
 }

  window.minimizeGrid=function (id,width,tx='') {
    if (width=='col-md-12') {
        var GridWh='col-md-6';
    } else  if (width=='col-md-6') {
        var GridWh='col-md-12';
    }
   if($("#"+id).hasClass("largeitem")) {
      $('#'+id).addClass(GridWh).removeClass(width); 
      $("#icon"+id).html(' <i class="'+tx+' ms-Icon ms-Icon--Checkbox"></i>');
   }
   else {
      $('#'+id).addClass(width).removeClass(GridWh); 
      $("#icon"+id).html('<i class="'+tx+' ms-Icon ms-Icon--ChromeRestore"></i>');
   }
   $("#"+id).toggleClass('largeitem');
 }
