// tatiye/Modal/myCrawler.js
import tatiyeNet,{components,itemSpinner} from "../es6.js"; 
export function myCrawler(resheader,title,width,route,inikey) { 
    console.log(tatiyeNet)
const assets = new components();
assets.from({
       "autoload"   :"body"+resheader,
       "storage"    :[false, 'demo','add'],
       "action"     :{
           'nama'       :['search',    '12' ,'URL']
      },
});
var send='oky';
window.useKeywordsStorage=function(page) {
          $.ajax({
                type: 'POST',
                 url: tatiyeNet.hostName+"/api/v1/select/crawler",
                 data: 'url='+page+'&userid='+tatiyeNet.userid+'&send='+send,
                 beforeSend: function() {
                  $("#bodymyCrawler").html(itemSpinner('modal'));    
                },
                success: function(response) {
                var desain=`
                    <div class="media">
                      <img style="width:150px" src="${response.thumbnail}" class="wd-100 rounded mg-r-20" alt="">
                      <div class="media-body">
                        <h6 class="mg-b-0 tx-inverse">${response.title}</h6>
                        <dd>${response.segment}</dd>
                        ${response.description}
                      </div>
                     </div>`;

                       $("#body"+resheader).html(desain);
                       var destroyLoader = setTimeout(function(){
                          $("#"+resheader).html('');
                           $(".table").DataTable().ajax.reload( null, false);
                             // window.app.storage();
                        },2000); 
               }
           });

}
  
   let myModal=`

 <div id="${resheader}" class="modal2" data-headerkey="${resheader}">
        <div class="modal-contentNg">
          <div id="wzeLage${resheader}" style="margin:auto;  margin-top:60px; background-color: #fff; width:${width}">
           <div class="setModal shadow-lg">
           <div id="modalHeader${resheader}">
           <div class="
                  setModal-header
                  d-flex
                  align-items-center
                  justify-content-between
                ">
                <h5 id="headerTitelx"  class="mg-b-0">${title}</h5>
                   <div class="d-flex"> 
                  <a onclick="minimize('${resheader}','${title}');" class="link-03 ml-10px mr-5px"href="javascript:void(0);">
                     <i class="fs-15px ms-Icon ms-Icon--ChromeMinimize"></i>
                  </a>   
                   <a  id="icon${resheader}" onclick="minimizeLage('${resheader}','${width}');" class="link-03 ml-10px mr-5px"href="javascript:void(0);">
                      <i class="ms-Icon ms-Icon--ChromeRestore" id="welcomeMinimize"></i>
                  </a>
                  <a id="myclose${resheader}" class="link-03 ml-10px mr-5px"href="javascript:void(0);">
                       <i class="ms-Icon ms-Icon--ChromeClose"></i>
                  </a>
                </div>
              </div>
            </div>
            <div class="setModal-body">
              <div id="setScrol"style="max-height:500px;">
                      <div class="row row-xs" id="content${resheader}"></div>
                      <div class="row row-xs" id="body${resheader}"></div>
                      <div class="row row-xs" id="pagination${resheader}"></div>
             
               </div>
            </div>
          </div>
          </div>
        </div>   
        </div>   
    
`;

return myModal;


};
