// tatiye/Oauth/index.js
 import tatiyeNet  from "../es6.js";
 import Vue from 'https://cdn.jsdelivr.net/npm/vue@2.7.14/dist/vue.esm.browser.js';
export class From {
    constructor(storage, action) {
        this.storage = storage
        this.action = action
    }
    setFromAction(model,row) {
          var myFrom        ='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
          console.log(row.key)
          $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                 beforeSend: function() {
                 },
                })
               .done(function (data) {
               // const info = document.getElementById(model);
                let innerHTML=`
                  <form id="form_${row.segment}"  method="POST" enctype="multipart/form-data"> 
                      <div class="row row-xs">
                         ${data}
                       <div class="col-md-12" style="padding-right:2.4px;">
                           <div class="setModal-footer">
                             <button type="button" class="myclose btn btn-xs btn-white mr-10px">Batalkan</button>
                              <button id="sendEntri${row.segment}" class="btn btn-xs btn-primary bold" type="button">Simpan </button>
                              <button style="display:none;" id="endEntri${row.segment}" class="mr-3px btn btn-xs btn-primary" type="button" disabled="">
                               <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span> Loading...
                             </button>
                           </div>
                        </div> 
                     </div>
                 </form>
                  `;
                   $("#"+model).html(innerHTML);
                   let modal     =document.getElementById("myModal");
                   let sendEntri =document.querySelectorAll("#sendEntri"+row.segment);
                   let myclose = document.querySelectorAll(".myclose");
                   const headerClose = event => {
                     modal.style.display = "none";
                    };
                    myclose.forEach(btn => {
                      btn.addEventListener('click', headerClose);
                    });
                   const setForumStorage = event => {
                             let form =$('#form_'+row.segment).serialize();
                             $.ajax({
                                    "type": 'POST',
                                    "url": tatiyeNet.package+'/'+row.route+'/controllers/send/'+row.segment,
                                    "data": form+'&token='+tatiyeNet.token+'&key='+row.key,
                                    "datatype": "json",
                                     "content": 'application/json;charset=utf-8',
                                     "headers": {
                                         "Authorization":tatiyeNet.token
                                     },
                                    "async": false,
                                     beforeSend: function() {
                                         $("#sendEntri"+row.segment).hide();
                                         $("#endEntri"+row.segment).show();  
                                     },
                                    success: function(data) {
                                             //$("#nugi").html(data);
                                              var add = JSON.parse(data);
                                              if (add.hasil !== "sukses") {
                                                      $.each(add.error, function( key, value ) {
                                                         var setId =$('#'+key).text();
                                                         var sedLabel=setId+' '+value;
                                                         $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                                                 $("#sendEntri"+row.segment).show();
                                                                 $("#endEntri"+row.segment).hide();
                                                     });
                                                      $.each(add.sukses, function( key, value ) {
                                                         
                                                         $('#info_'+key).html(' '); 
                                                     });
                                              } else {
                                                     $.each(add.sukses, function( key, value ) {
                                                         var label=$('#'+key).text(); 
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                    $("#sendEntri"+row.segment).hide();
                                                    $("#endEntri"+row.segment).show();
                                                  var destroyLoader = setTimeout(function(){
                                                    $("#sendEntri"+row.segment).show();
                                                    $("#endEntri"+row.segment).hide();
                                                     modal.style.display = "none";
                                                     // window.indexOn.setReload();
                                                 },1000);
                                                     window.app.storage();
                                                $('input[type="text"],select, textarea').val('');//kosongkan form
                                                $('input[type="email"],select,form textarea').val('');//kosongkan form
                                                $('input[type="password"], select, textarea').val('');//kosongkan form
                                                $('input[type="radio"]').prop('checked', false); 
                                                $('input[type="checkbox"]').prop('checked', false); 
                                              }
                                      }
                               }); 


                    };
                    sendEntri.forEach(entri => {
                      entri.addEventListener('click', setForumStorage);
                    });
           });
        
    }
}
