// tatiye/Charts/index.js
import tatiyeNet,{
    replace,
    setSplit,
    ngoreiDomElement,
    routerNavigasi,
    mainrouteReturn,
    explode,
    hashtagPage,
    mainrouteReturnHasPage,
    navigasiRequere,
    switchPage,
    tabDOM,
    internalDOM,
    userAgent
}  from "../es6.js";  
 /** PIE CHART **/
// export function chartPieDOM(item,dspage='1',dskeywords='') {
//  console.log('aaaaaaaaa')

// }
var txData   =[];
var txLabels   =[];
var txColor   =[];
export function chartPieDOM(item,countId) {

    item.forEach(set => {
     txLabels.push(set.label)
     txColor.push(set.clr)
     txData.push(set.data)

    })
   var datapie = {
          labels: txLabels,
          datasets: [{
            data: txData,
            backgroundColor:txColor
          }]
        };
        var optionpie = {
          maintainAspectRatio: false,
          responsive: true,
          legend: {
            display: false,
          },
          animation: {
            animateScale: true,
            animateRotate: true
          }
        };

        // return {
        //   data: datapie,
        //   options: optionpie
        // }

        // For a pie chart
         var ctx2 = document.getElementById(countId);
         var myDonutChart = new Chart(ctx2, {
           type: 'doughnut',
           data: datapie,
           options: optionpie
         });


}
