// tatiye/From/element.js
import tatiyeNet,{
modalEntri,
mainRouterId,
modalRouteFrom,
useHandel,
useTabPages,
useHandelID,
setModal,
setFrom,
useTabelToken,
components,
useHandelIDTabel,
itemSpinner,
setIndexedDB,
setDecode,
useHandelSheetkey
}  from "../es6.js"; 
  let count = 0;
 // import '../pagination.min.js';
export function setElementFrom(read,property,key='') {
    console.log(property)
  
   const app = new components();
   count++;
    
          var setPackage     =read['controllers'];
          var setPackageTabel=read['Query']['tabel'];

         if (property[1]=='add') {
              var expoModelTitel=read['title'];
         } else {
              var rowsetIData =useHandelIDTabel(setPackageTabel+','+property[1]);
              var expoModelTitel=property[2];

         }


              var expoModel=expoModelTitel.replace(/[^A-Za-z0-9]/g, '')+count

          // if (read['storage'][3]) {
          //       if (read['storage'][3]=='public') {
                  var privacy='public'; 
          //       } else {
          //          var privacy='privacy|'+read['storage'][3]; 
          //       } 
          //    } else {
          //          var privacy='public'; 
          //    }
          //  if (read.footer) {
          //    var  myFooter=read.footer;
          //   } else {
             var  myFooter='';
            // }
   if (property[0]=='from') {
              setModal({
                   'titel'   :expoModelTitel,
                   'width'   :read['storage'][2],
                   'key'     :property[1],
                   'model'   :expoModel,
                   'route'   :setPackage,
                   'tabel'   :setPackageTabel,
                   'content' :'fromAction',
                   'element' :property[0],
                   'footer'  :myFooter, //public,privacy 
                   'page'    :read['limit'],
                   'setdata' :privacy,
               });
                setFrom({
                   "storage"      :{
                     'label'       :true,
                     'tabel'       :setPackageTabel,
                     'segment'     :read['segment'],
                     'az'          :read['storage'][1],
                     'key'         :property[1],
                     'model'       :expoModel,
                     'route'       :setPackage,
                     'content'     :'from', //Outside,QuickSend 
                     'page'        :read['limit'],
                     'footer'      :myFooter, //public,privacy 
                     'setdata'     :privacy //public,privacy 
                   },
                   "action"      :read['action'],
                });
                //console.log(expoModel)
              if (read.footer) {
             var  myFooter=read.footer[0];
                 $("#sendEntri"+expoModel).html(myFooter);
                
               } 

           } else if (property[0]=='firebase') {

              setModal({
                   'titel':expoModelTitel,
                   'width':read['storage'][2],
                   'key'  :property[1],
                   'footer'  :myFooter, //public,privacy 
                   'model':expoModel,
                   'route':setPackage,
                   'tabel':setPackageTabel,
                   'content':'fromAction',
                   'element':property[0],
                   'page'   :read['limit'],
                   'setdata' :privacy,
               });
              
                setFrom({
                   "storage"      :{
                     'label'    :true,
                     'tabel'    :setPackageTabel,
                     'segment'  :read['segment'],
                     'az'       :read['storage'][1],
                     'key'      :property[1],
                     'model'    :expoModel,
                     'route'    :setPackage,
                     'content'  :'firebase', //Outside,QuickSend 
                     'page'     :read['limit'],
                      'footer'      :myFooter, //public,privacy 
                     'setdata'  :privacy, //public,privacy 
                   },
                   "action"      :read['action'],
                });





          /*
            |--------------------------------------------------------------------------
            | Initializes MODAL,TARGET BOOSTARP 
            |--------------------------------------------------------------------------
            | Develover Tatiye.Net 2018
            | @Date 
            */
           } else if (property[0]=='modal' || property[0]=='target'|| property[0]=='bootstrap') {
                const getAtr = document.querySelector(read['app']); 
                const getAtrElment = document.querySelector(read['atrID']+' '+getAtr.getAttribute('indexOn')); 
                let setAtrElment  ='element-'+getAtrElment.getAttribute('element')
                   let myDb=useHandel({
                       "token"    :read['token'],
                       "storage"       :{
                          "limit":1,
                          "page":1,
                          "keywords":property[1]
                       }
                 })

                 if (property[0]=='target') {
                      app.modalStandar({
                          "app":getAtrElment.getAttribute('element'),
                          "title"    :expoModelTitel,
                          "console"  :true,
                          "element":setAtrElment,
                          "property":property,
                          "token" :getAtr.getAttribute('token'),
                          "storage":myDb.storage,
                          "keywords":property[1],
                       })
                 } else if (property[0]=='bootstrap') {
                  // console.log(property[0])
                    const getElment = document.querySelector(setAtrElment); 
                    app.bootstrap({
                          "id":getElment.getAttribute('id'),
                          "modal":expoModel,
                          "bootstrap":getElment.getAttribute('bootstrap'),
                          "app":getAtrElment.getAttribute('element'),
                          "title"    :myDb.storage[0][property[2]],
                          "console"  :true,
                          "element":setAtrElment,
                          "property":property,
                          "token" :getAtr.getAttribute('token'),
                          "storage":myDb.storage,
                          "keywords":property[1],
                       })
                   
                 } else {
                      app.modalStandar({
                          "app":getAtrElment.getAttribute('element'),
                          "title"    :expoModelTitel,
                          "console"  :true,
                          "element":setAtrElment,
                          "property":property,
                          "token" :getAtr.getAttribute('token'),
                          "storage":myDb.storage,
                          "keywords":property[1],
                       })
                 }
       } else if (property[0]=='devices') {
              var appfile =useHandelIDTabel('appfile,'+property[1]);
              if (appfile['fileType'] =="png" || appfile['fileType'] =="jpg" || appfile['fileType'] =="jpeg") {
               setModal({
                   'titel':expoModelTitel,
                   'width':read['storage'][2],
                   'key'  :property[1],
                   'model':expoModel,
                   'route':setPackage,
                   'footer'  :myFooter, //public,privacy 
                   'tabel':setPackageTabel,
                   'content':'fromAction',
                   'element':'images',
                   'filename':tatiyeNet.hostName+'/'+appfile['filename'],
                   'page'   :read['limit'],
                   'setdata' :privacy,
                  });
              } else if (appfile['fileType']=="pdf") {
               setModal({
                   'titel':expoModelTitel,
                   'width':"600px",
                   'key'  :property[1],
                   'model':expoModel,
                   'route':setPackage,
                   'footer'  :myFooter, //public,privacy 
                   'tabel':setPackageTabel,
                   'content':'fromAction',
                   'element':'myURLPdf',
                   'filename':tatiyeNet.hostName+'/'+appfile['filename'],
                   'page'   :read['limit'],
                   'setdata' :privacy,
                  });
                }

        
       } else if (property[0]=='restore') {

            setModal({
               'titel':expoModelTitel,
               'width':'400px',
               'key'  :property[1],
               'model':expoModel,
               'route':setPackage,
               'tabel':setPackageTabel,
               'footer'  :myFooter, //public,privacy 
               'content':'restore',
               'element':"restore",
               'type':read['segments'],
               'reload'   :property[3],
               'page'   :1,
               'setdata' :privacy,
              });  
       } else if (property[0]=='archive') {
        setModal({
           'titel':expoModelTitel,
           'width':'500px',
           'key'  :property[1],
           'model':expoModel,
           'route':'devices',
           'footer'  :myFooter, //public,privacy 
           'tabel':'apparchive',
           'content':'fromAction',
           'element':"from",
           'page'   :read['limit'],
           'setdata' :privacy,
          });
           setFrom({
                     storage      :{
                       'label'    :true,
                       'tabel'    :'apparchive',
                       'segment'  :'archive',
                       'az'       :'a',
                       'key'      :'add',
                       'model'    :expoModel,
                       'route'    :'devices',
                       'footer'      :myFooter, //public,privacy 
                       'content'  :'from', 
                       'page'     :1,
                     },
                     action      :{
                          'deskripsi'  :['textarea',    '12' ,'Isi Deskripsi dari Archive'],
                          'nama'       :['hidden',      '12' ,expoModelTitel],
                          'categori'   :['hidden',      '12' ,'archive'],
                          'keyid'      :['hidden',      '12' ,property[1]],
                          'nmtabel'    :['hidden',      '12' ,setPackageTabel],
                          'package'    :['hidden',      '12' ,setPackage],
                      },
            });



       } else if (property[0]=='indexImages') {
        setModal({
           'titel':expoModelTitel,
           'width':'500px',
           'key'  :'add',
           'model':expoModel,
           'route':setPackage,
           'footer'  :myFooter, //public,privacy 
           'tabel':'appfile',
           'content':'fromAction',
           'element':"get",
           'page'   :1,
           'setdata' :privacy,
          });  
            $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/devices/controllers/indexImages.php',
                         data: 'key=appfile'+'&colom='+expoModelTitel+'&id='+property[1]+'&tabel='+setPackageTabel,
                         beforeSend: function() {
                          $("#body"+expoModel).html(itemSpinner());    
                        },
                        success: function(data) {
                          $("#body"+expoModel).html(data);
                          
                       }
                   }); 




       } else if (property[0]=='crawler') {
        
        setModal({
           'titel':'Crawler',
           'width':'500px',
           'key'  :'add',
           'model':expoModel,
           'route':setPackage,
           'footer'  :myFooter, //public,privacy 
           'tabel':setPackageTabel,
           'content':'fromAction',
           'element':"crawler",
           'page'   :1,
           'setdata' :privacy,
          });  

       } else if (property[0]=='sortable') {


        setModal({
           'titel':expoModelTitel,
           'width':'500px',
           'key'  :'add',
           'model':expoModel,
           'route':setPackage,
           'footer'  :myFooter, //public,privacy 
           'tabel':setPackageTabel,
           'content':'fromAction',
           'element':"get",
           'page'   :1,
           'setdata' :privacy,
          });  
            $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/devices/controllers/sortable.php',
                         data: 'key='+setPackageTabel+'&colom='+property[2],
                         beforeSend: function() {
                          $("#body"+expoModel).html(itemSpinner());    
                        },
                        success: function(data) {
                          $("#body"+expoModel).html(data);
                          $("#headerTitelx").html('Sortable');
                       }
                   }); 

       } else if (property[0]=='package') {
     
             history.replaceState('setRhref', '', tatiyeNet.host+'/'+setPackage+'/set/'+property[4]+'#'+setDecode(rowsetIData['id']));
             $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/'+property[4],
                         data: 'key='+rowsetIData['id']+'&resheader='+expoModelTitel,
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   }); 
                   // var JSONdata = JSON.parse(rowsetIData) 






       } else if (property[0]=='dialog') {
        setModal({
           'titel':expoModelTitel,
           'width':property[3],
           'key'  :property[1],
           'model':expoModel,
           'footer'  :myFooter, //public,privacy 
           'route':setPackage,
           'tabel':setPackageTabel,
           'content':'fromAction',
           'element':"get",
           'page'   :read['limit'],
           'setdata' :privacy,
          });
        if (property[4]=='undefined') {
          var routes=tatiyeNet.routes+'/resquire/pec/devices/controllers/dialog.php';  
        } else {
          var routes=tatiyeNet.routes+'/resquire/pec/'+property[4];  
        }
          $.ajax({
                type: 'POST',
                 url: routes,
                 data: 'key='+property[1]+'&tabel='+setPackageTabel,
                 beforeSend: function() {
                  $("#body"+expoModel).html(itemSpinner());    
                },
                success: function(data) {
                   $("#body"+expoModel).html(data);
               }
           }); 
          //          // var


        
       } else if (property[0]=='spam') {
        setModal({
           'titel':expoModelTitel,
           'width':'500px',
           'key'  :property[1],
           'model':expoModel,
           'route':setPackage,
           'tabel':setPackageTabel,
           'footer'  :myFooter, //public,privacy 
           'content':'fromAction',
           'element':"from",
           'type':read['segments'],
           'reload'   :property[3],
           'page'   :1,
           'setdata' :privacy,
          });
           setFrom({
                     storage      :{
                       'label'    :true,
                       'tabel'    :'apparchive',
                       'segment'  :'spam',
                       'az'       :'a',
                       'key'      :'add',
                       'model'    :expoModel,
                       'route'    :'devices',
                        'footer'      :myFooter, //public,privacy 
                       'content'  :'from', 
                       'page'     :1,
                     },
                     action      :{
                          'deskripsi'  :['textarea',    '12' ,'Isi Deskripsi dari Spam'],
                          'nama'       :['hidden',      '12' ,expoModelTitel],
                          'categori'   :['hidden',      '12' ,'spam'],
                          'keyid'      :['hidden',      '12' ,property[1]],
                          'nmtabel'    :['hidden',      '12' ,setPackageTabel],
                          'package'    :['hidden',      '12' ,setPackage],
                      },
            });
     } else if (property[0]=='hedtabel') {
         var rowfile =useHandelIDTabel(setPackageTabel+','+property[1]);
         setModal({
           'titel':expoModelTitel,
           'width':'600px',
           'key'  :property[1],
           'model':expoModel,
           'route':setPackage,
           'footer'  :myFooter, //public,privacy 
           'tabel':setPackageTabel,
           'content':'fromAction',
           'element':'from',
           'page'   :1,
           'setdata' :privacy,
          });
                setFrom({
                   "storage"      :{
                     'label'    :true,
                     'tabel'    :setPackageTabel,
                     'segment'  :read['segment'],
                     'az'       :read['storage'][1],
                     'key'      :property[1],
                     'model'    :expoModel,
                     'route'    :setPackage,
                     'content'  :'xlsx', //Outside,QuickSend 
                     'page'     :read['limit'],
                     'tableHead':rowfile.tableHead,
                      'footer'  :myFooter, //public,privacy 
                     'setdata'  :privacy //public,privacy 
                   },
                   "action"      :read['action'],
                });

     } else if (property[0]=='downloadfile') {
        var rowfile =useHandelIDTabel(setPackageTabel+','+property[1]);
        var wh=`'${rowfile.id}' AND nmtabel='${setPackageTabel}'`;
        var setFile =useHandelSheetkey("appfile,keyid="+wh);
        var retFile=tatiyeNet.hostName+'/'+setFile.data[0]['filename'];
        window.location.href = retFile;
       } else if (property[0]=='bookmark') {
   
       setModal({
           'titel':'Bookmark',
           'width':'500px',
           'key'  :property[1],
           'model':'Bookmark',
           'route':setPackage,
           'footer'  :myFooter, //public,privacy 
           'tabel':setPackageTabel,
           'content':'fromAction',
           'element':'from',
           'page'   :1,
           'setdata' :'public',
          });
           setFrom({
                     "storage"      :{
                       'label'    :true,
                       'tabel'    :'apparchive',
                       'segment'  :'book',
                       'az'       :'boo',
                       'key'      :property[1],
                       'model'    :'Bookmark',
                       'route'    :'devices',
                        'footer'      :myFooter, //public,privacy 
                       'content'  :'from', 
                       'page'     :1,
                     },
                     "action"      :{
                          'nama'            :['text',  '12' ,'Deskripsi'],
                          'deskripsi'       :['link',  '8' ,'URL Link'],
                          'categori'        :['hidden',      '12' ,'bookmark'],
                          'bookmark'        :['select','4','Categori','custom',['Link','youtube','facebook','github']],

    
                      },
            });


     } else if (property[0]=='label') {
        setModal({
           'titel':expoModelTitel,
           'width':'400px',
           'key'  :property[1],
           'model':expoModel,
           'route':'devices',
           'footer'  :myFooter, //public,privacy 
           'tabel':'apparchive',
           'content':'fromAction',
           'element':"from",
           'page'   :read['limit'],
           'setdata' :privacy,
          });
           setFrom({
                     storage      :{
                       'label'    :true,
                       'tabel'    :'apparchive',
                       'segment'  :'label',
                       'az'       :'a',
                       'key'      :'add',
                       'model'    :expoModel,
                       'route'    :'devices',
                       'content'  :'from', 
                        'footer'      :myFooter, //public,privacy 
                       'page'     :1,
                     },
                     action      :{
                          'deskripsi'  :['text',        '12' ,'Isi Deskripsi dari '+property[4]],
                          'nama'       :['hidden',      '12' ,expoModelTitel],
                          'categori'   :['hidden',      '12' ,property[3]],
                          'keyid'      :['hidden',      '12' ,property[1]],
                          'nmtabel'    :['hidden',      '12' ,setPackageTabel],
                          'package'    :['hidden',      '12' ,setPackage],
                      },
            });


     } else if (property[0]=='myXlsx') {
            history.replaceState('setRhref', '', tatiyeNet.host+'/'+property[3]+'/xlsx'+'#'+setDecode(rowsetIData['id']));
             $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/'+property[3]+'/xlsx',
                         data: 'key='+rowsetIData['newToken'],
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   }); 
             

     } else if (property[0]=='myOffice') {
             history.replaceState('setRhref', '', tatiyeNet.host+'/office/'+rowsetIData['fileType']+'#'+rowsetIData['newToken']);
             $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/office/'+rowsetIData['fileType'],
                         data: 'key='+rowsetIData['newToken'],
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   });                
     } else if (property[0]=='myOfficeDevices') {
        if (rowsetIData['package']=='office') {
        history.replaceState('setRhref', '', tatiyeNet.host+'/office/'+rowsetIData['fileType']+'#'+rowsetIData['newToken']);
              $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/office/'+rowsetIData['fileType'],
                         data: 'key='+rowsetIData['newToken'],
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   });

        } else if (rowsetIData['package']=='sisinfo') {

          history.replaceState('setRhref', '', tatiyeNet.host+'/sisinfo/xlsx'+'#'+setDecode(rowsetIData['newToken']));
             $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/sisinfo/xlsx',
                         data: 'key='+rowsetIData['newToken'],
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   }); 



        } else {
        history.replaceState('setRhref', '', tatiyeNet.host+'/devices/'+rowsetIData['categori']+'/package#'+rowsetIData['package']);
              $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/devices/pacakage',
                         data: 'key='+rowsetIData['newToken'],
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   });
        }

     } else if (property[0]=='myOfficeDevicesLabel') {
        if (rowsetIData['package']=='office') {
        history.replaceState('setRhref', '', tatiyeNet.host+'/office/'+rowsetIData['fileType']+'#'+rowsetIData['newToken']);
              $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/office/'+rowsetIData['fileType'],
                         data: 'key='+rowsetIData['fileType'],
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   });

        } else if (rowsetIData['package']=='sisinfo') {

          history.replaceState('setRhref', '', tatiyeNet.host+'/sisinfo/xlsx'+'#'+setDecode(rowsetIData['newToken']));
             $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/sisinfo/xlsx',
                         data: 'key='+rowsetIData['newToken'],
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   }); 

        } else {
        history.replaceState('setRhref', '', tatiyeNet.host+'/devices/'+rowsetIData['categori']+'#'+rowsetIData['package']);
        var mylabel=rowsetIData['categori'].split('label_')
              $.ajax({
                        type: 'POST',
                         url: tatiyeNet.routes+'/resquire/pec/devices/labelOn',
                         data: 'key='+mylabel[1],
                         beforeSend: function() {
                          $("#mainRoute").html(itemSpinner());    
                        },
                        success: function(data) {
                           $("#mainRoute").html(data);
                       }
                   });
        }


     } else if (property[0]=='apiNews') {
        setModal({
           'titel'   :'Create Api News',
           'width'   :'300px',
           'key'     :'add',
           'model'   :expoModel,
           'route'   :setPackage,
           'footer'  :myFooter, //public,privacy 
           'tabel'   :'appfile',
           'content' :'fromAction',
           'element' :"get",
           'page'    :1,
           'setdata' :privacy,
          });  
        // console.log(tatiyeNet)
            $.ajax({
                        type: 'POST',
                         url: tatiyeNet.hostName+'/api/v2/eyJkaXIiOiJuZXdzXC9BcGlcLzAuMVwvdXBkYXRlLnBocCIsInVpZCI6MX0',
                         data: 'key=appfile',
                         beforeSend: function() {
                          $("#body"+expoModel).html(itemSpinner('modal'));    
                        },
                        success: function(data) {
                          $(".table").DataTable().ajax.reload( null, false);
                          $("#"+expoModel).html('');
                          
                       }
                   }); 
     } else if (property[0]=='newlabel') {
     setModal({
           'titel':'Buat Label',
           'width':'300px',
           'key'  :property[1],
           'model':'Label',
           'footer'  :myFooter, //public,privacy 
           'route':'devices',
           'tabel':'appindikator',
           'content':'fromAction',
           'element':'from',
           'page'   :1,
           'setdata' :privacy,
          });
           setFrom({
                     storage      :{
                       'label'    :true,
                       'tabel'    :'appindikator',
                       'segment'  :'indikator',
                       'az'       :'a',
                       'key'      :property[1],
                       'model'    :expoModel,
                       'route'    :'devices',
                        'footer'      :myFooter, //public,privacy 
                       'content'  :'from', 
                       'page'     :1,
                     },
                     action      :{
                          'name'       :['text',        '12' ,'Nama Label'],
                          'value'      :['hidden',      '12' ,'label'],
                      },
            });
       } else if (property[0]=='query') {
           setModal({
           'titel':expoModelTitel,
           'width':'700px',
           'key'  :property[1],
           'model':expoModel,
           'route':'office',
           'footer'  :myFooter, //public,privacy 
           'tabel':'appoffice',
           'content':'fromAction',
           'element':"from",
           'page'   :read['limit'],
           'setdata' :privacy,
          });
           setFrom({
                     storage      :{
                       'label'    :true,
                       'tabel'    :'appoffice',
                       'segment'  :'fromquery1',
                       'az'       :'q',
                       'key'      :'add',
                       'model'    :expoModel,
                       'route'    :'office',
                        'footer'      :myFooter, //public,privacy 
                       'content'  :'codeMirror', 
                       'page'     :1,
                     },
                     action      :{
                          'deskripsi'  :['texquery',     '12',rowsetIData['deskripsi']],
                          'keyid'      :['hidden',      '12' ,property[1]],
                         
                      },
            });
       } else if (property[0]=='route') {
        setModal({
           'titel':expoModelTitel,
           'width':property[3],
           'key'  :property[1],
           'model':expoModel,
           'footer'  :myFooter, //public,privacy 
           'route':setPackage,
           'tabel':setPackageTabel,
           'content':'fromAction',
           'element':"get",
           'page'   :read['limit'],
           'setdata' :privacy,
          });
          var routes=tatiyeNet.routes+'/resquire/pec/'+property[4];  
          $.ajax({
                type: 'POST',
                 url: routes,
                 data: 'key='+property[1]+'&tabel='+setPackageTabel,
                 beforeSend: function() {
                  $("#body"+expoModel).html(itemSpinner());    
                },
                success: function(data) {
                   $("#body"+expoModel).html(data);
               }
           }); 
         } else if (property[0]=='stack') {
            // console.log(property[4])
           
             setModal({
                  'titel':expoModelTitel,
                  'width':read['storage'][2],
                  'key'  :property[1],
                  'model':expoModel,
                  'route':setPackage,
                  'tabel':setPackageTabel,
                  'footer'  :myFooter, //public,privacy 
                  'content':'fromAction',
                  'element':property[0],
                  'page'   :read['limit'],
                  'setdata' :privacy,
                 });
                     // if (property[3]=='datatables') {
                     //     // var  JSONdata = JSON.parse(property[4]);
                     //     console.log(JSON.stringify(property[4]))
                     // } else {
             
                               setFrom({
                                  "storage"      :{
                                    'label'    :true,
                                    'tabel'    :setPackageTabel,
                                    'segment'  :read['stack']['segment'],
                                    'az'       :read['storage'][1],
                                    'key'      :property[1],
                                    'model'    :expoModel,
                                    'route'    :setPackage,
                                    'content'  :'from', //Outside,QuickSend 
                                    'page'     :read['limit'],
                                     'footer'      :myFooter, //public,privacy 
                                    'setdata'  :privacy //public,privacy 
                                  },
                                  "action"      :read['stack']['action'],
                               });
                     // }
             } else if (property[0]=='recycle' || property[0]=='delete') {
                     setModal({
                       'titel':expoModelTitel,
                       'width':'300px',
                       'key'  :property[1],
                       'model':expoModel,
                       'route':false,
                       'package':setPackage,
                       'footer'  :myFooter, //public,privacy 
                       'tabel':setPackageTabel,
                       'content':property[0],
                       'page'   :1,
                       'setdata' :privacy,
                      });
               } else if (property[0]=='pridRaw') {
                         setModal({
                             'titel':expoModelTitel,
                             'width':read['storage'][2],
                             'key'  :property[1],
                             'model':expoModel,
                             'footer'  :myFooter, //public,privacy 
                             'route':false,
                             'tabel':false,
                             'content':'fromAction',
                             'element':property[0],
                             'page'   :1,
                             'setdata' :privacy,
                            });

                       app.myfromModal({
                           storage      :{
                             'label'    :false,
                             'tabel'    :setPackageTabel,
                             'segment'  :read['pridRaw']['segment'],
                             'az'       :read['storage'][1],
                             'key'      :property[1],
                             'model'    :expoModel,
                             'route'    :setPackage,
                             'content'  :'pridRaw',
                             'element'  :property[0],
                             'prind'    :'/raw/'+setPackage+'/'+property[3],
                             'page'     :1,
                             'setdata'  :privacy //public,privacy 
                           },
                           action      :read['pridRaw']['action'],
                        });
             } else if (property[0]=='upload' 
                        || property[0]=='document'
                        || property[0]=='dompdf'
                        //|| property[0]=='pridRaw'
                        || property[0]=='import') {
                             if (property[0]=='document') {
                                var doc='file';
                                var element='uploadFile';
                                var segment=false;
                                var syntax='';
                                var az=false;
                                var width='400px';
                             } else if (property[0]=='import') {
                                var segment='msexcel';
                                var element='office';
                                var doc='import';
                                var syntax=property[3]
                                var az=false;
                                 var width='400px';
                             } else if (property[0]=='dompdf') {
                              // console.log(property)
                                var segment=property[4];
                                var element='dompdf';
                                var doc='import';
                                var syntax=''
                                var az=property[3]
                                var width=property[5];
                             } else {
                                var doc='images';
                                var segment=false;
                                var element=property[0]
                                var syntax='';
                                var az=false;
                                var width='400px';
                             }
                            setModal({
                                 'titel':expoModelTitel,
                                 'width':width,
                                 'key'  :property[1],
                                 'model':expoModel,
                                 'route':setPackage,
                                 'tabel':setPackageTabel,
                                 'footer'  :myFooter, //public,privacy 
                                 'content':'fromAction',
                                 'element':element,
                                 'page'   :read['limit'],
                                 'setdata' :privacy,
                                });
                              setFrom({
                               storage      :{
                                 'headerId' :expoModelTitel,
                                 'label'    :false,
                                 'tabel'    :setPackageTabel,
                                 'segment'  :segment,
                                 'az'       :az,
                                 'key'      :property[1],
                                 'model'    :expoModel,
                                 'route'    :setPackage,
                                 'content'  :element, //Outside,QuickSend 
                                 'page'     :read['limit'],
                                  'footer'      :myFooter, //public,privacy 
                                 'setdata'  :privacy, //public,privacy 
                                 'syntax'   :syntax
                               },
                                   action      :{
                                       'filename'    :[doc,  '12' ,'Images']
                                     }
                            });
                }
};
