// tatiye/From/FromTables.js
 import tatiyeNet,{useHandelIDTabel,setDateTime,explode,useSpinner}  from "../es6.js";
 import {privacy} from "./privacy.js";
export class FromTables {
    constructor(storage, action,content) {
        this.storage = storage
        this.action = action
        this.content = content
      
 
    }
    userid(model,row,content,resheader){
          var dirFile='upload';
          var myFrom='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                  // "resheader"   :model
                 beforeSend: function() {
                  $("#bodyFrom"+model).html(useSpinner('text-primary','40px')); 
                 },
                })
               .done(function (template) {
                $("#bodyFrom"+resheader).html(template);
                  $("#form_"+resheader).on('submit', function(e){
                          e.preventDefault();
                              var formData = new FormData($(this)[0]);
                              formData.append('userid',tatiyeNet.userid);
                           $.ajax({
                               type: 'POST',
                               url: tatiyeNet.api+'/v1/userid/'+dirFile+'/appuserprofil/'+row.key,
                               data: formData,
                               contentType: false,
                               cache: false,
                               processData:false,
                               beforeSend: function(){
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();  
                               },
                               success: function(data){
                               /*
                               |--------------------------------------------------------------------------
                               | Initializes title 
                               |--------------------------------------------------------------------------
                               | Develover Tatiye.Net 2020
                               | @Date 
                               */ 
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();
                                 $("#uploadStatus").html(data.status);  
                                var destroyLoader = setTimeout(function(){
                                  $("#"+resheader).html('');
                                     // window.app.storage();
                                   $(".table").DataTable().ajax.reload( null, false);
                                },1000);    
                               $("input[type=file]").val('');
                               /*
                               |--------------------------------------------------------------------------
                               | AND title 
                               |--------------------------------------------------------------------------
                               */
                               }
                           });
                      });  
                });
    }
    upload(model,row,content,resheader){
      // console.log(this.storage)
        if (content=='uploadFile') {
          var dirFile='document';
        } else {
          var dirFile='upload';
        }
          var myFrom='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                  // "resheader"   :model
                 beforeSend: function() {
                  $("#bodyFrom"+model).html(useSpinner('text-primary','40px')); 
                 },
                })
               .done(function (template) {
                $("#bodyFrom"+resheader).html(template);
                  $("#form_"+resheader).on('submit', function(e){
                          e.preventDefault();
                              var formData = new FormData($(this)[0]);
                              formData.append('userid',tatiyeNet.userid);
                           $.ajax({
                               type: 'POST',
                               url: tatiyeNet.api+'/v1/select/'+dirFile+'/'+row.tabel+'/'+row.key,
                               data: formData,
                               contentType: false,
                               cache: false,
                               processData:false,
                               beforeSend: function(){
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();  
                               },
                               success: function(data){
                               /*
                               |--------------------------------------------------------------------------
                               | Initializes title 
                               |--------------------------------------------------------------------------
                               | Develover Tatiye.Net 2020
                               | @Date 
                               */ 
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();
                                 $("#uploadStatus").html(data.status);  
                                var destroyLoader = setTimeout(function(){
                                  $("#"+resheader).html('');
                                   $(".table").DataTable().ajax.reload( null, false);
                                     // window.app.storage();
                                },1000);    
                               $("input[type=file]").val('');
                               /*
                               |--------------------------------------------------------------------------
                               | AND title 
                               |--------------------------------------------------------------------------
                               */
                               }
                           });
                      });  
                });


    }
    from(model,row) {
      
          var myFrom        ='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
          $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                  beforeSend: function() {
                     $("#bodyFrom"+model).html(useSpinner('text-primary','40px'));  
                 },
                })
               .done(function (desain) {
                   $("#bodyFrom"+model).html(desain);
                   privacy(row);
                const setForumStorage = event => {
                             let form =$('#form_'+model).serialize();
                             $.ajax({
                                    "type": 'POST',
                                    "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                                    "data": form,
                                    "datatype": "json",
                                     "content": 'application/x-www-form-urlencoded',
                                     "headers": {
                                         "Authorization":tatiyeNet.token,
                                         "Key":row.key,
                                         "userid":tatiyeNet.userid
                                     },
                                    "async": false,
                                     beforeSend: function() {
                                         $("#sendEntri"+model).hide();
                                         $("#endEntri"+model).show();  
                                     },
                                    success: function(add) {
                                              if (add.hasil !== "sukses") {
                                                      $.each(add.error, function( key, value ) {
                                                         var setId =$('#'+key).text();
                                                         var sedLabel=setId+' '+value;
                                                         $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                                                 $("#sendEntri"+model).show();
                                                                 $("#endEntri"+model).hide();
                                                     });
                                                      $.each(add.sukses, function( key, value ) {
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                     
                                              } else {
                                                     $.each(add.sukses, function( key, value ) {
                                                         var label=$('#'+key).text(); 
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                    $("#sendEntri"+model).hide();
                                                    $("#endEntri"+model).show();
                                                    var destroyLoader = setTimeout(function(){
                                                    $("#sendEntri"+model).show();
                                                    $("#endEntri"+model).hide();
                                                      // modal.style.display = "none";
                                                        $("#"+model).html('');
                                                         $(".table").DataTable().ajax.reload( null, false);
                                                        // console.log(this.storage)
                                                     // window.indexOn.setReload();
                                                    // window.app.storage();
                                                 },200);
                                                $('input[type="text"],select, textarea').val('');//kosongkan form
                                                $('input[type="email"],select,form textarea').val('');//kosongkan form
                                                $('input[type="password"], select, textarea').val('');//kosongkan form
                                                $('input[type="radio"]').prop('checked', false); 
                                                $('input[type="checkbox"]').prop('checked', false); 
                                              }
                                      }
                               }); 
                    };

                    let sendEntri =document.querySelectorAll("#sendEntri"+model);
                    sendEntri.forEach(entri => {
                      entri.addEventListener('click', setForumStorage);
                    });
           });
        







    }
}
