// tatiye/From/index.js
 import tatiyeNet,{
  Firebase,
  setDateTime,
  useHandelIDTabel,
  useHandelIDTabelFire,
  itemSpinner
}  from "../es6.js";
  import {privacy,privacyRaw} from "./privacy.js";

export class From {
    constructor(storage, action,content) {
        this.storage = storage
        this.action = action
        this.content = content
      
    }
    // useElement(model,row,content){
    //      setElementFrom(model,row,content);
    // }
    userid(model,row,content,resheader){
          var dirFile='upload';
          var myFrom='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                  // "resheader"   :model
                 beforeSend: function() {
                   $("#bodyFrom"+resheader).html(itemSpinner('modal'));
                 },
                })
               .done(function (template) {
                $("#bodyFrom"+resheader).html(template);
                  $("#form_"+resheader).on('submit', function(e){
                          e.preventDefault();
                              var formData = new FormData($(this)[0]);
                              formData.append('userid',tatiyeNet.userid);
                           $.ajax({
                               type: 'POST',
                               url: tatiyeNet.api+'/v1/userid/'+dirFile+'/appuserprofil/'+row.key,
                               data: formData,
                               contentType: false,
                               cache: false,
                               processData:false,
                               beforeSend: function(){
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();  
                               },
                               success: function(data){
                               /*
                               |--------------------------------------------------------------------------
                               | Initializes title 
                               |--------------------------------------------------------------------------
                               | Develover Tatiye.Net 2020
                               | @Date 
                               */ 
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();
                                 $("#uploadStatus").html(data.status);  
                                var destroyLoader = setTimeout(function(){
                                  $("#"+resheader).html('');
                                     // window.app.storage();
                                },1000);    
                               $("input[type=file]").val('');
                               /*
                               |--------------------------------------------------------------------------
                               | AND title 
                               |--------------------------------------------------------------------------
                               */
                               }
                           });
                      });  
                });
    }


        dompdf(model,row,content,resheader){
          $("#bodyFrom"+resheader).html('<iframe width="100%" height="1300px"src="'+tatiyeNet.routes+'/pdf/'+row.route+'/'+row.az+'/'+row.segment+'?tn='+row.key+'"></iframe>');
        }


        office(model,row,content,resheader){
          var myFrom='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                 beforeSend: function() {
                   $("#bodyFrom"+resheader).html(itemSpinner('modal')); 
                 },
                })
               .done(function (template) {
                $("#bodyFrom"+resheader).html(template);
                  $("#form_"+resheader).on('submit', function(e){
                          e.preventDefault();
                              var formData = new FormData($(this)[0]);
                              formData.append('userid',tatiyeNet.userid);
                           $.ajax({
                               type: 'POST',
                               url: tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                               data: formData,
                               contentType: false,
                               cache: false,
                               processData:false,
                               beforeSend: function(){
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();  
                               },
                               success: function(data){
                               /*
                               |--------------------------------------------------------------------------
                               | Initializes title 
                               |--------------------------------------------------------------------------
                               | Develover Tatiye.Net 2020
                               | @Date 
                               */ 
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();
                                 $("#uploadStatus").html(data.status);  
                                var destroyLoader = setTimeout(function(){
                                  $("#"+resheader).html('');
                                    // reload(this.storage.page);
                                  reload(row.page);
                                  //console.log(row.page)
                                     //window.app.storage();
                                },1000);    
                               $("input[type=file]").val('');
                               /*
                               |--------------------------------------------------------------------------
                               | AND title 
                               |--------------------------------------------------------------------------
                               */
                               }
                           });
                      });  
                });


    }
    upload(model,row,content,resheader){
      // console.log(this.storage)
        if (content=='uploadFile') {
          var dirFile='document';
        } else {
          var dirFile='upload';
        }
          var myFrom='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                 beforeSend: function() {
                  $("#bodyFrom"+resheader).html(itemSpinner('modal'));  
                 },
                })
               .done(function (template) {
                $("#bodyFrom"+resheader).html(template);
                  $("#form_"+resheader).on('submit', function(e){
                          e.preventDefault();
                              var formData = new FormData($(this)[0]);
                              formData.append('userid',tatiyeNet.userid);
                           $.ajax({
                               type: 'POST',
                               url: tatiyeNet.api+'/v1/select/'+dirFile+'/'+row.tabel+'/'+row.key,
                               data: formData,
                               contentType: false,
                               cache: false,
                               processData:false,
                               beforeSend: function(){
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();  
                               },
                               success: function(data){
                               /*
                               |--------------------------------------------------------------------------
                               | Initializes title 
                               |--------------------------------------------------------------------------
                               | Develover Tatiye.Net 2020
                               | @Date 
                               */ 
                                 $("#sendEntriFile"+resheader).hide();
                                 $("#endEntri"+resheader).show();
                                 $("#uploadStatus").html(data.status);  
                                var destroyLoader = setTimeout(function(){
                                  $("#"+resheader).html('');
                                    // reload(this.storage.page);
                                  reload(row.page);
                                  //console.log(row.page)
                                     //window.app.storage();
                                },1000);    
                               $("input[type=file]").val('');
                               /*
                               |--------------------------------------------------------------------------
                               | AND title 
                               |--------------------------------------------------------------------------
                               */
                               }
                           });
                      });  
                });


    }
    dialog(model,row) {
    var myFrom        ='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
          $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                  beforeSend: function() {
                    $("#bodyFrom"+model).html(itemSpinner('modal'));  
                 },
                })
               .done(function (desain) {
                   $("#bodyFrom"+model).html(desain);
                const setForumStorage = event => {
                             let form =$('#form_'+model).serialize();
                             $.ajax({
                                    "type": 'POST',
                                    "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                                    "data": form,
                                    "datatype": "json",
                                     "content": 'application/x-www-form-urlencoded',
                                     "headers": {
                                         "Authorization":tatiyeNet.token,
                                         "Key":row.key,
                                         "userid":tatiyeNet.userid
                                     },
                                    "async": false,
                                     beforeSend: function() {
                                         $("#sendEntri"+model).hide();
                                         $("#endEntri"+model).show();  
                                     },
                                    success: function(add) {
                                              if (add.hasil !== "sukses") {
                                                      $.each(add.error, function( key, value ) {
                                                         var setId =$('#'+key).text();
                                                         var sedLabel=setId+' '+value;
                                                         $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                                                 $("#sendEntri"+model).show();
                                                                 $("#endEntri"+model).hide();
                                                     });
                                                      $.each(add.sukses, function( key, value ) {
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                     
                                              } else {
                                                     $.each(add.sukses, function( key, value ) {
                                                         var label=$('#'+key).text(); 
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                    $("#sendEntri"+model).hide();
                                                    $("#endEntri"+model).show();
                                                    var destroyLoader = setTimeout(function(){
                                                    $("#sendEntri"+model).show();
                                                    $("#endEntri"+model).hide();
                                                      // modal.style.display = "none";
                                                        $("#"+model).html('');
                                                     // window.indexOn.setReload();
                                                     // window.app.storage();

                                                 },200);
                                                $('input[type="text"],select, textarea').val('');//kosongkan form
                                                $('input[type="email"],select,form textarea').val('');//kosongkan form
                                                $('input[type="password"], select, textarea').val('');//kosongkan form
                                                $('input[type="radio"]').prop('checked', false); 
                                                $('input[type="checkbox"]').prop('checked', false); 
                                              }
                                      }
                               }); 
                    };

                    let sendEntri =document.querySelectorAll("#sendEntri"+model);
                    sendEntri.forEach(entri => {
                      entri.addEventListener('click', setForumStorage);
                    });
           });
        


    }



    xlsx(model,row,page=1) {
          var myFrom        ='/helpers/Form/selectFromxlsx';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(row.tableHead);
          $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :row.tableHead,
                  beforeSend: function() {
                    $("#bodyFrom"+model).html(itemSpinner('modal'));  
                 },
                })
               .done(function (desain) {
                   $("#bodyFrom"+model).html(desain);
                   //  let rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
                  // console.log(row) public/package/office/controllers/send/xlsxHeadTabel.php
                   privacy(row);

                    const setForumStorage = event => {
                             let form =$('#form_'+model).serialize();
                             $.ajax({
                                    "type": 'POST',
                                    "url": tatiyeNet.api+'/v1/package/office/controllers/xlsxHeadTabel',
                                    "data": form,
                                    "datatype": "json",
                                     "content": 'application/x-www-form-urlencoded',
                                     "headers": {
                                         "Authorization":tatiyeNet.token,
                                         "Key":row.key,
                                         "from":row.tableHead,
                                         "userid":tatiyeNet.userid
                                     },
                                    "async": false,
                                     beforeSend: function() {
                                         $("#sendEntri"+model).hide();
                                         $("#endEntri"+model).show();  
                                     },
                                    success: function(add) {
                                    
                                              if (add.hasil !== "sukses") {
                                                      $.each(add.error, function( key, value ) {
                                                         var setId =$('#'+key).text();
                                                         var sedLabel=setId+' '+value;
                                                         $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                                                 $("#sendEntri"+model).show();
                                                                 $("#endEntri"+model).hide();
                                                     });
                                                      $.each(add.sukses, function( key, value ) {
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                     
                                              } else {
                                                     $.each(add.sukses, function( key, value ) {
                                                         var label=$('#'+key).text(); 
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                    $("#sendEntri"+model).hide();
                                                    $("#endEntri"+model).show();
                                                    var destroyLoader = setTimeout(function(){
                                                    $("#sendEntri"+model).show();
                                                    $("#endEntri"+model).hide();
                                                      // modal.style.display = "none";
                                                        $("#"+model).html('');
                                                     // window.indexOn.setReload();
                                                     //window.app.storage();
                                                        // if (row.reload==false) {} else {
                                                        //    reload(row.key);
                                                        // }
                                                       
                                                       
                                                 },200);
                                                $('input[type="text"],select, textarea').val('');//kosongkan form
                                                $('input[type="email"],select,form textarea').val('');//kosongkan form
                                                $('input[type="password"], select, textarea').val('');//kosongkan form
                                                $('input[type="radio"]').prop('checked', false); 
                                                $('input[type="checkbox"]').prop('checked', false); 
                                              }
                                      }
                               }); 
                    };

                    let sendEntri =document.querySelectorAll("#sendEntri"+model);
                    sendEntri.forEach(entri => {
                      entri.addEventListener('click', setForumStorage);
                    });
           });
        


    }

    from(model,row,page=1) {

          var myFrom        ='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
          $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                  beforeSend: function() {
                    $("#bodyFrom"+model).html(itemSpinner('modal'));  
                 },
                })
               .done(function (desain) {
                   $("#bodyFrom"+model).html(desain);
                   //  let rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
                  // console.log(row) modal
                   privacy(row);

                    const setForumStorage = event => {
                             let form =$('#form_'+model).serialize();
                             $.ajax({
                                    "type": 'POST',
                                    "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                                    "data": form,
                                    "datatype": "json",
                                     "content": 'application/x-www-form-urlencoded',
                                     "headers": {
                                         "Authorization":tatiyeNet.token,
                                         "Key":row.key,
                                         "userid":tatiyeNet.userid
                                     },
                                    "async": false,
                                     beforeSend: function() {
                                         $("#sendEntri"+model).hide();
                                         $("#endEntri"+model).show();  
                                     },
                                    success: function(add) {
                                              
                                              if (add.hasil !== "sukses") {
                                                      $.each(add.error, function( key, value ) {
                                                         var setId =$('#'+key).text();
                                                         var sedLabel=setId+' '+value;
                                                         $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                                                 $("#sendEntri"+model).show();
                                                                 $("#endEntri"+model).hide();
                                                     });
                                                      $.each(add.sukses, function( key, value ) {
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                     
                                              } else {
                                                     $.each(add.sukses, function( key, value ) {
                                                         var label=$('#'+key).text(); 
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                    $("#sendEntri"+model).hide();
                                                    $("#endEntri"+model).show();
                                                    var destroyLoader = setTimeout(function(){
                                                    $("#sendEntri"+model).show();
                                                    $("#endEntri"+model).hide();
                                                      // modal.style.display = "none";
                                                        $("#"+model).html('');
                                                     // window.indexOn.setReload();
                                                     //window.app.storage();
                                                        if (row.reload==false) {} else {
                                                           reload(row.key);
                                                        }
                                                       
                                                       
                                                 },200);
                                                $('input[type="text"],select, textarea').val('');//kosongkan form
                                                $('input[type="email"],select,form textarea').val('');//kosongkan form
                                                $('input[type="password"], select, textarea').val('');//kosongkan form
                                                $('input[type="radio"]').prop('checked', false); 
                                                $('input[type="checkbox"]').prop('checked', false); 
                                              }
                                      }
                               }); 
                    };

                    let sendEntri =document.querySelectorAll("#sendEntri"+model);
                    sendEntri.forEach(entri => {
                      entri.addEventListener('click', setForumStorage);
                    });
           });
    }


firebase(model,row,action) {
          console.log(row)
          var myFrom        ='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
          $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                  beforeSend: function() {
                     $("#bodyFrom"+model).html(itemSpinner('modal'));  
                 },
                })
               .done(function (desain) {
                   $("#bodyFrom"+model).html(desain);
                 
                    privacy(row);
                    const setForumStorage = event => {
                             let form =$('#form_'+model).serialize();
                             let db=tatiyeNet.firebase.database().ref('exsampel');
                              const d = new Date();
                              let time = d.getTime();
                              var formdata = $('#form_'+model).serializeArray();
                              var dataFire = {};
                              let keys = Object.keys(action);
                              $(formdata ).each(function(index, obj){
                                  $(keys).each(function(indexKey, objKey){
                                     dataFire[objKey] = obj.value;
                                  });
                              });
                         
                               

                             $.ajax({
                                    "type": 'POST',
                                    "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                                    "data": form,
                                    "datatype": "json",
                                     "content": 'application/x-www-form-urlencoded',
                                     "headers": {
                                         "Authorization":tatiyeNet.token,
                                         "Key":row.key,
                                         "userid":tatiyeNet.userid,
                                         "fireid":tatiyeNet.oauth.userid+''+time
                                     },
                                    "async": false,
                                     beforeSend: function() {
                                         $("#sendEntri"+model).hide();
                                         $("#endEntri"+model).show();  
                                     },
                                    success: function(add) {
                                              
                                              if (add.hasil !== "sukses") {
                                                      $.each(add.error, function( key, value ) {
                                                         var setId =$('#'+key).text();
                                                         var sedLabel=setId+' '+value;
                                                         $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                                                 $("#sendEntri"+model).show();
                                                                 $("#endEntri"+model).hide();
                                                     });
                                                      $.each(add.sukses, function( key, value ) {
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                     
                                              } else {
                                                     $.each(add.sukses, function( key, value ) {
                                                         var label=$('#'+key).text(); 
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                    $("#sendEntri"+model).hide();
                                                    $("#endEntri"+model).show();
                                                    var destroyLoader = setTimeout(function(){
                                                    $("#sendEntri"+model).show();
                                                    $("#endEntri"+model).hide();
                                                      // modal.style.display = "none";
                                                        $("#"+model).html('');
                                                     

                                                          if (row.key=='add') {

                                                          let useragent = {
                                                              id:tatiyeNet.oauth.userid+''+time,
                                                              userid:tatiyeNet.oauth.userid,
                                                              avatar:tatiyeNet.oauth.avatar,
                                                              username:tatiyeNet.oauth.name,
                                                              date:setDateTime("EN"),
                                                              time:setDateTime("jam")
                                                          };
                                                           let margeData = Object.assign({}, dataFire,useragent);
                                                            db.child(tatiyeNet.oauth.userid+''+time).set(margeData);
                                                          } else {
                                                             var rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
                                                              let useragentUpdate = {
                                                                  id:rowsetIData.fireid,
                                                                  userid:tatiyeNet.oauth.userid,
                                                                  avatar:tatiyeNet.oauth.avatar,
                                                                  username:tatiyeNet.oauth.name,
                                                                  date:setDateTime("EN"),
                                                                  time:setDateTime("jam")
                                                              };
                                                               let margeDataUp = Object.assign({}, dataFire,useragentUpdate);
                                                                db.child(rowsetIData.fireid).update(margeDataUp);
                                                                
                                                          }

                                                         
                                                     // window.indexOn.setReload();
                                                    
                                                        if (row.reload==false) {} else {
                                                           reload(row.key);
                                                        }
                                                       
                                                       
                                                 },200);
                                                $('input[type="text"],select, textarea').val('');//kosongkan form
                                                $('input[type="email"],select,form textarea').val('');//kosongkan form
                                                $('input[type="password"], select, textarea').val('');//kosongkan form
                                                $('input[type="radio"]').prop('checked', false); 
                                                $('input[type="checkbox"]').prop('checked', false); 
                                              }
                                      }
                               }); 
                    };

                    let sendEntri =document.querySelectorAll("#sendEntri"+model);
                    sendEntri.forEach(entri => {
                      entri.addEventListener('click', setForumStorage);
                    });
           });
    }


 pridRaw(model,row,page=1) {
   console.log(row)
          var myFrom        ='/helpers/Form/selectFrom';
          var actionFrom    =JSON.stringify(this.action);
          var actionStorage =JSON.stringify(this.storage);
          $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 'storage'      :actionStorage,
                  beforeSend: function() {
                     $("#bodyFrom"+model).html(itemSpinner('modal'));  
                 },
                })
               .done(function (desain) {

                   $.post(tatiyeNet.etc,{
                        'etc'          :row.prind,
                        'key'          :row.key,
                        'resheader'    :row.key,
                         beforeSend: function() {
                        },
                    })
                   .done(function (desain) {
                       $("#raw"+model).html(desain);
                       
                    });





                   $("#bodyFrom"+model).html(desain);
                   //  let rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
                   // console.log(rowsetIData.nama)
                   privacyRaw(row);

                   const setForumStorage = event => {
                            printraw()
                             let form =$('#form_'+model).serialize();
                             $.ajax({
                                    "type": 'POST',
                                    "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                                    "data": form,
                                    "datatype": "json",
                                     "content": 'application/x-www-form-urlencoded',
                                     "headers": {
                                         "Authorization":tatiyeNet.token,
                                         "Key":row.key,
                                         "userid":tatiyeNet.userid
                                     },
                                    "async": false,
                                     beforeSend: function() {
                                         $("#sendEntri"+model).hide();
                                         $("#endEntri"+model).show();  
                                     },
                                    success: function(add) {
                                              if (add.hasil !== "sukses") {
                                                      $.each(add.error, function( key, value ) {
                                                         var setId =$('#'+key).text();
                                                         var sedLabel=setId+' '+value;
                                                         $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                                                 $("#sendEntri"+model).show();
                                                                 $("#endEntri"+model).hide();
                                                     });
                                                      $.each(add.sukses, function( key, value ) {
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                     
                                              } else {
                                                     $.each(add.sukses, function( key, value ) {
                                                         var label=$('#'+key).text(); 
                                                         $('#info_'+key).html(' '); 
                                                     });
                                                    $("#sendEntri"+model).hide();
                                                    $("#endEntri"+model).show();
                                                    var destroyLoader = setTimeout(function(){
                                                    $("#sendEntri"+model).show();
                                                    $("#endEntri"+model).hide();
                                                      // modal.style.display = "none";
                                                        $("#"+model).html('');
                                                     // window.indexOn.setReload();
                                                     //window.app.storage();
                                                       
                                                        reload(row.page);
                                                       
                                                 },200);
                                                $('input[type="text"],select, textarea').val('');//kosongkan form
                                                $('input[type="email"],select,form textarea').val('');//kosongkan form
                                                $('input[type="password"], select, textarea').val('');//kosongkan form
                                                $('input[type="radio"]').prop('checked', false); 
                                                $('input[type="checkbox"]').prop('checked', false); 
                                              }
                                      }
                               }); 
                    };

                    let sendEntri =document.querySelectorAll("#sendEntri"+model);
                    sendEntri.forEach(entri => {
                      entri.addEventListener('click', setForumStorage);
                    });
           });
        







    }







}
