// tatiye/From/index.js
import tatiyeNet,{
Firebase,
setDateTime,
useHandelIDTabel,
useHandelIDTabelFire,
itemSpinner,
packageIndex,
getItemStorage,
setItemStorage,
useHandel,
fetchIndex,
vendor,
validasi,
elDOM,
fileType,
queryElements,
indexComponents,
intervalPage,
useHandelSelect,
setDbFirebase
}  from "../es6.js";
import {privacy,privacyRaw} from "./privacy.js";
// import {elementsDOM4} from "../Assets/elementsDOM4.js";

export class From {
constructor(storage, action,content) {
const d = new Date();
let getTime  = d.getTime();
this.storage = storage
this.action = action
this.content = content
this.appid = tatiyeNet.userid+''+getTime

}
// useElement(model,row,content){
//      setElementFrom(model,row,content);
// }
userid(model,row,content,resheader){
var dirFile='upload';
var myFrom='/helpers/Form/selectFrom';
var actionFrom    =JSON.stringify(this.action);
var actionStorage =JSON.stringify(this.storage);
$.post(tatiyeNet.etc,{
'etc'          :myFrom,
'action'       :actionFrom,
'storage'      :actionStorage,
// "resheader"   :model
beforeSend: function() {
 $("#bodyFrom"+resheader).html(itemSpinner('modal'));
},
})
.done(function (template) {
$("#bodyFrom"+resheader).html(template);
$("#form_"+resheader).on('submit', function(e){
        e.preventDefault();
            var formData = new FormData($(this)[0]);
            formData.append('userid',tatiyeNet.userid);
         $.ajax({
             type: 'POST',
             url: tatiyeNet.api+'/v1/userid/'+dirFile+'/appuserprofil/'+row.key,
             data: formData,
             contentType: false,
             cache: false,
             processData:false,
             beforeSend: function(){
               $("#sendEntriFile"+resheader).hide();
               $("#endEntri"+resheader).show();  
             },
             success: function(data){
             /*
             |--------------------------------------------------------------------------
             | Initializes title 
             |--------------------------------------------------------------------------
             | Develover Tatiye.Net 2020
             | @Date 
             */ 
               $("#sendEntriFile"+resheader).hide();
               $("#endEntri"+resheader).show();
               $("#uploadStatus").html(data.status);  
              var destroyLoader = setTimeout(function(){
                $("#"+resheader).html('');
                   // window.app.storage();
              },1000);    
             $("input[type=file]").val('');
             /*
             |--------------------------------------------------------------------------
             | AND title 
             |--------------------------------------------------------------------------
             */
             }
         });
    });  
});
}


dompdf(model,row,content,resheader){
$("#bodyFrom"+resheader).html('<iframe width="100%" height="1300px"src="'+tatiyeNet.routes+'/pdf/'+row.route+'/'+row.az+'/'+row.segment+'?tn='+row.key+'"></iframe>');
}


office(model,row,content,resheader){
    var myFrom='/helpers/Form/selectFrom';
    var actionFrom    =JSON.stringify(this.action);
    var actionStorage =JSON.stringify(this.storage);
    $.post(tatiyeNet.etc,{
    'etc'          :myFrom,
    'action'       :actionFrom,
    'storage'      :actionStorage,
    beforeSend: function() {
     $("#bodyFrom"+resheader).html(itemSpinner('modal')); 
    },
    })
    .done(function (template) {
    $("#bodyFrom"+resheader).html(template);
    $("#form_"+resheader).on('submit', function(e){
        e.preventDefault();
            var formData = new FormData($(this)[0]);
            formData.append('userid',tatiyeNet.userid);
         $.ajax({
             type: 'POST',
             url: tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
             data: formData,
             contentType: false,
             cache: false,
             processData:false,
             beforeSend: function(){
               $("#sendEntriFile"+resheader).hide();
               $("#endEntri"+resheader).show();  
             },
             success: function(data){
             /*
             |--------------------------------------------------------------------------
             | Initializes title 
             |--------------------------------------------------------------------------
             | Develover Tatiye.Net 2020
             | @Date 
             */ 
               $("#sendEntriFile"+resheader).hide();
               $("#endEntri"+resheader).show();
               $("#uploadStatus").html(data.status);  
              var destroyLoader = setTimeout(function(){
                $("#"+resheader).html('');
                  // reload(this.storage.page);
                reload(row.page);
                //console.log(row.page)
                   //window.app.storage();
              },1000);    
             $("input[type=file]").val('');
             /*
             |--------------------------------------------------------------------------
             | AND title 
             |--------------------------------------------------------------------------
             */
             }
         });
    });  
});


}
upload(model,row,content,resheader){
    // console.log(this.storage)
    if (content=='uploadFile') {
    var dirFile='document';
    } else {
    var dirFile='upload';
    }
    var myFrom='/helpers/Form/selectFrom';
    var actionFrom    =JSON.stringify(this.action);
    var actionStorage =JSON.stringify(this.storage);
    $.post(tatiyeNet.etc,{
    'etc'          :myFrom,
    'action'       :actionFrom,
    'storage'      :actionStorage,
    beforeSend: function() {
    $("#bodyFrom"+resheader).html(itemSpinner('modal'));  
    },
    })
    .done(function (template) {
    $("#bodyFrom"+resheader).html(template);
    $("#form_"+resheader).on('submit', function(e){
            e.preventDefault();
                var formData = new FormData($(this)[0]);
                formData.append('userid',tatiyeNet.userid);
             $.ajax({
                 type: 'POST',
                 url: tatiyeNet.api+'/v1/select/'+dirFile+'/'+row.tabel+'/'+row.key,
                 data: formData,
                 contentType: false,
                 cache: false,
                 processData:false,
                 beforeSend: function(){
                   $("#sendEntriFile"+resheader).hide();
                   $("#endEntri"+resheader).show();  
                 },
                 success: function(data){
                 /*
                 |--------------------------------------------------------------------------
                 | Initializes title 
                 |--------------------------------------------------------------------------
                 | Develover Tatiye.Net 2020
                 | @Date 
                 */ 
                   $("#sendEntriFile"+resheader).hide();
                   $("#endEntri"+resheader).show();
                   $("#uploadStatus").html(data.status);  
                  var destroyLoader = setTimeout(function(){
                    $("#"+resheader).html('');
                      // reload(this.storage.page);
                    reload(row.page);
                    //console.log(row.page)
                       //window.app.storage();
                  },1000);    
                 $("input[type=file]").val('');
                 /*
                 |--------------------------------------------------------------------------
                 | AND title 
                 |--------------------------------------------------------------------------
                 */
                 }
             });
        });  
    });
}
dialog(model,row) {
var myFrom        ='/helpers/Form/selectFrom';
var actionFrom    =JSON.stringify(this.action);
var actionStorage =JSON.stringify(this.storage);
$.post(tatiyeNet.etc,{
'etc'          :myFrom,
'action'       :actionFrom,
'storage'      :actionStorage,
beforeSend: function() {
  $("#bodyFrom"+model).html(itemSpinner('modal'));  
},
})
.done(function (desain) {
 $("#bodyFrom"+model).html(desain);
const setForumStorage = event => {
           let form =$('#form_'+model).serialize();
           $.ajax({
                  "type": 'POST',
                  "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                  "data": form,
                  "datatype": "json",
                   "content": 'application/x-www-form-urlencoded',
                   "headers": {
                       "Authorization":tatiyeNet.token,
                       "Key":row.key,
                       "userid":tatiyeNet.userid
                   },
                  "async": false,
                   beforeSend: function() {
                       $("#sendEntri"+model).hide();
                       $("#endEntri"+model).show();  
                   },
                  success: function(add) {
                            if (add.hasil !== "sukses") {
                                    $.each(add.error, function( key, value ) {
                                       var setId =$('#'+key).text();
                                       var sedLabel=setId+' '+value;
                                       $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                               $("#sendEntri"+model).show();
                                               $("#endEntri"+model).hide();
                                   });
                                    $.each(add.sukses, function( key, value ) {
                                       $('#info_'+key).html(' '); 
                                   });
                                   
                            } else {
                                   $.each(add.sukses, function( key, value ) {
                                       var label=$('#'+key).text(); 
                                       $('#info_'+key).html(' '); 
                                   });
                                  $("#sendEntri"+model).hide();
                                  $("#endEntri"+model).show();
                                  var destroyLoader = setTimeout(function(){
                                  $("#sendEntri"+model).show();
                                  $("#endEntri"+model).hide();
                                    // modal.style.display = "none";
                                      $("#"+model).html('');
                                   // window.indexOn.setReload();
                                   // window.app.storage();

                               },200);
                              $('input[type="text"],select, textarea').val('');//kosongkan form
                              $('input[type="email"],select,form textarea').val('');//kosongkan form
                              $('input[type="password"], select, textarea').val('');//kosongkan form
                              $('input[type="radio"]').prop('checked', false); 
                              $('input[type="checkbox"]').prop('checked', false); 
                            }
                    }
             }); 
  };

  let sendEntri =document.querySelectorAll("#sendEntri"+model);
  sendEntri.forEach(entri => {
    entri.addEventListener('click', setForumStorage);
  });
});



}



xlsx(model,row,page=1) {
var myFrom        ='/helpers/Form/selectFromxlsx';
var actionFrom    =JSON.stringify(this.action);
var actionStorage =JSON.stringify(row.tableHead);
$.post(tatiyeNet.etc,{
'etc'          :myFrom,
'action'       :actionFrom,
'storage'      :row.tableHead,
beforeSend: function() {
  $("#bodyFrom"+model).html(itemSpinner('modal'));  
},
})
.done(function (desain) {
 $("#bodyFrom"+model).html(desain);
 //  let rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
// console.log(row) public/package/office/controllers/send/xlsxHeadTabel.php
 privacy(row);

  const setForumStorage = event => {
           let form =$('#form_'+model).serialize();
           $.ajax({
                  "type": 'POST',
                  "url": tatiyeNet.api+'/v1/package/office/controllers/xlsxHeadTabel',
                  "data": form,
                  "datatype": "json",
                   "content": 'application/x-www-form-urlencoded',
                   "headers": {
                       "Authorization":tatiyeNet.token,
                       "Key":row.key,
                       "from":row.tableHead,
                       "userid":tatiyeNet.userid
                   },
                  "async": false,
                   beforeSend: function() {
                       $("#sendEntri"+model).hide();
                       $("#endEntri"+model).show();  
                   },
                  success: function(add) {
                  
                            if (add.hasil !== "sukses") {
                                    $.each(add.error, function( key, value ) {
                                       var setId =$('#'+key).text();
                                       var sedLabel=setId+' '+value;
                                       $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                               $("#sendEntri"+model).show();
                                               $("#endEntri"+model).hide();
                                   });
                                    $.each(add.sukses, function( key, value ) {
                                       $('#info_'+key).html(' '); 
                                   });
                                   
                            } else {
                                   $.each(add.sukses, function( key, value ) {
                                       var label=$('#'+key).text(); 
                                       $('#info_'+key).html(' '); 
                                   });
                                  $("#sendEntri"+model).hide();
                                  $("#endEntri"+model).show();
                                  var destroyLoader = setTimeout(function(){
                                  $("#sendEntri"+model).show();
                                  $("#endEntri"+model).hide();
                                    // modal.style.display = "none";
                                      $("#"+model).html('');
                                   // window.indexOn.setReload();
                                   //window.app.storage();
                                      // if (row.reload==false) {} else {
                                      //    reload(row.key);
                                      // }
                                     
                                     
                               },200);
                              $('input[type="text"],select, textarea').val('');//kosongkan form
                              $('input[type="email"],select,form textarea').val('');//kosongkan form
                              $('input[type="password"], select, textarea').val('');//kosongkan form
                              $('input[type="radio"]').prop('checked', false); 
                              $('input[type="checkbox"]').prop('checked', false); 
                            }
                    }
             }); 
  };

  let sendEntri =document.querySelectorAll("#sendEntri"+model);
  sendEntri.forEach(entri => {
    entri.addEventListener('click', setForumStorage);
  });
});



}

from(model,row,page=1) {

var myFrom        ='/helpers/Form/selectFrom';
var actionFrom    =JSON.stringify(this.action);
var actionStorage =JSON.stringify(this.storage);
$.post(tatiyeNet.etc,{
'etc'          :myFrom,
'action'       :actionFrom,
'storage'      :actionStorage,
beforeSend: function() {
  $("#bodyFrom"+model).html(itemSpinner('modal'));  
},
})
.done(function (desain) {
 $("#bodyFrom"+model).html(desain);
 //  let rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
 privacy(row);
  const setForumStorage = event => {
           let form =$('#form_'+model).serialize();
           $.ajax({
                  "type": 'POST',
                  "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                  "data": form,
                  "datatype": "json",
                   "content": 'application/x-www-form-urlencoded',
                   "headers": {
                       "Authorization":tatiyeNet.token,
                       "Key":row.key,
                       "userid":tatiyeNet.userid
                   },
                  "async": false,
                   beforeSend: function() {
                       $("#sendEntri"+model).hide();
                       $("#endEntri"+model).show();  
                   },
                  success: function(add) {
                    
                            
                            if (add.hasil !== "sukses") {
                                    $.each(add.error, function( key, value ) {
                                       var setId =$('#'+key).text();
                                       var sedLabel=setId+' '+value;
                                       $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                               $("#sendEntri"+model).show();
                                               $("#endEntri"+model).hide();
                                   });
                                    $.each(add.sukses, function( key, value ) {
                                       $('#info_'+key).html(' '); 
                                   });
                                   
                            } else {
                                   $.each(add.sukses, function( key, value ) {
                                       var label=$('#'+key).text(); 
                                       $('#info_'+key).html(' '); 
                                   });
                                  $("#sendEntri"+model).hide();
                                  $("#endEntri"+model).show();
                                  var destroyLoader = setTimeout(function(){
                                  $("#sendEntri"+model).show();
                                  $("#endEntri"+model).hide();
                                    // modal.style.display = "none";
                                      $("#"+model).html('');
                                   // window.indexOn.setReload();
                                   //window.app.storage();
                                      if (row.reload==false) {} else {
                                       //  reload(row.key);
                                      }
                                     
                                     
                               },200);
                              $('input[type="text"],select, textarea').val('');//kosongkan form
                              $('input[type="email"],select,form textarea').val('');//kosongkan form
                              $('input[type="password"], select, textarea').val('');//kosongkan form
                              $('input[type="radio"]').prop('checked', false); 
                              $('input[type="checkbox"]').prop('checked', false); 
                            }
                    }
             }); 
  };

  let sendEntri =document.querySelectorAll("#sendEntri"+model);
  sendEntri.forEach(entri => {
    entri.addEventListener('click', setForumStorage);
  });
});
}

setOauth(model,row,page=1) {
  //  console.log(row.signin)
  // console.log('setOauth')
    // console.log(model)
      var tokentype=fetchIndex(row.segment);
      // const getReload=getItemStorage('onPress_data')
      // console.log(getReload)
      const pressModal=getItemStorage('onPressModal')
      var myFrom        ='/helpers/Form/Action/form';
      var actionVariabel    =this.action;
      var actionFrom    =JSON.stringify(this.action);
      var actionStorage =JSON.stringify(this.storage);
      var setItem =JSON.stringify(row.setItem);
      $.post(tatiyeNet.etc,{
         'etc'          :myFrom,
         'setItem'      :setItem,
         'action'       :actionFrom,
         'storage'      :actionStorage,
      beforeSend: function() {
        $("#bodyFrom"+model).html(itemSpinner('modal'));  
      },
      })
      .done(function (desain) {
       $("#bodyFrom"+model).html(desain);
       const setForumStorage = event => {
            $("#sendEntri"+model).hide();
            $("#endEntri"+model).show();  
            var oauth_name=$('input[name=oauth_name]').val();
            var oauth_uid=$('input[name=oauth_uid]').val();
            var oauth_pws=$('input[name=oauth_pws]').val();
            var oauth_confirmasi=$('input[name=oauth_confirmasi]').val();
            var myHeaders = new Headers();
            myHeaders.append("Content-Type", "application/json");
            if (model=='signup') {
                var raw = JSON.stringify({
                  "fullname":oauth_name,
                  "username": oauth_uid,
                  "password": oauth_pws,
                  "confirmasi":oauth_confirmasi
                });
                var requestOptions = {
                  method: 'POST',
                  headers: myHeaders,
                  body: raw,
                  redirect: 'follow'
                };
                fetch(tatiyeNet.api+"/registrasi", requestOptions)
                  .then(response => response.json())
                  .then(result => {
                     $("#sendEntri"+model).show();
                     $("#endEntri"+model).hide();  
                     if (result.success) {
                   let sendFrom={
                            "oauth_name"          :'success', 
                            "oauth_uid"           :'success',
                            "oauth_pws"           :'success', 
                            "oauth_confirmasi"    :'success',
                    };
                     validasi(sendFrom,"sendEntri"+model);
                       //   var setEl=elDOM('el','alert');
                       // $("#informasi").after(setEl.template);
                       // $("#alert").html("Success Mendaftar silahkan masuk");
                     } else {
                       let sendFrom={
                            "oauth_name"          :'', 
                            "oauth_uid"           :'',
                            "oauth_pws"           :'', 
                            "oauth_confirmasi"    :'', 
                       };    
                       validasi(sendFrom,"sendEntri"+model);
                       // var setEl=elDOM('el','alert');
                       // $("#informasi").after(setEl.template);
                       // $("#alert").html(result.messages[0]);
                     }
 
                   

                   })
                  .catch(error => console.log('error', error));

              } else {
            // LOGIN AKUN

            var raw = JSON.stringify({
              "username": oauth_uid,
              "password": oauth_pws
            });
        

            var requestOptions = {
              method: 'POST',
              headers: myHeaders,
              body: raw,
              redirect: 'follow'
            };

            fetch(tatiyeNet.api+"/login", requestOptions)
              .then(response => response.json())
              .then(result => {
                   $("#sendEntri"+model).show();
                   $("#endEntri"+model).hide();  
                   if (result.success) {
                   let sendFrom={
                         "oauth_uid"    :result.data.email,
                         "oauth_pws"    :result.data.password, 
                    };
                     validasi(sendFrom,"sendEntri"+model);
                     vendor('bootstrap')
                      Cookies.set('hostName',    result.data.sub_domain, { expires: 10000});



           //  if (navigator.geolocation) {
           //    navigator.geolocation.getCurrentPosition(setPosition);
           //  } 
           // function setPosition(position) {
           //    Cookies.set('geolocation',JSON.stringify([position.coords.latitude,position.coords.longitude]), { expires: 10000});
           // }







                      if (row.signin) {
                        if (row.signin =='open') {
                           window.location.href =result.data.sub_domain;
                        } else {
                           window.location.href =tatiyeNet.routes+'/'+row.signin;
                        }
                      } else {
                        location.reload(true);
                      }
                    $("#alert").remove();
                    } else {
                      let sendFrom={
                            "oauth_uid"    :'',
                            "oauth_pws"    :'', 
                       };    
                    validasi(sendFrom,"sendEntri"+model);
                    //  var setEl=elDOM('el','alert');
                    // $("#informasi").after(setEl.template);
                    // $("#alert").html(result.messages[0]);
                    }
             }).catch(error => console.log('error', error));
           }

       };
      let sendEntri =document.querySelectorAll("#sendEntri"+model);
      sendEntri.forEach(entri => {
        entri.addEventListener('click', setForumStorage);
      });
});
}
uploadOutside(model,row,content,resheader){
   // console.log(row)
 // console.log('uploadOutside')

      var myFrom        ='/helpers/Form/Action/form';
      var actionVariabel    =this.action;
      var actionFrom    =JSON.stringify(this.action);
      var actionStorage =JSON.stringify(this.storage);
      var setItem =JSON.stringify(row.setItem);
      $.post(tatiyeNet.etc,{
         'etc'          :myFrom,
         'setItem'      :setItem,
         'action'       :actionFrom,
         'storage'      :actionStorage,
       beforeSend: function() {
            $("#bodyFrom"+resheader).html(itemSpinner('modal'));  
       },
    })

    .done(function (template) {
         $("#bodyFrom"+resheader).html(template);

       if (row.key=='add') {
         
       } else {
           var rowsetIData =useHandelIDTabel(row.tabel+','+row.key);         
              $.each(actionVariabel,function(variabel, set){
                if(set[0]=="images") {
                //  $("#bodyFrom"+row.segment+" #imagePreview").attr("src", tatiyeNet.routes+"/"+rowsetIData[variabel]);
                // } else if(set[0]=="drive") {
                //   var appfile =useHandelIDTabel('appfile,'+row.key);
                //   $("#bodyFrom"+row.segment+" #imagePreview").attr("src", fileType(appfile['fileType']));
                // } else {
                //   $("#bodyFrom"+model+" input[name="+variabel+"]").val(rowsetIData[variabel]);  
                //     // console.log(set) 
                // }
                   $("#bodyFrom"+row.segment+" #imagePreview").attr("src", tatiyeNet.routes+"/assets/images/images.svg");
                 } else if(set[0]=="drive") {
                   $("#bodyFrom"+row.segment+" #imagePreview").attr("src", tatiyeNet.routes+"/assets/images/Drive.png");
                 }
            })
       
       }

         $("#form_"+resheader).on('submit', function(e){
            e.preventDefault();
                 const getItem=getItemStorage('onPress')
                 if (getItem) {
                     var statusKey=getItem.id;
                  } else {
                     var statusKey=row.key;
                  }
                var formData = new FormData(this);
                formData.append('id',statusKey);
                formData.append('userid',tatiyeNet.userid);
             $.ajax({
                 type: 'POST',
                 url: tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                 data: formData,
                 contentType: false,
                 cache: false,
                 processData:false,
                 headers: {
                       "Authorization":tatiyeNet.token,
                       "initkey":row.initkey,
                       
                   },
                 beforeSend: function(){
                     $("#sendEntri"+model).hide();
                     $("#endEntri"+model).show();  
                 },
                 success: function(add){
                    // console.log(add)
                 /*
                 |--------------------------------------------------------------------------
                 | Initializes title 
                 |--------------------------------------------------------------------------
                 | Develover Tatiye.Net 2020
                 | @Date 
                 */ 
                     if (add.hasil !== "sukses") {
                                    $.each(add.error, function( key, value ) {
                                       var setId =$('#'+key).text();
                                       var sedLabel=value;
                                         $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                               $("#sendEntri"+model).show();
                                               $("#endEntri"+model).hide();
                                   });
                                    $.each(add.sukses, function( key, value ) {
                                       $('#info_'+key).html(' '); 
                                   });
                               
                            } else {
                                   $.each(add.sukses, function( key, value ) {
                                       var label=$('#'+key).text(); 
                                       $('#info_'+key).html(' '); 
                                   });
                             

                                 
                                  $("#sendEntri"+model).hide();
                                  $("#endEntri"+model).show();
                                  var destroyLoader = setTimeout(function(){
                                  $("#sendEntri"+model).show();
                                  $("#endEntri"+model).hide();
                              
                                       if (getItem) {
                                             var rowsetIData =useHandelIDTabel(row.tabel+','+getItem.id);
                                             // console.log(rowsetIData)
                                            $("#sendEntri"+model).text(getItem.button);
                                             
                                             setItemStorage({
                                                'item' :'onPress',
                                                'data' :''
                                              }) 
                                        

                                   var entriStatus='Update';
                                  } else {
                                    if (row.key !=='add') {
                                     var entriStatus='Update';
                                    } else {
                                     var entriStatus='Insert';   

                                    }
                                  } 

                                  // console.log(entriStatus)

                                  if (row.page=="multiple") {
                                     $("#"+row.model).remove();
                                  } else {
                                    vendor('bootstrap',500)
                                  }
                                 //  indexComponents(row.component,entriStatus)
                                   // components
                                 if (row.datatables) {
                                        $(row.datatables).DataTable().ajax.reload( null, false);
                                    } else {
                                      reload();  
                                    }
                                  


                                    if (row.format=="drive") {
                                      $("#bodyFrom"+row.segment+" #imagePreview").attr("src", tatiyeNet.routes+"/assets/images/Drive.png");
                                    } else {
                                      $("#bodyFrom"+row.segment+" #imagePreview").attr("src", tatiyeNet.routes+"/assets/images/images.svg");
                                    }


                               },200);
                              $('input[type="file"]').val('');//kosongkan form
                              $('input[type="text"],select, textarea').val('');//kosongkan form
                              $('input[type="email"],select,form textarea').val('');//kosongkan form
                              $('input[type="password"], select, textarea').val('');//kosongkan form
                              $('input[type="radio"]').prop('checked', false); 
                              $('input[type="checkbox"]').prop('checked', false); 
                            }
                 /*
                 |--------------------------------------------------------------------------
                 | AND title 
                 |--------------------------------------------------------------------------
                 */
                 }
             });
        });  
    });
}

fromOutside(model,row,page=1) {   
   // var keyappId=this.appid; 
    // console.log(row)
      var myFrom            ='/helpers/Form/Action/form';
      var actionVariabel    =this.action;
      var actionFrom        =JSON.stringify(this.action);
      var actionStorage     =JSON.stringify(this.storage);
      var setItem           =JSON.stringify(row.setItem);
      var setcomponents     =JSON.stringify(row.components);
      $.post(tatiyeNet.etc,{
         'etc'          :myFrom,
         'setItem'      :setItem,
         'action'       :actionFrom,
         'storage'      :actionStorage,
         'components'   :setcomponents,
      beforeSend: function() {
        $("#bodyFrom"+model).html(itemSpinner('modal'));  
      },
      })
      .done(function (desain) {
       $("#bodyFrom"+model).html(desain);
       // console.log(row.key)

       if (row.key=='add') {
         
       } else {
        if (row.tabel !=="appuserpackage") {
            var rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
            $.each(actionVariabel,function(variabel, set){
              $("#bodyFrom"+model+" input[name="+variabel+"]").val(rowsetIData[variabel]);  
           })    
        }

       
       }
       if (row.tabel !=="appuserpackage") {
         privacy(row);
       }

        //
        const setForumStorage = event => {
           let form =$('#form_'+model).serialize();
           const getItem=getItemStorage('onPress')
           if (getItem) {
              var statusKey=getItem.id;
            } else {
             var statusKey=row.key;
            }
           $.ajax({
                  "type": 'POST',
                  "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                  "data": form,
                  "datatype": "json",
                  "content": 'application/x-www-form-urlencoded',
                   "headers": {
                       "Authorization":tatiyeNet.token,
                       "initkey":row.initkey,
                       "Key":statusKey,
                       "userid":tatiyeNet.userid
                   },
                  "async": false,
                   beforeSend: function() {
                       $("#sendEntri"+model).hide();
                       $("#endEntri"+model).show();  
                   },
                  success: function(add) {
                    // console.log(add)
                            if (add.hasil !== "sukses") {
                                    $.each(add.error, function( key, value ) {
                                       var setId =$('#'+key).text();
                                       var sedLabel=value;
                                       $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                               $("#sendEntri"+model).show();
                                               $("#endEntri"+model).hide();
                                   });
                                    $.each(add.sukses, function( key, value ) {
                                       $('#info_'+key).html(' '); 
                                   });
                               
                            } else {
                                   $.each(add.sukses, function( key, value ) {
                                       var label=$('#'+key).text(); 
                                       $('#info_'+key).html(' '); 
                                   });
                             
                                  $("#sendEntri"+model).hide();
                                  $("#endEntri"+model).show();
                                  var destroyLoader = setTimeout(function(){
                                  $("#sendEntri"+model).show();
                                  $("#endEntri"+model).hide();
                                   if (getItem) {
                                     var rowsetIData =useHandelIDTabel(row.tabel+','+getItem.id);
                                     $("#sendEntri"+model).text(getItem.button);
                                      setItemStorage({
                                         'item' :'onPress',
                                         'data' :''
                                      }) 

                                  var entriStatus='Update';
                                  } else {
                                    if (row.key !=='add') {
                                     var entriStatus='Update';
                                    } else {
                                     var entriStatus='Insert';   
                                    }
                                  } 

                        
         
                                  if (row.page=="multiple") {
                                     $("#"+row.model).remove();
                                  } else {
                                    vendor('bootstrap',500)
                                  }
                                   indexComponents(row.component,entriStatus)
                                   // components
    

                                    if (row.datatables) {
                                        $(row.datatables).DataTable().ajax.reload( null, false);
                                    } else {
                                        if (row.reload !=='false') {
                                           reload(); 
                                         } 
                                    }

                               },200);
                              $('input[type="text"],select, textarea').val('');//kosongkan form
                              $('input[type="email"],select,form textarea').val('');//kosongkan form
                              $('input[type="password"], select, textarea').val('');//kosongkan form
                              $('input[type="radio"]').prop('checked', false); 
                              $('input[type="checkbox"]').prop('checked', false); 
                            }
                    }
             }); 
  };
  let sendEntri =document.querySelectorAll("#sendEntri"+model);
  sendEntri.forEach(entri => {
    entri.addEventListener('click', setForumStorage);
  });
});
}

firebase(model,row,page=1) {
  const dbfireKey=setDbFirebase();
  if (dbfireKey.status) {
      var appfirebase = firebase.initializeApp(dbfireKey.database);
   } else {
      var appfirebase =false;
   }         

          var myFrom            ='/helpers/Form/Action/form';
          var actionVariabel    =this.action;
          var actionFrom        =JSON.stringify(this.action);
          var actionStorage     =JSON.stringify(this.storage);
          var setItem           =JSON.stringify(row.setItem);
          var setcomponents     =JSON.stringify(row.components);
          $.post(tatiyeNet.etc,{
             'etc'          :myFrom,
             'setItem'      :setItem,
             'action'       :actionFrom,
             'storage'      :actionStorage,
             'components'   :setcomponents,
          beforeSend: function() {
            $("#bodyFrom"+model).html(itemSpinner('modal'));  
          },
          })
          .done(function (desain) {
           $("#bodyFrom"+model).html(desain);
           // console.log(row.key)

           if (row.key=='add') {
             
           } else {
            if (row.tabel !=="appuserpackage") {
                var rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
                $.each(actionVariabel,function(variabel, set){
                  $("#bodyFrom"+model+" input[name="+variabel+"]").val(rowsetIData[variabel]);  
               })    
            }

           
           }
           if (row.tabel !=="appuserpackage") {
             privacy(row);
           }

            //
            const setForumStorage = event => {
               let form =$('#form_'+model).serialize();
               const getItem=getItemStorage('onPress')
               if (getItem) {
                  var statusKey=getItem.id;
                } else {
                 var statusKey=row.key;
                }
               $.ajax({
                      "type": 'POST',
                      "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                      "data": form,
                      "datatype": "json",
                      "content": 'application/x-www-form-urlencoded',
                       "headers": {
                           "Authorization":tatiyeNet.token,
                           "Key":statusKey,
                           "userid":tatiyeNet.userid
                       },
                      "async": false,
                       beforeSend: function() {
                           $("#sendEntri"+model).hide();
                           $("#endEntri"+model).show();  
                       },
                      success: function(add) {
                        //console.log(add)
                                if (add.hasil !== "sukses") {
                                        $.each(add.error, function( key, value ) {
                                           var setId =$('#'+key).text();
                                           var sedLabel=value;
                                           $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                                   $("#sendEntri"+model).show();
                                                   $("#endEntri"+model).hide();
                                       });
                                        $.each(add.sukses, function( key, value ) {
                                           $('#info_'+key).html(' '); 
                                       });
                                   
                                } else {
                                       $.each(add.sukses, function( key, value ) {
                                           var label=$('#'+key).text(); 
                                           $('#info_'+key).html(' '); 
                                       });
                                 
                                      $("#sendEntri"+model).hide();
                                      $("#endEntri"+model).show();
                                      var destroyLoader = setTimeout(function(){
                                      $("#sendEntri"+model).show();
                                      $("#endEntri"+model).hide();
                                       if (getItem) {
                                         var rowsetIData =useHandelIDTabel(row.tabel+','+getItem.id);
                                         $("#sendEntri"+model).text(getItem.button);
                                          setItemStorage({
                                             'item' :'onPress',
                                             'data' :''
                                          }) 

                                      var entriStatus='Update';
                                      } else {
                                        if (row.key !=='add') {
                                         var entriStatus='Update';
                                        } else {
                                         var entriStatus='Insert';   
                                        }
                                      } 

                            
             
                                      if (row.page=="multiple") {
                                         $("#"+row.model).remove();
                                      } else {
                                        vendor('bootstrap',500)
                                      }
                                       indexComponents(row.component,entriStatus)
                                       // components
        

                                        if (row.datatables) {
                                            $(row.datatables).DataTable().ajax.reload( null, false);
                                        } else {
                                            //if (row.reload !=='false') {
                                               reload(); 
                                             //} 
                                        }

                                              if (appfirebase) {
                                                  let db=appfirebase.database().ref(row.tabel);
                                                  if (add.data) {
                                                        var addRTDB=useHandelSelect({
                                                             "tabel":row.tabel,
                                                             "where":'appid="'+add.data+'"'
                                                         });
                                                      db.child(add.data).set(addRTDB);
                                                  } else {
                                                         var addRTDB=useHandelSelect({
                                                             "tabel":row.tabel,
                                                             "where":'id='+statusKey,
                                                         });  
                                                      db.child(addRTDB.appid).update(addRTDB);   
                                                  }
                                              }
                                   },200);
                                  $('input[type="text"],select, textarea').val('');//kosongkan form
                                  $('input[type="email"],select,form textarea').val('');//kosongkan form
                                  $('input[type="password"], select, textarea').val('');//kosongkan form
                                  $('input[type="radio"]').prop('checked', false); 
                                  $('input[type="checkbox"]').prop('checked', false); 
                                }
                        }
                 }); 
      };
      let sendEntri =document.querySelectorAll("#sendEntri"+model);
      sendEntri.forEach(entri => {
        entri.addEventListener('click', setForumStorage);
      });
    });
}

pridRaw(model,row,page=1) {
console.log(row)
var myFrom        ='/helpers/Form/selectFrom';
var actionFrom    =JSON.stringify(this.action);
var actionStorage =JSON.stringify(this.storage);
$.post(tatiyeNet.etc,{
'etc'          :myFrom,
'action'       :actionFrom,
'storage'      :actionStorage,
beforeSend: function() {
   $("#bodyFrom"+model).html(itemSpinner('modal'));  
},
})
.done(function (desain) {

 $.post(tatiyeNet.etc,{
      'etc'          :row.prind,
      'key'          :row.key,
      'resheader'    :row.key,
       beforeSend: function() {
      },
  })
 .done(function (desain) {
     $("#raw"+model).html(desain);
     
  });





 $("#bodyFrom"+model).html(desain);
 //  let rowsetIData =useHandelIDTabel(row.tabel+','+row.key);
 // console.log(rowsetIData.nama)
 privacyRaw(row);

 const setForumStorage = event => {
          printraw()
           let form =$('#form_'+model).serialize();
           $.ajax({
                  "type": 'POST',
                  "url": tatiyeNet.api+'/v1/package/'+row.route+'/controllers/send/'+row.segment,
                  "data": form,
                  "datatype": "json",
                   "content": 'application/x-www-form-urlencoded',
                   "headers": {
                       "Authorization":tatiyeNet.token,
                       "Key":row.key,
                       "userid":tatiyeNet.userid
                   },
                  "async": false,
                   beforeSend: function() {
                       $("#sendEntri"+model).hide();
                       $("#endEntri"+model).show();  
                   },
                  success: function(add) {
                            if (add.hasil !== "sukses") {
                                    $.each(add.error, function( key, value ) {
                                       var setId =$('#'+key).text();
                                       var sedLabel=setId+' '+value;
                                       $('#info_'+key).html('<span style="color: #f0506e">'+sedLabel+'</span>'); 
                                               $("#sendEntri"+model).show();
                                               $("#endEntri"+model).hide();
                                   });
                                    $.each(add.sukses, function( key, value ) {
                                       $('#info_'+key).html(' '); 
                                   });
                                   
                            } else {
                                   $.each(add.sukses, function( key, value ) {
                                       var label=$('#'+key).text(); 
                                       $('#info_'+key).html(' '); 
                                   });
                                  $("#sendEntri"+model).hide();
                                  $("#endEntri"+model).show();
                                  var destroyLoader = setTimeout(function(){
                                  $("#sendEntri"+model).show();
                                  $("#endEntri"+model).hide();
                                    // modal.style.display = "none";
                                      $("#"+model).html('');
                                   // window.indexOn.setReload();
                                   //window.app.storage();
                                     
                                      reload(row.page);
                                     
                               },200);
                              $('input[type="text"],select, textarea').val('');//kosongkan form
                              $('input[type="email"],select,form textarea').val('');//kosongkan form
                              $('input[type="password"], select, textarea').val('');//kosongkan form
                              $('input[type="radio"]').prop('checked', false); 
                              $('input[type="checkbox"]').prop('checked', false); 
                            }
                    }
             }); 
  };

  let sendEntri =document.querySelectorAll("#sendEntri"+model);
  sendEntri.forEach(entri => {
    entri.addEventListener('click', setForumStorage);
  });
});








}







}
