// tatiye/From/privacy.js
 import tatiyeNet,{
    useHandelIDTabel,
    setDateTime,
    explode,
    useHandelIDTabelFire,
    useFetchUser
    }  from "../es6.js";

 let count = 0;


export function privacy(row) { 
   var validasiAngka = /^[0-9]+$/;
       var setkeyID = row.key;
       if (row.key !=='add') {
          var titleFooter='Update';
       } else {
           var titleFooter='Simpan'; 
       }
     if (row.footer) {
           var headtitleFooter=row.footer[0];
       } else {
           var headtitleFooter=titleFooter; 
       }

            if (titleFooter =='Update') {
                 let uidkey=explode(row.setdata,'|',1);
                 let setUid=useHandelIDTabel(row.tabel+','+row.key+','+uidkey);
                 if (setUid.userid==tatiyeNet.userid) {

                     $("#sendEntri"+row.model).html(headtitleFooter);
                 } else {
                     if (explode(row.setdata,'|',0)=='privacy') {
                       var mydataSet= $("#headerTitelx").text();

            let dataTime=explode(setUid.time,':',1);
            if (dataTime) {
               var setUidTime=setUid.time;
            } else {
               var setUidTime=setDateTime(setUid.time,'time');
            }
            if (setUid.name==null) {
              var nmSpam='Noname';
            } else {
              var nmSpam=setUid.name;
            }
           var deasin=`
           <div class="col-md-12">
              <div class="list-group-item d-flex align-items-center">
                <img src="${setUid.avatar}" class="wd-40 ht-40 rounded-circle mg-r-15" alt="">
                <div>
                  <span class="pull-right"><i class="tx-danger  ti-info-alt"></i></span>
                  <h6 class="tx-13 tx-inverse tx-semibold mg-b-0">${nmSpam}</h6>
                  <span class="d-block tx-11 text-muted">Data  <b>${mydataSet}</b> 
                  dikelolah oleh ${nmSpam}. history data terkahir 
                  <b>${setDateTime('DIN',setUid.date)} Jam ${setUidTime}</b> 
                   
                  </span>
                </div>
              </div>
              </div>
           `;
            $(".modal-info").html(deasin);
            $("#profil"+row.model).html(deasin);
            $("#sendEntri"+row.model).remove();
            $("#disabled"+row.model).show();  
            $("#privacy").show();  

         } else {
            $("#sendEntri"+row.model).html(headtitleFooter);
         }
      }
  } 
  if (tatiyeNet.userid) {
      var dtsUser=useFetchUser();
  } else {
    $("#body"+row.model).html(clientLogin(row.model))        
  }

  // if (dtsUser.uidkey==3) {
  //   $("#body"+row.model).html(clientBanned(dtsUser))        
  // } 
      
};

export function privacyRaw(row) { 
   var validasiAngka = /^[0-9]+$/;
       var setkeyID = row.key;
       if (row.key !=='add') {

          var titleFooter='Prind';
       } else {
           var titleFooter='Simpan'; 
       }
            if (titleFooter =='Prind') {
                 let uidkey=explode(row.setdata,'|',1);
                 // console.log(uidkey)
                 let setUid=useHandelIDTabel(row.tabel+','+row.key+','+uidkey);
                 if (setUid.userid==tatiyeNet.userid) {
                     $("#sendEntri"+row.model).html(titleFooter);
                 } else {
                     if (explode(row.setdata,'|',0)=='privacy') {
                       var mydataSet= $("#headerTitelx").text();

            let dataTime=explode(setUid.time,':',1);
            if (dataTime) {
               var setUidTime=setUid.time;
            } else {
               var setUidTime=setDateTime(setUid.time,'time');
            }
            if (setUid.name==null) {
              var nmSpam='Noname';
            } else {
              var nmSpam=setUid.name;
            }
           var deasin=`
           <div class="col-md-12">
              <div class="list-group-item d-flex align-items-center">
                <img src="${setUid.avatar}" class="wd-40 ht-40 rounded-circle mg-r-15" alt="">
                <div>
                  <span class="pull-right"><i class="tx-danger  ti-info-alt"></i></span>
                  <h6 class="tx-13 tx-inverse tx-semibold mg-b-0">${nmSpam}</h6>
                  <span class="d-block tx-11 text-muted">Data  <b>${mydataSet}</b> 
                  dikelolah oleh ${nmSpam}. history data terkahir hari
                  <b>${setDateTime('IND',setUid.date)} Jam ${setUidTime}</b> 
                   
                  </span>
                </div>
              </div>
              </div>
           `;
            $("#profil"+row.model).html(deasin);
            $("#sendEntri"+row.model).remove();
            $("#disabled"+row.model).show();  
            $("#privacy").show();
         } else {
            $("#sendEntri"+row.model).html(titleFooter);
         }
      }
  } 
 
  if (tatiyeNet.userid) {
      var dtsUser=useFetchUser();
  } else {
    $("#body"+row.model).html(clientLogin(row.model))        
  }

  if (dtsUser.uidkey==3) {
    $("#body"+row.model).html(clientBanned(dtsUser))        
  } 






};
export function privacy2(row,resheader) {
      let uidkey=explode(row.setdata,'|',1);
      if (uidkey) {
      let setUid=useHandelIDTabel(row.tabel+','+row.key+','+uidkey);
         if (setUid.userid==tatiyeNet.userid) {
         } else {
            if (setUid.name==null) {
              var nmSpam='Noname';
            } else {
              var nmSpam=setUid.name;
            }
            let dataTime=explode(setUid.time,':',1);
            if (dataTime) {
               var setUidTime=setUid.time;
            } else {
               var setUidTime=setDateTime(setUid.time,'time');
            }



           var deasin=`
           <div class="col-md-12 mb-10px">
              <div class="list-group-item d-flex align-items-center">
                <img src="${setUid.avatar}" class="wd-40 ht-40 rounded-circle mg-r-15" alt="">
                <div>
                  <span class="pull-right"><i class="tx-danger  ti-info-alt"></i></span>
                  <h6 class="tx-13 tx-inverse tx-semibold mg-b-0">${nmSpam}</h6>
                  <span class="d-block tx-11 text-muted">Data  <b>${row.titel}</b> 
                  dikelolah oleh ${nmSpam}. history data terkahir hari
                   <b>${setDateTime('IND',setUid.date)} Jam ${setUidTime}</b> 
                  </span>
                </div>
              </div>
              </div>
           `;
            $("#body"+resheader+' p').remove();
            $("#profil"+resheader).html(deasin);
            $("#uidDelete"+resheader).hide();
            $("#disabled"+resheader).show();  
            $("#privacy").show();

         }


      } 

  if (tatiyeNet.userid) {
    var dtsUser=useFetchUser();
  } else {
    $("#body"+resheader+' p').remove()
    $("#profil"+resheader).html(clientLogin(row.model))
    $(".setModal-footer").hide();
  }


  if (dtsUser.uidkey==3) {
    $("#body"+row.model).html(clientBanned(dtsUser))        
  } 


}

export function privacy3(row,resheader) {
      let uidkey=explode(row.setdata,'|',1);
      if (uidkey) {
       let setUid=useHandelIDTabel(row.tabel+','+row.key+','+uidkey);
         if (setUid.userid==tatiyeNet.userid) {
           
         } else {
            if (setUid.name==null) {
              var nmSpam='Noname';
            } else {
              var nmSpam=setUid.name;
            }
            let dataTime=explode(setUid.time,':',1);
            if (dataTime) {
               var setUidTime=setUid.time;
            } else {
               var setUidTime=setDateTime(setUid.time,'time');
            }



           var deasin=`
           <div class="col-md-12 mb-10px">
              <div class="list-group-item d-flex align-items-center">
                <img src="${setUid.avatar}" class="wd-40 ht-40 rounded-circle mg-r-15" alt="">
                <div>
                  <span class="pull-right"><i class="tx-danger  ti-info-alt"></i></span>
                  <h6 class="tx-13 tx-inverse tx-semibold mg-b-0">${nmSpam}</h6>
                  <span class="d-block tx-11 text-muted">Data  <b>${row.titel}</b> 
                  dikelolah oleh ${nmSpam}. history data terkahir hari
                   <b>${setDateTime('IND',setUid.date)} Jam ${setUidTime}</b> 
                  </span>
                </div>
              </div>
              </div>
           `;

            $("#body"+resheader+' p').remove();
            $("#profil"+resheader).html(deasin);
            $("#uidDelete"+resheader).hide();
            $("#disabled"+resheader).show();  
            $("#privacy").show();

         }


      } 
     
  if (tatiyeNet.userid) {
    var dtsUser=useFetchUser();
  } else {
    $("#bodyrecycle"+row.model).html(clientLogin(row.model))
  }


  if (dtsUser.uidkey==3) {
    $("#body"+row.model).html(clientBanned(dtsUser))        
  } 

}

/*
|--------------------------------------------------------------------------
| Initializes clientBanned 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2022
| @Date  
*/
 function clientBanned(row){
    var desain=`
            <div class="col-md-12 list-group-item d-flex align-items-top">
              <img src="${row.avatar}" class="wd-40 ht-40 rounded-circle mg-r-8" alt="">
              <div>
                <h6 class="tx-13 tx-inverse tx-semibold mg-b-0 pt-5px">Akun Banned </h6>
                <span class="d-block tx-12 text-muted mb-10px">
                 <b>${row.name}</b>  akun anda dibekukan  untuk sentara waktu 
                </span>
              </div>
            </div>
    `;
      return desain;
    
}
/* and class clientBanned */
/*
|--------------------------------------------------------------------------
| Initializes client 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2022
| @Date  
*/
 function clientLogin(key){
        count++;
     var resender=key+count;
    var desain=`
        <div class="col-md-12 list-group-item d-flex align-items-top">
          <img src="${tatiyeNet.hostName}/images/admin.jpeg" class="wd-50 ht-50 rounded-circle mg-r-15" alt="">
          <div>
            <h6 class="tx-13 tx-inverse tx-semibold mg-b-0">Oopps. Anda Blum Login</h6>
            <span class="d-block tx-11 text-muted mb-10px">Silahkan masuk atau mendaftar untuk melanjutkan.</span>
            <button onclick="useClient(['${resender}','SignIn']);" type="button" class="btn btn-xs btn-outline-primary">Login</button>
            <button onclick="useClient(['${resender}','SignUp']);" type="button" class="btn btn-xs btn-primary">Mendaftar</button>
          </div>
        </div>
    `;
      return desain;
    
}
/* and class client */
