// tatiye/app.js
import {Modal}  from "./Modal/index.js";
import {From}  from "./From/index.js";
import {FromTables}  from "./From/FromTables.js";
import {setGrid}  from "./Grid/index.js";
import {setGridTabel}  from "./Grid/setGridTabel.js";
import {Tables}  from "./Tables/index.js";
import {readRouter}  from "./Router/index.js";

import {BASE_URL,BASE_API,PATROUTES} from "./tatiyeNetConfig.js";
import 'https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js';
// import 'https://cdnjs.cloudflare.com/ajax/libs/qrious/4.0.2/qrious.min.js';
// import styles from './tatiyeNetStyle.css' assert { type: "css" };
//   document.adoptedStyleSheets = [styles];
if (Cookies.get("uid")) {
    var newtoken=Cookies.get("accessToken");
    var uid=Cookies.get("uid");
    var userid=Cookies.get("user_id");
} else {
    var uid='';
    var userid='';
    var newtoken='';
}
let tatiyeNet = {
      token     :newtoken,
      uid       :uid,
      userid    :userid,
      host      :Cookies.get("hostName"),
      etc       :BASE_URL+'/etc',
      autoload  :BASE_URL+'/autoload',
      package   :BASE_URL+'/package',
      route     :BASE_URL+'/route',
      routes    :BASE_URL,
      hostName  :BASE_URL,
      domain    :BASE_URL,
      api       :BASE_API,
      patroutes :PATROUTES
      
  };

export default tatiyeNet;



export function navLink(main) {
    if (main.status=='dev') {
          var myFrom='/helpers/Form/Router/navLink';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });
    }

} 

export function navLinkPackage(main) {
    if (main.status=='dev') {
          var myFrom='/helpers/Form/Router/navLinkPackage';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });
    }
      window.ClickPackage  = function (key) {
         mainRouter({
            "spinner"    :main.spinner,
            "mainroute"  :main.mainroute,
            "autoload"  :key,
            "patroutes" :1,
         },'package')
      }


     historyRouter({
        "spinner"    :main.spinner,
        "mainroute"  :main.mainroute,
        "explode"   :'#',
        "patroutes" :1,
       },'package')   

} 

export function objectStorage(main) {
      if (main.status=='dev') {
          var myFrom='/helpers/Form/Router/objectStorage';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                //$("#nugi").html(template);
                });
    }
}  
export function navLinkTheme(main) {
    if (main.status=='dev') {
          var myFrom='/helpers/Form/Router/navLinkHashtag';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });
    }
      window.ClickTheme   = function (key) {
         mainRouter({
            "spinner"    :main.spinner,
            "mainroute"  :main.mainroute,
            "autoload"  :key,
            "patroutes" :1,
         })
      }
      historyRouter({
       "spinner"    :main.spinner,
        "mainroute" :main.mainroute,
        "explode"   :'#',
        "patroutes" :1,
      })
} 


export function historyRouter(main,Rout='') { 
   
    let history=$(location).attr('href').split(main.explode).splice(main.patroutes,5).join("/");
    if (history) {
        if (Rout=='package') {
          const myRouter = new readRouter(main);
          var segmenGet=$(location).attr('href').split("/").splice(4,12).join("/");
          return myRouter.package(main,explode(segmenGet,main.explode,0)+'/'+history);
        } else {
           var segmenGet=$(location).attr('href').split("/").splice(3,12).join("/");
           mainRouter({
             "spinner"   :main.spinner,
             "mainroute" :main.mainroute,
             "autoload"  :explode(segmenGet,main.explode,0)+'/'+history,
             "patroutes" :main.patroutes,
          })
        } 
    }

}
export function mainRouter(main,Rout='') { 

    const myRouter = new readRouter(main);
    if (Rout=='package') {
        var segmenGet=main.autoload.split("/").splice(4,5).join("/");
        return myRouter.package(main,segmenGet);  
    } else {
       return myRouter.hashtag(main);
    }
}
  


export function setRouter(main) { 
    const buttonClickHandler = event => {
       
        history.pushState({}, "", event.target.href)
        if (event.target.href !==null) {
           var segmenGet=event.target.href.split("/").splice(3,5).join("/");
       
        const myRouter = new readRouter(main);
        return myRouter.template(main,segmenGet);           
        }

    };
    $.each(main.selector,function(key1, row){
       const button = document.querySelector('#'+row);
       button.addEventListener('click', buttonClickHandler);
    });
}

export function Router(main) { 
 $('a').click(function(){
       var ulr=$(this).attr('href');
       if (ulr !=='javascript:void(0);') {
            history.replaceState('setRhref', '', ulr);
            var segmenGet=$(location).attr('href').split("/").splice(4,5).join("/");
            // console.log(segmenGet)
            const myRouter = new readRouter(main);
            return myRouter.package(main,segmenGet);  
        } 
  });
}

export function setTables(data) { 
  return new Tables(data)
  // console.log(data)
}

export function gridSystem(main) { 
    return setGrid(main)
}

export function gridSystemContent(main) { 
    return setGridTabel(main)
}

export function Qrcode(main,style) { 
    // console.log(main)
    if (style) {
     var qrcode = new QRious({
       element: document.getElementById(style.selector),
       background: '#ffffff',
       backgroundAlpha: 1,
       foreground: style.color?style.color:'#5868bf',
       foregroundAlpha: 1,
       level: 'H',
       padding: 0,
       size: style.size?style.size:128,
       value: main
     });
     return qrcode;
    } else {
     return  new QRious({element: document.getElementById("qrcode"), value: main});
    }
}



export class setHttp {
    get(url) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/'+url)
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }
    select(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/select/'+url, {
                "method": 'POST',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }
    entri(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/entri/'+url, {
                "method": 'POST',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }

    post(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/'+url, {
                "method": 'POST',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }
    put(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/'+url, {
                 "method": 'PUT',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }
    delete(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/'+url, {
                "method": 'DELETE',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(() => resolve('Deleted successfully'))
                .catch(err => reject(err))

        })
    }
}


  let count = 0;
/*
|--------------------------------------------------------------------------
| AND Http 
|--------------------------------------------------------------------------
*/


export function setFrom(type) {
  const myFrom = new From(type.storage,type.action,type.storage,type.storage.content)
  if (type.storage.content=='upload') {
       myFrom.upload(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='uploadFile'){
       myFrom.upload(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='userid'){
       myFrom.userid(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='from'){
       myFrom.from(type.storage.model,type.storage)
  } else {
       myFrom.dialog(type.storage.model,type.storage)
  } 
}


export function setFromTables(type) {
  const myFrom = new FromTables(type.storage,type.action,type.storage,type.storage.content)
  if (type.storage.content=='upload') {
      myFrom.upload(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='uploadFile'){
       myFrom.upload(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='userid'){
       myFrom.userid(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else {
       myFrom.from(type.storage.model,type.storage)
  } 
 
}

/*
|--------------------------------------------------------------------------
| Initializes Modal 
form_demoModal011692702825018
form_demoModal011692702825018
|--------------------------------------------------------------------------
| Develover Tatiye.Net 
| @Date 8/16/2023 6:11:41 PM
*/
export function  Images(imageUrl) {

 
}


export function fileType(ekstensi) {
             if (ekstensi=='docx') {
                var  myIcon='https://img.freepik.com/free-icon/word_318-674258.jpg';
              } else if (ekstensi=='doc') {
                  myIcon='https://img.freepik.com/free-icon/word_318-674258.jpg';
             
              } else if (ekstensi=='pdf') {
                  myIcon='https://cdn-icons-png.flaticon.com/512/179/179483.png';
              } else if (ekstensi=='xlsx') {
                  myIcon='https://img.freepik.com/free-icon/excel_318-566085.jpg';
                
              } else if (ekstensi=='pptx') {
                  myIcon='https://img.freepik.com/free-icon/powerpoint_318-674249.jpg';
             
              } else if (ekstensi=='csv') {
                   myIcon='https://img.freepik.com/free-icon/excel_318-566085.jpg';
              } else if (ekstensi=='odes') {
                   myIcon='https://img.freepik.com/free-icon/excel_318-566085.jpg';
              } else {
                   myIcon='';
              }
             return myIcon;
}
export function modalEntri(body) {
      setModal({
       'titel':explode(body,',',0),
       'width':explode(body,',',1),
       'key':'add',
       'model':'modalBody',
       'route':explode(body,',',3),
       'content':explode(body,',',2),
      });
}
export function modalRoute(header,id,style,content,path,tabel='',model='') {
       setModal({
           'titel':header,
           'width':style,
           'key'  :id,
           'model':model,
           'route':path,
           'tabel':tabel,
           'content':content,
          });
       //console.log(header);
}
export function setModal(style) {
        count++;
        const addModal = new Modal(
            style.titel, 
            style.width, 
            style.key, 
            style.model,
            style.route,
            style.tabel,
            style.content,
            count
            );
        // console.log(addModal)
         var resheader =style.content+style.titel.replace(/[^A-Za-z0-9]/g, '')+count;
         if(style.content == 'profil') {
            
            addModal.profil(style.titel,resheader,style.model)  
         } else if (style.content == 'delete'){
           addModal.delete(style.titel,resheader,style.model)  
         } else if (style.content == 'from'){
           addModal.from(style.titel,resheader,style.model)  
         } else if (style.content == 'upload'){
           addModal.upload(style.titel,resheader,style.model)  
         } else {
           addModal.modal(style.titel,resheader,style.model)  
         }        
}


/*
|--------------------------------------------------------------------------
| AND Modal 
|--------------------------------------------------------------------------
*/
var arrbulanEn  =["January","February","March","April","May","June","July","August","September","October","November","Desember"];
var arrbulan    =["Januari","Februari","Maret","April","Mei","Juni","Juli","Agustus","September","Oktober","November","Desember"];
var hariarray =new Array("Minggu","Senin","Selasa","Rabu","Kamis","Jum'at","Sabtu");
var getlocation=$(location).attr('href').split("/").splice(1).join("/");
var tn        =getlocation.split('/');


export function terbilang(a){
   var bilangan = ['','Satu','Dua','Tiga','Empat','Lima','Enam','Tujuh','Delapan','Sembilan','Sepuluh','Sebelas'];

   // 1 - 11
   if(a < 12){
      var kalimat = bilangan[a];
   }
   // 12 - 19
   else if(a < 20){
      var kalimat = bilangan[a-10]+' Belas';
   }
   // 20 - 99
   else if(a < 100){
      var utama = a/10;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%10;
      var kalimat = bilangan[depan]+' Puluh '+bilangan[belakang];
   }
   // 100 - 199
   else if(a < 200){
      var kalimat = 'Seratus '+ terbilang(a - 100);
   }
   // 200 - 999
   else if(a < 1000){
      var utama = a/100;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%100;
      var kalimat = bilangan[depan] + ' Ratus '+ terbilang(belakang);
   }
   // 1,000 - 1,999
   else if(a < 2000){
      var kalimat = 'Seribu '+ terbilang(a - 1000);
   }
   // 2,000 - 9,999
   else if(a < 10000){
      var utama = a/1000;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%1000;
      var kalimat = bilangan[depan] + ' Ribu '+ terbilang(belakang);
   }
   // 10,000 - 99,999
   else if(a < 100000){
      var utama = a/100;
      var depan = parseInt(String(utama).substr(0,2));
      var belakang = a%1000;
      var kalimat = terbilang(depan) + ' Ribu '+ terbilang(belakang);
   }
   // 100,000 - 999,999
   else if(a < 1000000){
      var utama = a/1000;
      var depan = parseInt(String(utama).substr(0,3));
      var belakang = a%1000;
      var kalimat = terbilang(depan) + ' Ribu '+ terbilang(belakang)+ ' Rupiah ';
   }
   // 1,000,000 -    99,999,999
   else if(a < 100000000){
      var utama = a/1000000;
      var depan = parseInt(String(utama).substr(0,4));
      var belakang = a%1000000;
      var kalimat = terbilang(depan) + ' Juta '+ terbilang(belakang);
   }
   else if(a < 1000000000){
      var utama = a/1000000;
      var depan = parseInt(String(utama).substr(0,4));
      var belakang = a%1000000;
      var kalimat = terbilang(depan) + ' Juta '+ terbilang(belakang);
   }
   else if(a < 10000000000){
      var utama = a/1000000000;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%1000000000;
      var kalimat = terbilang(depan) + ' Milyar '+ terbilang(belakang);
   }
   else if(a < 100000000000){
      var utama = a/1000000000;
      var depan = parseInt(String(utama).substr(0,2));
      var belakang = a%1000000000;
      var kalimat = terbilang(depan) + ' Milyar '+ terbilang(belakang);
   }
   else if(a < 1000000000000){
      var utama = a/1000000000;
      var depan = parseInt(String(utama).substr(0,3));
      var belakang = a%1000000000;
      var kalimat = terbilang(depan) + ' Milyar '+ terbilang(belakang);
   }
   else if(a < 10000000000000){
      var utama = a/10000000000;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%10000000000;
      var kalimat = terbilang(depan) + ' Triliun '+ terbilang(belakang);
   }
   else if(a < 100000000000000){
      var utama = a/1000000000000;
      var depan = parseInt(String(utama).substr(0,2));
      var belakang = a%1000000000000;
      var kalimat = terbilang(depan) + ' Triliun '+ terbilang(belakang);
   }

   else if(a < 1000000000000000){
      var utama = a/1000000000000;
      var depan = parseInt(String(utama).substr(0,3));
      var belakang = a%1000000000000;
      var kalimat = terbilang(depan) + ' Triliun '+ terbilang(belakang);
   }

  else if(a < 10000000000000000){
      var utama = a/1000000000000000;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%1000000000000000;
      var kalimat = terbilang(depan) + ' Kuadriliun '+ terbilang(belakang);
   }

   var pisah = kalimat.split(' ');
   var full = [];
   for(var i=0;i<pisah.length;i++){
    if(pisah[i] != ""){full.push(pisah[i]);}
   }
   var Rp=full.join(' ');

   return Rp;
}
function Encode(argument) {
var decodedStringBtoA =argument;
var encodedStringBtoA = btoa(decodedStringBtoA);
var resheader = encodedStringBtoA.replace(/[^A-Za-z0-9]/g, '');
return resheader;
}

function Decode(argument) {
  var encodedStringAtoB = argument;
  var decodedStringAtoB = atob(encodedStringAtoB);
  return decodedStringAtoB;
}


 function newToken(myData){
       var varjs=JSON.stringify(myData, null, 10);
       var myToken=Encode(varjs);
        return myToken;
  }
  /* and class setToken */
  /*
  |--------------------------------------------------------------------------
  | Initializes getToken 
  |--------------------------------------------------------------------------
  | Develover Tatiye.Net 2022
  | @Date  
  */
   function ngetToken(myData){
        var addset=JSON.parse(Decode(myData)); 
        return addset;
      
  }
  /* and class getToken */

function text_shorten(str, length, ending=''){
    if (length == null) {
      length = 100;
    }
    if (ending == null) {
      ending = '...';
    }
    if (str.length > length) {
      return str.substring(0, length - ending.length) + ending;
    } else {
      return str;
    }
  };
  /*
  |--------------------------------------------------------------------------
  | Initializes title 
  |--------------------------------------------------------------------------
  | Develover Tatiye.Net 2022
  | @Date  
  */

 
  /* and class title */
export function rupiah(num) {
  const numb = num;
  const format = numb.toString().split('').reverse().join('');
  const convert = format.match(/\d{1,3}/g);
  const rupiah = convert.join('.').split('').reverse().join('');
  return rupiah;
}
export function romawi(num) {
  var roman = {
    M: 1000,
    CM: 900,
    D: 500,
    CD: 400,
    C: 100,
    XC: 90,
    L: 50,
    XL: 40,
    X: 10,
    IX: 9,
    V: 5,
    IV: 4,
    I: 1
  }
  var result = '';
  for (var key in roman) {
    if (num == roman[key]) {
      return result +=key;
    }
    var check = num > roman[key];
    if(check) {
      result = result + key.repeat(parseInt(num/roman[key]));
      num = num%roman[key];
    }
  }
 return result;
}
export function substr(text='',str=0,str1=0) {
     var res = text.substring(str,str1);
    return res;
}
export function readmore(text,str) {
  return text_shorten(text,str);
}
export function explode(text,str,str1) {
     var tn =text.split(str);
   return tn[str1];
}
export function expo(num) {
     return tn[num];
}
export function replace(textToReplace,bstr,bstr1) {
  let text = textToReplace; 
let result = text.replace(bstr, bstr1);
return result;
}

export function setDecode(code,key='') {
    var dcodeset=(typeof(code) === 'number' || typeof(code) === "string" && code.trim() !== '') && !isNaN(code);  
   if (dcodeset==true) {
      return Encode(code);
   } else {
      return Decode(code);
   }
}

export function setToken(myData) {
  let nToken=newToken(myData);
   return nToken; 
}

export function getToken(myData) {
   return ngetToken(myData);
}




export function setDateTime(dy,flag='',mydate='') {

   // if(OPENID_ID == 'st') {
  var tw = new Date();
  // if (tw.getTimezoneOffset() == 0) (a=tw.getTime() + ( 7 *60*60*1000))
  // else (a=tw.getTime());
  // tw.setTime(a);
 var tahun= tw.getFullYear ();
 var hari= tw.getDay ();
 var bulan= tw.getMonth();
 var tanggal= tw.getDate ();
 var hours = tw.getHours();
 var minutes = "0" + tw.getMinutes();
// Seconds part from the timestamp
var seconds = "0" + tw.getSeconds();
 var hariarray=new Array("Minggu","Senin","Selasa","Rabu","Kamis","Jum'at","Sabtu");
 var bulanarray=new Array("Januari","Februari","Maret","April","Mei","Juni","Juli","Agustus","September","Oktober","November","Desember");
 var bln=bulan+1;
 var ampmd = hours >= 12 ? 'PM' : 'AM';
 var tedTime = hours + ':' + minutes.substr(-2) + ':' + seconds.substr(-2)+' '+ampmd;
 var dcodeset=(typeof(dy) === 'number' || typeof(dy) === "string" && dy.trim() !== '') && !isNaN(dy);  
         if (dcodeset==true) {
            var seTime='number';
         } else {
            var seTime='text';
         }
 if(flag !== '') {
      var month =flag.split('/');
     var getMonth=arrbulan[month[1]-1];
     var getDetmOnd=  arrbulanEn[month[1]-1]+' '+month[2]+', '+month[0];
     var tw = new Date(getDetmOnd);
     var hari= tw.getDay();
  
      if (dy =='IN') {
         return month[2]+' '+getMonth+' '+month[0]; 
      } else if (dy=='EN') {
         return month[0]+' '+getMonth+' '+month[2]; 
      } else if (dy=='Bln') {
           return month[1]; 
      } else if (dy=='Year') {
          return month[0]; 
      } else if (dy=='Month') {
          return getMonth; 
      } else if (dy=='Day') {
         return  hariarray[hari]; 
      } else if (dy=='IND') {
         return  hariarray[hari]+', '+month[2]+' '+getMonth+' '+month[0]; 
      } else {
        if (seTime =='number') {
                let unix_timestamp = dy;
                var date = new Date(unix_timestamp * 1000);
                // Hours part from the timestamp
                var hours = date.getHours();
                // Minutes part from the timestamp
                var minutes = "0" + date.getMinutes();
                // Seconds part from the timestamp
                var seconds = "0" + date.getSeconds();

                 var hari   = date.getDay ();
                 var bulan  = date.getMonth ();
                 var tanggal= date.getDate ();
                 var tahun  = date.getFullYear ();
                 var ampm = hours >= 12 ? 'PM' : 'AM';

                // Will display time in 10:30:23 format
                var formattedTime =tanggal+" "+arrbulan[bulan]+" "+tahun;
                var formatgetDay =hariarray[hari]+", "+tanggal+" "+arrbulan[bulan]+" "+tahun;
                var tedTime = hours + ':' + minutes.substr(-2) + ':' + seconds.substr(-2)+' '+ampm;
                    if (flag=='dmy') {
                       return formatgetDay;
                    } else if (flag=='v') {
                       return formatgetDay+', '+tedTime;   
                    } else if (flag=='time') {
                         return tedTime;  
                    } else if (flag=='Day') {
                        return hariarray[hari];
                    } else if (flag=='Month') {
                        return arrbulan[bulan];
                    } else if (flag=='Year') {
                         return tahun;
                    } else {
                       return formattedTime;
                    }

        } else {      
          return  month[2]+' '+getMonth+' '+month[0]; 
        }
      }

 } else {
    if (bln ==10) {
      var ampm = bln;
    } else if (bln==11) {
        var ampm = bulan;
    } else if (bln==12) {
        var ampm = bln;
    } else {
         var ampm = '0'+bln;
    }
    if (tanggal < 10) {
        var mytgl = '0'+tanggal;
    } else {
        var mytgl = tanggal;
    }

    if (dy=='hari') {
        return hariarray[hari];
     } else if (dy=='bulan') {
        return bulanarray[bulan];
     } else if (dy=='tahun') {
        return tahun;
     } else if (dy=='jam') {
        return tedTime;
     } else if (dy=='EN') {
        return tahun+'/'+ampm+'/'+mytgl;
     } else if (dy=='IN') {
        return mytgl+'/'+ampm+'/'+tahun;
    } else if (dy=='DIN') {
        return mytgl+' '+bulanarray[bulan]+' '+tahun;
     } else {
           return hariarray[hari]+", "+mytgl+" "+bulanarray[bulan]+" "+tahun;
     }
 }
}
