// tatiye/es6.js
import {Modal}  from "./Modal/index.js";
import {From}  from "./From/index.js";
import {FromTables}  from "./From/FromTables.js";
import {setGrid}  from "./Grid/index.js";
import {setGridTabel}  from "./Grid/setGridTabel.js";
import {Tables}  from "./Tables/index.js";
import {readRouter}  from "./Router/index.js";
import {routerLink}  from "./Router/routerLink.js";
import {Assets}  from "./Assets/index.js";
import {share}  from "./Assets/shareLink.js";
import {Client}  from "./Oauth/Client.js";
import {spinner,spinnerModal} from "./Router/spinner.js";
import {BASE_URL,BASE_API,PATROUTES,SPINNER,firebaseConfig} from "./tatiyeNetConfig.js";
import 'https://www.gstatic.com/firebasejs/9.12.1/firebase-app-compat.js';
import 'https://www.gstatic.com/firebasejs/9.12.1/firebase-database-compat.js';
// import 'shortcuts.js';
if (firebaseConfig) {
   var appfirebase = firebase.initializeApp(firebaseConfig);
} else {
  var appfirebase =false;
}

// import 'https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js';
// import 'https://cdnjs.cloudflare.com/ajax/libs/qrious/4.0.2/qrious.min.js';
 // import styles from './ngorei.css' assert { type: "css" };
//   document.adoptedStyleSheets = [styles];
if (Cookies.get("sso")) {
    var newtoken=Cookies.get("sso");
    var row=getToken(newtoken);
    var accessToken=row.token;
    var uid=row.uid;
    var userid=row.userid;
    var appSSO=row;
} else {
    var appSSO='';
    var uid='';
    var userid='';
    var newtoken='';
    var accessToken='';
}
var getLocation=window.location.origin;
let tatiyeNet = {
      token     :accessToken,
      uid       :uid,
      userid    :userid,
      host      :Cookies.get("hostName"),
      etc       :getLocation+'/etc',
      autoload  :getLocation+'/autoload',
      package   :getLocation+'/package',
      route     :getLocation+'/route',
      assets    :getLocation+'/assets',
      apps      :getLocation+'/webview',
      mobile    :getLocation+'/mobile',
      api       :getLocation+'/api',
      routes    :getLocation,
      hostName  :getLocation,
      domain    :getLocation,
      spinner   :SPINNER,
      patroutes :PATROUTES,
      oauth     :appSSO,
      firebase  :appfirebase
      
  };

export default tatiyeNet;
export function storLocation(key='') {
    if (key) {
        return window.location[key]
    } else {
        return window.location
    }
  
}


let useLisensiApi='https://tatiye.net';
export function createLicense() {
let mydata  = [];
$.ajax({
    url:  tatiyeNet.hostName+"/api/qrGgrrxm2xu9LjtXCooLeL6gPiNrY5SCKvNtHb66EZKF1OBwmfxq",
       
    headers: {
      "userid":tatiyeNet.userid,
      "Content-Type": "application/json"
    },
    method: 'POST',
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  }); 
  return mydata;
}




 export function shareLink(key){
   return  share(key);
 }
export function framework7Async(main){
     let dabatabse=useHandel({
        "token"    :main.token,
        "storage"       :{
           "limit": main.storage.limit,
           "page":main.storage.page,
           "keywords":main.storage.keywords
        }
     });
     const item= setItemStorage({
       'item' :'async',
       'data' :dabatabse
     })
     return dabatabse;
 }

export function framework7Object(key){

    let results=setSplit(1,'://',key[1])
    if (results) {
      var baseURL=key[1];
    } else {
        var baseURL=tatiyeNet.hostName+'/'+key[1];
    }

  var mainRouter=setItemStorage({
    'item' :'routes',
    'data' :{
       'titel':key[2],
       'path':key[0],
       'url':baseURL
    }
  })
 

         $.ajax({
                type: 'POST',
                 url: tatiyeNet.hostName+'/framework7',
                 data: 'package='+key[0]+'&key=0'+'&url='+key[1],
                 beforeSend: function() {
                   //$("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                  $("#ngorei").html(data);
                   // history.replaceState(null, null, '');
               }
           });

}
export function framework7Autoload(){
         var getSegment=getItemStorage('routes');
         
         $.ajax({
                type: 'POST',
                 url: tatiyeNet.hostName+'/framework7',
                 data: 'package='+getSegment.path+'&key=0'+'&url='+getSegment.url,
                 beforeSend: function() {
                   //$("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                  $("#ngorei").html(data);
                   // history.replaceState(null, null, '');
               }
           });
         return getSegment;



}
// framework7Autoload()
export function framework7Components(components){
        let mydata ='';
        $.ajax({
            url:tatiyeNet.routes+'/assets/mobilenavigasi.json',
            method: 'GET',
            datatype: "json",
            async: false,
            success: function(data){
                mydata=data;  
            }
          });
         return mydata; 

}
export function framework7Routes(key=''){
  var mainRouter=setItemStorage({
    'item' :'routes',
    'data' :{
       'titel':key.titel,
       'path':key.path,
       'url':key.dir
    }
  })

         $.ajax({
                type: 'POST',
                 url: tatiyeNet.hostName+'/framework7',
                 data: 'package='+key.path+'&key=0'+'&url='+key.dir,
                 beforeSend: function() {
                   //$("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                  $("#"+key.id).html(data);
                   // history.replaceState(null, null, '');
               }
           });
  var getSegment=getItemStorage('routes');
return getSegment;
    // const getItem=getItemStorage('f7router-view_0-history')
// console.log(localStorage.getItem('f7router-view_0-history'));
}

 export function clientLogin(key){
    count++;
    var resender=key+count;
    var desain=`
        <div  class="col-md-12 list-group-item1 d-flex align-items-top">
          <img src="${tatiyeNet.hostName}/images/admin.jpeg" class="wd-50 ht-50 rounded-circle mg-r-15" alt="">
          <div>
            <h6 class="tx-13 tx-inverse tx-semibold mg-b-0">Oopps. Anda Blum Login</h6>
            <span class="d-block tx-11 text-muted mb-10px">Silahkan masuk atau mendaftar untuk melanjutkan.</span>
            <button onclick="useClient(['${resender}','SignIn']);" type="button" class="btn btn-xs btn-outline-primary">Login</button>
            <button onclick="useClient(['${resender}','SignUp']);" type="button" class="btn btn-xs btn-primary">Mendaftar</button>
          </div>
        </div>
    `;
    $("#"+key).html(desain);
}
 window.useClient=function (property) {
    //  let count = 0;
    console.log(property[1]+property[0])
    // $("#"+property[0]).remove();
        setModal({
           'titel':property[1],
           'width':'400px',
           'key'  :'add',
           'model':property[1]+property[0],
           'route':'devices',
           'tabel':'apparchive',
           'content':'fromAction',
           'element':"get",
           'page'   :1,
           'setdata' :'public',
          });
        Client(property,property[1]+property[0])
 }



export function useLisensi() {
let mydata  = [];
let mydata2 = [];
$.ajax({
    url:  tatiyeNet.hostName+"/api/license",
    method: 'GET',
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
    $.ajax({
         url: useLisensiApi+"/api/license/"+mydata.key,
        method: 'GET',
        datatype: "json",
        async: false,
        success: function(data){
           
            if (data.status=='active') {
                 Cookies.set('trial', data.sso, { expires: 10000});
            } else if (data.status=='expair') {
                Cookies.set('trial', data.sso, { expires: 10000});
            } else {
                 Cookies.set('trial', '', { expires: 10000});

            }
            // console.log(data.status)
                 var add=JSON.stringify(data, null, 10);
                 mydata2 = JSON.parse(add);      
        }
      });
    return mydata2;
}
// useLisensi()

export function setLink (e,main='') {
  return  routerLink(e,main);
}
export function Firebase (e) {
    // if (appSSO.userid) {
            let db=appfirebase.database().ref(e.tabel[0]);
            if (e.tabel[1]=='insert') {
              const d = new Date();
              let time = d.getTime();

              let useragent = {
                  id:appSSO.userid+''+time,
                  userid:appSSO.userid,
                  avatar:appSSO.avatar,
                  username:appSSO.name,
                  date:setDateTime("EN"),
                  time:setDateTime("jam")
              };

             let margeData = Object.assign({}, e.data,useragent);

              db.child(time).set(margeData);
            } else if (e.tabel[1]=='update') {
                db.on("value", function (select) {
                    select.forEach(function(row){
                        var post = row.val();
                        if (post.id==e.data.id) {
                            db.child(row.key).update(e.data);
                        }
                    });
                });
            } else  if (e.tabel[1]=='delete') {
                db.on("value", function (select) {
                    select.forEach(function(row){
                        var post = row.val();
                        if (post.id==e.id) {
                            db.child(row.key).remove();
                        }
                    });
                });  
            } else  if (e.tabel[1]=='select') {    
                 return db;
            }
    // } else {
    //   return false;
    // }

   
}



let db;
const dbRequest = indexedDB.open('ngorey', 1);
dbRequest.onsuccess = function(event) {
  db = event.target.result;
};

dbRequest.onupgradeneeded = function(event) {
  db = event.target.result;

  const objStore = db.createObjectStore('data', { keyPath: 'id' });

  objStore.transaction.oncomplete = function(event) {
    const productsStore = db
      .transaction('data', 'readwrite')
      .objectStore('data');
    productsStore.add({
      id: 'index',
      baseLink: BASE_URL,
      created: new Date(),
      tags: ['Expensive', 'Luxury']
    });
  };
};

dbRequest.onerror = function(event) {
  console.log('ERROR!');
};



export function userAgent (e) {
let mydata = [];
var agent={
      'indexOn':e.indexOn,
      'href':storLocation('href')  //Online ,Offline
    }
$.ajax({
    url: tatiyeNet.api+"/v1/select/userAgent",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json"
    },
    method: 'POST',
    data: JSON.stringify(agent),
            datatype: "json",
            async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
return mydata
}
 let dataset=userAgent({
      'indexOn':'Desktop'
    })
// console.log(dataset)
/*
|--------------------------------------------------------------------------
| Initializes localStorage 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2022
| @Date 10/20/2023 11:07:05 PM 
*/
export function setItemStorage(row){
      localStorage.setItem(row.item, JSON.stringify(row.data));
      return row;
}

export function getItemStorage(row){
      const extracted= JSON.parse(localStorage.getItem(row));
      return extracted;
}
/* and class localStorage */
/*
|--------------------------------------------------------------------------
| Initializes Cookies 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2022
| @Date 10/20/2023 11:07:05 PM 
*/
export function setItemCookie(row){
      Cookies.set(row.item,JSON.stringify(row.data), { expires: 10000});
      return row;
}
export function getItemCookie(row){
        var useData=Cookies.get(row);
        var  JSONdata = JSON.parse(useData);
      return JSONdata;
}
/* and class Cookie */
/*
|--------------------------------------------------------------------------
| Initializes IndexedDB 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2022
| @Date 10/20/2023 11:07:05 PM 
*/
export function setIndexedDB(row){
  if (!db) {
    return;
  }
  const productsStore = db
    .transaction('data', 'readwrite')
    .objectStore('data');
    if (row.item=='put') {
      productsStore.put(row.data);
      return row;
    } else if (row.item=='get') {
        const request = productsStore.get(row.data);
        return request;
    } else {
      productsStore.add(row.data);
      return row;
    }
}
/* and class IndexedDB */
/*
|--------------------------------------------------------------------------
| Initializes useHandelSave 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2018
| @Date 9/2/2023 3:51:00 PM
*/
export function useHandel(e) {
let mydata = [];
$.ajax({
    url: tatiyeNet.api+"/v2/"+e.token,
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(e.storage),
            datatype: "json",
            async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
return mydata

}

export function bdColor(key) {
     var color={
      '0'  :'bd-pink',
      '1'  :'bd-primary',
      '2'  :'bd-teal',
      '3'  :'bd-success',
      '4'  :'bd-warning',
      '5'  :'bd-danger',
      '6'  :'bd-info',
      '7'  :'bd-dark',
      '8'  :'bd-white',
      '9'  :'bd-indigo',
      '10' :'bd-purple',
      '11' :'bd-success',
      '12' :'bd-warning',
      '13' :'bd-danger',
      '14' :'bd-info',
      '15' :'bd-dark',
      '16' :'bd-white',
      '17' :'bd-pink',
      '18' :'bd-indigo',
      '19' :'bd-purple',
      '20' :'bd-teal',
      '21' :'bd-primary',
    };
    return color[key];

}
export function bgColor(key) {
        var color={
         '0'  :'bg-pink',
         '1'  :'bg-primary',
         '2'  :'bg-teal',
         '3'  :'bg-success',
         '4'  :'bg-warning',
         '5'  :'bg-danger',
         '6'  :'bg-info',
         '7'  :'bg-dark',
         '8'  :'bg-white',
         '9'  :'bg-indigo',
         '10' :'bg-purple',
         '11' :'bg-success',
         '12' :'bg-warning',
         '13' :'bg-danger',
         '14' :'bg-info',
         '15' :'bg-dark',
         '16' :'bg-white',
         '17' :'bg-pink',
         '18' :'bg-indigo',
         '19' :'bg-purple',
         '20' :'bg-teal',
         '21' :'bg-primary',
    };
    return color[key];
}
export function createModules(getAttribute) {
      var jsx = document.createElement("script");
      jsx.setAttribute("src", tatiyeNet.routes+'/modules/'+getAttribute);
      jsx.setAttribute("type", "module");
      document.head.appendChild(jsx);    

}

/*
|--------------------------------------------------------------------------
| Initializes headMeta 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2018
| @Date 9/2/2023 3:51:00 PM
*/
export function headMeta(key) {
    if (key=='') {
        var meta='Home';
    } else {
        var meta=key
    }
     $("title").html(meta);
 
}
export function useTabPages(ssltabel,limit) {
     let tabel =openSSLKey(ssltabel);
     const count=useHandelPages(tabel,limit);
     return count.peging
 
}
/*
|--------------------------------------------------------------------------
| Initializes useHandelID 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2018
| @Date 9/2/2023 3:51:00 PM
*/


export function useHandelIDTabel(e) {
 let tabel=explode(e,',',0);
 let id=explode(e,',',1);
 let key=explode(e,',',2);

let mydata = [];
var settings = {
       "tabel":tabel,
       "id":id,
       "expo":key,
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/useHandelIDTabel",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json"
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}

export function useHandelIDTabelFire(e) {
 let tabel=explode(e,',',0);
 let id=explode(e,',',1);
 let key=explode(e,',',2);

let mydata = [];
var settings = {
       "tabel":tabel,
       "id":id,
       "expo":key,
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/useHandelIDTabelFire",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}



export function useHandelID(e) {
 let tabel=explode(e,',',0);
 let id=explode(e,',',1);

let mydata = [];
var settings = {
       "tabel":tabel,
       "id":id,
       "expo":'id',
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/useHandelID",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}

export function useHandelCount(e) {
 let tabel=explode(e,',',0);
 let id=explode(e,',',1);

let mydata = [];
var settings = {
       "tabel":tabel,
       "syntax":id,
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/useHandelCount",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}


export function useHandelIDKey(e) {
 let tabel=explode(e,',',0);
 let id=explode(e,',',1);
 let syntax=explode(e,',',2);

let mydata = [];
var settings = {
       "tabel"  :tabel,
       "id"     :id,
       "syntax" :syntax,
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/useHandelIDKey",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}
export function useHandelSheet(e) {
 let tabel=explode(e,',',0);
 let id=explode(e,',',1);
 let syntax=explode(e,',',2);

let mydata = [];
var settings = {
       "tabel"  :tabel,
       "id"     :id,
       "syntax" :syntax,
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/useHandelSheet",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}

export function useHandelSheetkey(e) {
    // console.log(e)
 let tabel=explode(e,',',0);
 let id=explode(e,',',1);
 let syntax=explode(e,',',2);

let mydata = [];
var settings = {
       "tabel"  :tabel,
       "id"     :id,
       "syntax" :syntax,
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/useHandelSheetkey",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}


export function openSSLKey(e) {
 let tabel=e;
let mydata = [];
var settings = {
       "tabel"  :e
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/openSSLKey",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}

export function useFetchUser() {
let mydata = [];
var settings = {
      "userid":tatiyeNet.userid
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/useFetchUser",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}
export function useHandelPages(tabel,limit) {
let mydata = [];
var settings = {
      "tabel":tabel,
      "limit":limit,
      "page":"1",
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/pages",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}

export function useTabelToken(tabel) {
let mydata = [];
var settings = {
      "token":tabel
    };
$.ajax({
    url: tatiyeNet.api+"/v1/select/token",
    headers: {
      "Authorization":tatiyeNet.token,
      "Content-Type": "application/json",
      
    },
    method: 'POST',
    data: JSON.stringify(settings),
    datatype: "json",
    async: false,
    success: function(data){
             var add=JSON.stringify(data, null, 10);
             mydata = JSON.parse(add);      
    }
  });
  return mydata
}

/*
|--------------------------------------------------------------------------
| Initializes useHandelHttp 
|--------------------------------------------------------------------------
| Develover Tatiye.Net 2018
| @Date 9/2/2023 3:51:00 PM
*/
export function useHandelHttp(e) {
    var myHeaders = new Headers();
    myHeaders.append("Authorization",e.authorization);
    myHeaders.append("key", e.primarykey);
    myHeaders.append("userid", e.userid);
    myHeaders.append("Content-Type", "application/json");
    
    var requestOptions = {
      method:  e.method,
      headers: myHeaders,
      body: JSON.stringify(e.storage),
      redirect: 'follow'
    };
    let Entri ={
     "url"         :tatiyeNet.api+"/v2/"+e.token,
     "request"    :requestOptions,
    };
     return Entri;
}



export function elementsRender(main) {
     const app = document.querySelector('app-ngorei');
     if ($(app).attr('elements')) {
            const item= setItemStorage({
                'item' :'elements',
                'data' :main
              })
     }
}
export function objectStorage(main) {
      if (main.status=='dev') {
          var myFrom='/helpers/Form/Router/objectStorage';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                //$("#nugi").html(template);
                });
    }
}
export function mainRouter(main,Rout='') { 
    const myRouter = new readRouter(main);
    if (Rout=='package') {
        var segmenGet=main.autoload.split("/").splice(4,5).join("/");
        return myRouter.package(main,segmenGet);  
    } else {
        
       return myRouter.hashtag(main);
    }
}
  
export function routePath(myStack,main='mainRoute',spin='') {
    if (spin=='modal') {
         var spinId=spinnerModal(tatiyeNet.spinner);
    } else {
         var spinId=spinner(tatiyeNet.spinner);
    }
      $.ajax({
            type: 'POST',
             url: tatiyeNet.routes+'/resquire/'+myStack,
             data: 'key='+myStack,
             beforeSend: function() {
              $("#"+main).html(spinId);    
            },
            success: function(data) {
                $("#"+main).html(data);   
           }
       });
}  
export function itemSpinner(key='') {
    if (key=='modal') {
      return spinnerModal(tatiyeNet.spinner);
    } else {
      return spinner(tatiyeNet.spinner);
    }
   
}

export function sorTable(main) {

   $(main.atribut).sortable({ opacity: 0.6, cursor: 'move', update: function() {
        var getIdGrid = $(this).sortable("toArray"); 
        var actionFrom =JSON.stringify(getIdGrid, null, 10);
        var addset = JSON.parse(actionFrom); 
       $.ajax({
                type: 'POST',
                 url: tatiyeNet.routes+'/resquire/pec/devices/controllers/send/dragSend.php',
                 data: 'order='+actionFrom+'&tabel='+main.package,
                 beforeSend: function() {   
                },
                success: function(data) {
                    console.log(data)
               }
           }); 
        }                                 
        });
}
export function createRouter(main) {
    // console.log(main)
    if (main.status=='dev') {
         if (main.public=='theme') {
              var myFrom='/helpers/Form/Router/navLink';
              var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });  

         } else if (main.public=='simulator') {
              var myFrom='/helpers/Form/Router/navLink';
              var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });   
         } else if (main.public=='package') {
          var myFrom='/helpers/Form/Router/navLinkPackage';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });
         }
         if (main.hashtag) {
          var myFrom='/helpers/Form/Router/hashtag';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });
         }

    }
          setItemStorage({
            'item' :'domroutes',
            'data' :{
                "spinner"    :main.spinner,
                "mainroute"  :main.mainroute,
                "patroutes" :main.patroutes,
            }
          })

      window.useClickHashtag   = function (key,rotes='') {
        if (rotes) {
            var mainroute=rotes;
        } else {
            var mainroute=main.mainroute;
        }
         mainRouter({
            "spinner"    :main.spinner,
            "mainroute"  :mainroute,
            "autoload"  :key,
            "patroutes" :main.patroutes,
         })
      }


      historyRouter({
       "spinner"    :main.spinner,
        "mainroute" :main.mainroute,
        "explode"   :'#',
        "patroutes" :main.patroutes,
      })
} 

export function navLinkPackage(main) {
    if (main.status=='dev') {
          var myFrom='/helpers/Form/Router/navLinkPackage';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });
    }
      window.ClickPackage  = function (key) {
         mainRouter({
            "spinner"    :main.spinner,
            "mainroute"  :main.mainroute,
            "autoload"  :key,
            "patroutes" :1,
         },'package')
      }


     historyRouter({
        "spinner"    :main.spinner,
        "mainroute"  :main.mainroute,
        "explode"   :'#',
        "patroutes" :1,
       },'package')   

} 
export function routerHashtag(main) {
    if (main.status=='dev') {
          var myFrom='/helpers/Form/Router/navLinkHashtag';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });
    }
      window.ClickHashtag   = function (key) {
         mainRouter({
            "spinner"    :main.spinner,
            "mainroute"  :main.mainroute,
            "autoload"  :key,
            "patroutes" :main.patroutes,
         })
      }
      historyRouter({
       "spinner"    :main.spinner,
        "mainroute" :main.mainroute,
        "explode"   :'#',
        "patroutes" :main.patroutes,
      })
} 
export function navLinkTheme(main) {
    if (main.status=='dev') {
          var myFrom='/helpers/Form/Router/navLinkHashtag';
          var actionFrom    =JSON.stringify(main);
            $.post(tatiyeNet.etc,{
                 'etc'          :myFrom,
                 'action'       :actionFrom,
                 beforeSend: function() {
                 },
                })
               .done(function (template) {
                });
    }
      window.ClickTheme   = function (key) {
         mainRouter({
            "spinner"    :main.spinner,
            "mainroute"  :main.mainroute,
            "autoload"  :key,
            "patroutes" :1,
         })
      }
      historyRouter({
       "spinner"    :main.spinner,
        "mainroute" :main.mainroute,
        "explode"   :'#',
        "patroutes" :1,
      })
} 


export function index() {
    return 'Hello Net'
}
export function setSplit(expo=0,Rout='/',marge='') {
    if (marge) {
     var segmenGet=marge;
    } else {
     var segmenGet=$(location).attr('href').split("/").splice(3,10).join("/");
    } 
     var exp=segmenGet.split(Rout);
     return exp[expo]
 }
export function historyRouter(main,Rout='') { 
   
    let history=$(location).attr('href').split(main.explode).splice(main.patroutes,5).join("/");

    if (history) {
        if (Rout=='package') {
          const myRouter = new readRouter(main);
          var segmenGet=$(location).attr('href').split("/").splice(4,12).join("/");
          return myRouter.package(main,explode(segmenGet,main.explode,0)+'/'+history);
        } else {
           var segmenGet=$(location).attr('href').split("/").splice(3,15).join("/");
           mainRouter({
             "spinner"   :main.spinner,
             "mainroute" :main.mainroute,
             "autoload"  :explode(segmenGet,main.explode,0)+'/'+history,
             "patroutes" :main.patroutes,
          })
        } 
    }

}

  

export function useSpinner(main,top) {
 return  spinner(main,top) ;
}
  
export function mainRouterId(main,Rout='') { 
    const myRouter = new readRouter(main);
    if (Rout=='package') {
        var segmenGet=main.autoload;
        return myRouter.package(main,segmenGet);  
    } else {
       return myRouter.hashtag(main);
    }
}

export function setRouter(main) {
 $.each(main.selector,function(key1, row){
   const setID = document.querySelectorAll('#'+row+" a"); 
                $(setID).each(function( index ) {
                 if ($(this).attr("onclick")) {
                 } else {
                    $(this).attr("onclick", 'return false;');
                    //console.log($(this).attr("href"))
                  }
                 });

             $('#'+row+" a").click(function(){
                if ($(this).attr("href")) {
                    history.pushState({}, "", $(this).attr("href"))
                    var event=$(this).attr("href");
                    var segmenGet=event.split("/").splice(3,5).join("/");
                    var segmenMeta=event.split("/").splice(4,1).join("/");
                    var segmenMetaObj=event.split("/").splice(3,1).join("/");
                    headMeta(segmenMeta);
                    const myRouter = new readRouter(main);
                    return myRouter.template(main,segmenGet,segmenMetaObj);   
                 } 
             });
      });


}

export function getHistory(main) { 
      var segmenGet=$(location).attr('href').split("/").splice(4,10).join("/");
      let nmafile=setSplit(1,'#',segmenGet)
      if (nmafile) {
         if (main.segment=='package') {
            var  getDirektori='pec/'+replace(segmenGet,'#','/') 
         } else {
            var  getDirektori='theme/'+replace(segmenGet,'#','/') 
         }
         routePath(getDirektori,main.atribut);
      }
      if (main.content=='tabline') {
         $("#index").removeClass("active");
         $("#"+nmafile).addClass("active").siblings().removeClass("active"); 
      }
 
}

export function Router(main) { 
 $('a').click(function(){
       var ulr=$(this).attr('href');
       if (ulr !=='javascript:void(0);') {
            history.replaceState('setRhref', '', ulr);
            var segmenGet=$(location).attr('href').split("/").splice(4,5).join("/");
            var segmenGetdevices=$(location).attr('href').split("/").splice(4,1).join("/");
            var nav1=$(location).attr('href').split("/").splice(5,1).join("/");
            // if (true) {} else {}
            if (explode(nav1,'_',1)) {
                var myPackage=segmenGetdevices+'/'+explode(nav1,'_',0);
                var appsetKey=explode(nav1,'_',1);
            } else {
                var myPackage=segmenGet;
                var appsetKey='';
            }
            // console.log(segmenGet)
            headMeta(myPackage);

            if (segmenGetdevices=='') {
                var meta='home';
            } else {
                var meta=segmenGetdevices
            }
          

            $("#"+meta).addClass("active").siblings().removeClass("active"); 
            $("#nav"+meta).addClass("active").siblings().removeClass("active");
            $("#nav1"+nav1).addClass("active").siblings().removeClass("active");
            const myRouter = new readRouter(main);
           
            return myRouter.package(main,myPackage,appsetKey);  
        } 
  });
}

export function setTables(data) { 
  return new Tables(data)
  // console.log(data)
}

export function gridSystem(main) { 
    return setGrid(main)
}

// export function setAssets() { 
//     // return new Assets();
// }
const mysetAssets = new Assets();
export class components {
    tab(data) {
      return mysetAssets.tab(data)
    }
    useState(data) {
      return mysetAssets.useHandel(data)
    }
    
    dropdown(data) {
      return mysetAssets.dropdown(data)
    }
    tabel(data) {
      return mysetAssets.tabel(data)
    }

    dropdownTabel(data) {
      return mysetAssets.dropdownTabel(data)
    }
    from(data) {
      return mysetAssets.from(data)
    }
    myfromModal(data) {
      return mysetAssets.myfromModal(data)
    }
    myFrom(data) {
      return mysetAssets.myfromModal(data)
    }
    element(data) {
      return mysetAssets.element(data)
    }
    modal(data) {
      return mysetAssets.modal(data)
    }
    modalStandar(data) {
      return mysetAssets.modalStandar(data)
    }
    keywords(data) {
      return mysetAssets.keywords(data)
    }
    bootstrap(data) {
      return mysetAssets.bootstrap(data)
    }

    datatables(data) {
      return mysetAssets.datatables(data)
    }
    officeTabel(data) {
      return mysetAssets.officeDatatables(data)
    }
    dbXlsx(data) {
      return mysetAssets.datatablesXlsx(data)
    }
    setDevices(data) {
      return mysetAssets.useDevices(data)
    }


    
}




export function gridSystemContent(main) { 
    return setGridTabel(main)
}

export function Qrcode(main,style) { 
    // console.log(main)
    if (style) {
     var qrcode = new QRious({
       element: document.getElementById(style.selector),
       background: '#ffffff',
       backgroundAlpha: 1,
       foreground: style.color?style.color:'#5868bf',
       foregroundAlpha: 1,
       level: 'H',
       padding: 0,
       size: style.size?style.size:128,
       value: main
     });
     return qrcode;
    } else {
     return  new QRious({element: document.getElementById("qrcode"), value: main});
    }
}



export class setHttp {
    get(url) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/'+url)
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }
    select(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/select/'+url, {
                "method": 'POST',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }
    entri(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/entri/'+url, {
                "method": 'POST',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }

    post(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/'+url, {
                "method": 'POST',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }
    put(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/'+url, {
                 "method": 'PUT',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(data => resolve(data))
                .catch(err => reject(err))

        })
    }
    delete(url, data) {
        return new Promise((resolve, reject) => {
            fetch(tatiyeNet.api+'/'+url, {
                "method": 'DELETE',
                 "headers": {
                     "Authorization": tatiyeNet.token,
                     "Content-Type": "application/json"
                 },
                "body": JSON.stringify(data)
            })
                .then(res => res.json())
                .then(() => resolve('Deleted successfully'))
                .catch(err => reject(err))

        })
    }
}


  let count = 0;
/*
|--------------------------------------------------------------------------
| AND Http 
|--------------------------------------------------------------------------
*/


// export function useElementFrom(type) {
// }
export function setFrom(type) {
    // console.log(type)
  const myFrom = new From(type.storage,type.action,type.storage,type.storage.content)
  if (type.storage.content=='upload') {
       myFrom.upload(
        type.storage.model,
        type.storage,
        type.storage.content,
        type.storage.model
        )
  } else if (type.storage.content=='uploadFile'){
       myFrom.upload(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='userid'){
       myFrom.userid(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='codeMirror'){
       myFrom.codeMirror(type.storage.model,type.storage)
  } else if (type.storage.content=='xlsx'){
       myFrom.xlsx(type.storage.model,type.storage)
  } else if (type.storage.content=='firebase'){
       myFrom.firebase(
        type.storage.model,
        type.storage,
        type.action)
  } else if (type.storage.content=='from'){
       myFrom.from(type.storage.model,type.storage)
  } else if (type.storage.content=='office'){
      myFrom.office(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='dompdf'){
      myFrom.dompdf(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='pridRaw'){
        myFrom.pridRaw(type.storage.model,type.storage) 
  } else {
       myFrom.dialog(type.storage.model,type.storage)
  } 
}

export function setModal(style) {
  // console.log(style);
        count++;
        const addModal = new Modal(
            style.titel, 
            style.width, 
            style.key, 
            style.model,
            style.route,
            style.tabel,
            style.content,
            count,
            style.page,
            style
            );
        // console.log(addModal)
         var resheader =style.content+style.titel.replace(/[^A-Za-z0-9]/g, '')+count;
         if(style.content == 'profil') {
            addModal.profil(style.titel,resheader,style.model)  
         } else if (style.content == 'delete'){
           addModal.delete(style.titel,resheader,style.route,style.page)  
         } else if (style.content == 'recycle'){
           addModal.recycle(style.titel,resheader,style.model,style.page)  

         } else if (style.content == 'restore'){
           addModal.restore(style.titel,resheader,style.model,style.page)  

         } else if (style.content == 'from'){
           addModal.from(style.titel,resheader,style.model)  
           
          } else if (style.content == 'pridRaw'){
           addModal.pridRaw(style.titel,style.model,style.model,style)  
         } else if (style.content == 'fromAction'){
           addModal.fromAction(style.titel,style.model,style.model,style) 
         } else if (style.content == 'upload'){
           addModal.upload(style.titel,resheader,style.model)  
          } else if (style.content == 'office'){
           addModal.office(style.titel,resheader,style.model)  
          } else if (style.content == 'dompdf'){
           
           addModal.dompdf(style.titel,resheader,style.model)  
         } else {
           addModal.modal(style.titel,resheader,style.model,style.content)  
         }        
}









export function setFromTables(type) {
  const myFrom = new FromTables(type.storage,type.action,type.storage,type.storage.content)
  if (type.storage.content=='upload') {
      myFrom.upload(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='uploadFile'){
       myFrom.upload(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else if (type.storage.content=='userid'){
       myFrom.userid(type.storage.model,type.storage,type.storage.content,type.storage.model)
  } else {
       myFrom.from(type.storage.model,type.storage)
  } 
 
}

/*
|--------------------------------------------------------------------------
| Initializes Modal 
form_demoModal011692702825018
form_demoModal011692702825018
|--------------------------------------------------------------------------
| Develover Tatiye.Net 
| @Date 8/16/2023 6:11:41 PM
*/
export function  Images(imageUrl) {

 
}


export function fileType(ekstensi) {
             if (ekstensi=='docx') {
                var  myIcon='https://img.freepik.com/free-icon/word_318-674258.jpg';
              } else if (ekstensi=='doc') {
                  myIcon='https://img.freepik.com/free-icon/word_318-674258.jpg';
             
              } else if (ekstensi=='pdf') {
                  myIcon='https://cdn-icons-png.flaticon.com/512/179/179483.png';
              } else if (ekstensi=='xlsx') {
                  myIcon='https://img.freepik.com/free-icon/excel_318-566085.jpg';
                
              } else if (ekstensi=='pptx') {
                  myIcon='https://img.freepik.com/free-icon/powerpoint_318-674249.jpg';
             
              } else if (ekstensi=='csv') {
                   myIcon='https://img.freepik.com/free-icon/excel_318-566085.jpg';
              } else if (ekstensi=='odes') {
                   myIcon='https://img.freepik.com/free-icon/excel_318-566085.jpg';
              } else {
                   myIcon='';
              }
             return myIcon;
}
export function modalEntri(body,key='add') {
      setModal({
       'titel':explode(body,',',0),
       'width':explode(body,',',1),
       'key':key,
       'model':'modalBody',
       'route':explode(body,',',3),
       'content':explode(body,',',2),
      });
}
export function modalRoute(header,id,style,content,path,tabel='',model='',page='',privacy='public') {
       setModal({
           'titel':header,
           'width':style,
           'key'  :id,
           'model':model,
           'route':path,
           'tabel':tabel,
           'content':content,
           'page'   :page,
           'setdata'   :privacy,
          });
       //console.log(header);
}




export function replaceText(main) {
    return main.replace(/[^A-Za-z0-9]/g, ' ') 
}
export function modalRouteFrom(main,key='add',tabel,content,page='1') {

        modalRoute(
            explode(main,',',0),
            key,
            explode(main,',',1),
            explode(main,',',2),
            explode(main,',',3),
            tabel,
            content,
            page
        );
}
export function sendFrom(main,key='add',tabel,content,page='1',privacy='public') {
        modalRoute(
            explode(main,',',0),
            key,
            explode(main,',',1),
            explode(main,',',2),
            explode(main,',',3),
            tabel,
            content,
            page,
            privacy
        );
}


/*
|--------------------------------------------------------------------------
| AND Modal 
|--------------------------------------------------------------------------
*/
var arrbulanEn  =["January","February","March","April","May","June","July","August","September","October","November","Desember"];
var arrbulan    =["Januari","Februari","Maret","April","Mei","Juni","Juli","Agustus","September","Oktober","November","Desember"];
var hariarray =new Array("Minggu","Senin","Selasa","Rabu","Kamis","Jum'at","Sabtu");
var getlocation=$(location).attr('href').split("/").splice(1).join("/");
var tn        =getlocation.split('/');


export function terbilang(a){
   var bilangan = ['','Satu','Dua','Tiga','Empat','Lima','Enam','Tujuh','Delapan','Sembilan','Sepuluh','Sebelas'];

   // 1 - 11
   if(a < 12){
      var kalimat = bilangan[a];
   }
   // 12 - 19
   else if(a < 20){
      var kalimat = bilangan[a-10]+' Belas';
   }
   // 20 - 99
   else if(a < 100){
      var utama = a/10;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%10;
      var kalimat = bilangan[depan]+' Puluh '+bilangan[belakang];
   }
   // 100 - 199
   else if(a < 200){
      var kalimat = 'Seratus '+ terbilang(a - 100);
   }
   // 200 - 999
   else if(a < 1000){
      var utama = a/100;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%100;
      var kalimat = bilangan[depan] + ' Ratus '+ terbilang(belakang);
   }
   // 1,000 - 1,999
   else if(a < 2000){
      var kalimat = 'Seribu '+ terbilang(a - 1000);
   }
   // 2,000 - 9,999
   else if(a < 10000){
      var utama = a/1000;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%1000;
      var kalimat = bilangan[depan] + ' Ribu '+ terbilang(belakang);
   }
   // 10,000 - 99,999
   else if(a < 100000){
      var utama = a/100;
      var depan = parseInt(String(utama).substr(0,2));
      var belakang = a%1000;
      var kalimat = terbilang(depan) + ' Ribu '+ terbilang(belakang);
   }
   // 100,000 - 999,999
   else if(a < 1000000){
      var utama = a/1000;
      var depan = parseInt(String(utama).substr(0,3));
      var belakang = a%1000;
      var kalimat = terbilang(depan) + ' Ribu '+ terbilang(belakang)+ ' Rupiah ';
   }
   // 1,000,000 -    99,999,999
   else if(a < 100000000){
      var utama = a/1000000;
      var depan = parseInt(String(utama).substr(0,4));
      var belakang = a%1000000;
      var kalimat = terbilang(depan) + ' Juta '+ terbilang(belakang);
   }
   else if(a < 1000000000){
      var utama = a/1000000;
      var depan = parseInt(String(utama).substr(0,4));
      var belakang = a%1000000;
      var kalimat = terbilang(depan) + ' Juta '+ terbilang(belakang);
   }
   else if(a < 10000000000){
      var utama = a/1000000000;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%1000000000;
      var kalimat = terbilang(depan) + ' Milyar '+ terbilang(belakang);
   }
   else if(a < 100000000000){
      var utama = a/1000000000;
      var depan = parseInt(String(utama).substr(0,2));
      var belakang = a%1000000000;
      var kalimat = terbilang(depan) + ' Milyar '+ terbilang(belakang);
   }
   else if(a < 1000000000000){
      var utama = a/1000000000;
      var depan = parseInt(String(utama).substr(0,3));
      var belakang = a%1000000000;
      var kalimat = terbilang(depan) + ' Milyar '+ terbilang(belakang);
   }
   else if(a < 10000000000000){
      var utama = a/10000000000;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%10000000000;
      var kalimat = terbilang(depan) + ' Triliun '+ terbilang(belakang);
   }
   else if(a < 100000000000000){
      var utama = a/1000000000000;
      var depan = parseInt(String(utama).substr(0,2));
      var belakang = a%1000000000000;
      var kalimat = terbilang(depan) + ' Triliun '+ terbilang(belakang);
   }

   else if(a < 1000000000000000){
      var utama = a/1000000000000;
      var depan = parseInt(String(utama).substr(0,3));
      var belakang = a%1000000000000;
      var kalimat = terbilang(depan) + ' Triliun '+ terbilang(belakang);
   }

  else if(a < 10000000000000000){
      var utama = a/1000000000000000;
      var depan = parseInt(String(utama).substr(0,1));
      var belakang = a%1000000000000000;
      var kalimat = terbilang(depan) + ' Kuadriliun '+ terbilang(belakang);
   }

   var pisah = kalimat.split(' ');
   var full = [];
   for(var i=0;i<pisah.length;i++){
    if(pisah[i] != ""){full.push(pisah[i]);}
   }
   var Rp=full.join(' ');

   return Rp;
}
function Encode(argument) {
var decodedStringBtoA =argument;
var encodedStringBtoA = btoa(decodedStringBtoA);
var resheader = encodedStringBtoA.replace(/[^A-Za-z0-9]/g, '');
return resheader;
}

function Decode(argument) {
  var encodedStringAtoB = argument;
  var decodedStringAtoB = atob(encodedStringAtoB);
  return decodedStringAtoB;
}


 function newToken(myData){
       var varjs=JSON.stringify(myData, null, 10);
       var myToken=Encode(varjs);
        return myToken;
  }
  /* and class setToken */
  /*
  |--------------------------------------------------------------------------
  | Initializes getToken 
  |--------------------------------------------------------------------------
  | Develover Tatiye.Net 2022
  | @Date  
  */
   function ngetToken(myData){
        var addset=JSON.parse(Decode(myData)); 
        return addset;
      
  }
  /* and class getToken */

function text_shorten(str, length, ending=''){
    if (length == null) {
      length = 10;
    }
    if (ending == null) {
       ending = '...';
    }
    if (str.length > length) {
      return str.substring(0, length - ending.length) + '...';
    } else {
      return str;
    }
  };
  /*
  |--------------------------------------------------------------------------
  | Initializes title 
  |--------------------------------------------------------------------------
  | Develover Tatiye.Net 2022
  | @Date  
  */

 
  /* and class title */
export function rupiah(num) {
  const numb = num;
  const format = numb.toString().split('').reverse().join('');
  const convert = format.match(/\d{1,3}/g);
  const rupiah = convert.join('.').split('').reverse().join('');
  return rupiah;
}
export function romawi(num) {
  var roman = {
    M: 1000,
    CM: 900,
    D: 500,
    CD: 400,
    C: 100,
    XC: 90,
    L: 50,
    XL: 40,
    X: 10,
    IX: 9,
    V: 5,
    IV: 4,
    I: 1
  }
  var result = '';
  for (var key in roman) {
    if (num == roman[key]) {
      return result +=key;
    }
    var check = num > roman[key];
    if(check) {
      result = result + key.repeat(parseInt(num/roman[key]));
      num = num%roman[key];
    }
  }
 return result;
}
export function substr(text='',str=0,str1=0) {
     var res = text.substring(str,str1);
    return res;
}
export function readmore(text,str) {
  return text_shorten(text,str);
}
export function explode(text,str,str1) {
     var tn =text.split(str);
   return tn[str1];
}
export function expo(num) {
     return tn[num];
}
export function replace(textToReplace,bstr,bstr1) {
  let text = textToReplace; 
let result = text.replace(bstr, bstr1);
return result;
}

export function setDecode(code,key='') {
    var dcodeset=(typeof(code) === 'number' || typeof(code) === "string" && code.trim() !== '') && !isNaN(code);  
   if (dcodeset==true) {
      return Encode(code);
   } else {
      return Decode(code);
   }
}

export function setToken(myData) {
  let nToken=newToken(myData);
   return nToken; 
}

export function getToken(myData) {
   return ngetToken(myData);
}




export function setDateTime(dy,flag='',mydate='') {

   // if(OPENID_ID == 'st') {
  var tw = new Date();
  // if (tw.getTimezoneOffset() == 0) (a=tw.getTime() + ( 7 *60*60*1000))
  // else (a=tw.getTime());
  // tw.setTime(a);
 var tahun= tw.getFullYear ();
 var hari= tw.getDay ();
 var bulan= tw.getMonth();
 var tanggal= tw.getDate ();
 var hours = tw.getHours();
 var minutes = "0" + tw.getMinutes();
// Seconds part from the timestamp
var seconds = "0" + tw.getSeconds();
 var hariarray=new Array("Minggu","Senin","Selasa","Rabu","Kamis","Jum'at","Sabtu");
 var bulanarray=new Array("Januari","Februari","Maret","April","Mei","Juni","Juli","Agustus","September","Oktober","November","Desember");
 var bln=bulan+1;
 var ampmd = hours >= 12 ? 'PM' : 'AM';
 var tedTime = hours + ':' + minutes.substr(-2) + ':' + seconds.substr(-2)+' '+ampmd;
 var dcodeset=(typeof(dy) === 'number' || typeof(dy) === "string" && dy.trim() !== '') && !isNaN(dy);  
         if (dcodeset==true) {
            var seTime='number';
         } else {
            var seTime='text';
         }
 if(flag !== '') {
      var month =flag.split('/');
     var getMonth=arrbulan[month[1]-1];
     var getDetmOnd=  arrbulanEn[month[1]-1]+' '+month[2]+', '+month[0];
     var tw = new Date(getDetmOnd);
     var hari= tw.getDay();
  
      if (dy =='IN') {
         return month[2]+' '+getMonth+' '+month[0]; 
      } else if (dy=='EN') {
         return month[0]+' '+getMonth+' '+month[2]; 
      } else if (dy=='Bln') {
           return month[1]; 
      } else if (dy=='Year') {
          return month[0]; 
      } else if (dy=='Month') {
          return getMonth; 
      } else if (dy=='Day') {
         return  hariarray[hari]; 
      } else if (dy=='IND') {
         return  hariarray[hari]+', '+month[2]+' '+getMonth+' '+month[0]; 
      } else {
        if (seTime =='number') {
                let unix_timestamp = dy;
                var date = new Date(unix_timestamp * 1000);
                // Hours part from the timestamp
                var hours = date.getHours();
                // Minutes part from the timestamp
                var minutes = "0" + date.getMinutes();
                // Seconds part from the timestamp
                var seconds = "0" + date.getSeconds();

                 var hari   = date.getDay ();
                 var bulan  = date.getMonth ();
                 var tanggal= date.getDate ();
                 var tahun  = date.getFullYear ();
                 var ampm = hours >= 12 ? 'PM' : 'AM';

                // Will display time in 10:30:23 format
                var formattedTime =tanggal+" "+arrbulan[bulan]+" "+tahun;
                var formatgetDay =hariarray[hari]+", "+tanggal+" "+arrbulan[bulan]+" "+tahun;
                var tedTime = hours + ':' + minutes.substr(-2) + ':' + seconds.substr(-2)+' '+ampm;
                    if (flag=='dmy') {
                       return formatgetDay;
                    } else if (flag=='v') {
                       return formatgetDay+', '+tedTime;   
                    } else if (flag=='time') {
                         return tedTime;  
                    } else if (flag=='Day') {
                        return hariarray[hari];
                    } else if (flag=='Month') {
                        return arrbulan[bulan];
                    } else if (flag=='Year') {
                         return tahun;
                    } else {
                       return formattedTime;
                    }

        } else {      
          return  month[2]+' '+getMonth+' '+month[0]; 
        }
      }

 } else {
    if (bln ==10) {
      var ampm = bln;
    } else if (bln==11) {
        var ampm = bulan;
    } else if (bln==12) {
        var ampm = bln;
    } else {
         var ampm = '0'+bln;
    }
    if (tanggal < 10) {
        var mytgl = '0'+tanggal;
    } else {
        var mytgl = tanggal;
    }

    if (dy=='hari') {
        return hariarray[hari];
     } else if (dy=='bulan') {
        return bulanarray[bulan];
     } else if (dy=='tahun') {
        return tahun;
     } else if (dy=='jam') {
        return tedTime;
     } else if (dy=='EN') {
        return tahun+'/'+ampm+'/'+mytgl;
     } else if (dy=='IN') {
        return mytgl+'/'+ampm+'/'+tahun;
    } else if (dy=='DIN') {
        return mytgl+' '+bulanarray[bulan]+' '+tahun;
     } else {
           return hariarray[hari]+", "+mytgl+" "+bulanarray[bulan]+" "+tahun;
     }

 }


}



export function setLicense(key){
const assets = new components();
var row=createLicense();
 if (tatiyeNet.userid) {
   assets.from({
          "autoload"   :key,
          "storage"    :[true, 'applicense',row.id],
          "action"     :{
              'serial'       :['copy',    '12','License',row.serial],
              'password'     :['password','6' ,'Password',row.password],
              'date'         :['date',    '6' ,'Expair','yy/mm/dd',row.date],
         },
   });
 } else {
   clientLogin(key);
 }
}

// window.location.origin
  
// $("a").click(function(){

//      // var segmenGet=$(location).attr('href').split("/").splice(4,10).join("/");
//      // let results=setSplit(1,'#',$(this).attr('href'))
//      // // if (results) {
//      // console.log(results);
//      console.log($(this).attr('href'))   
//     // }

// });
