// tatiye/Router/index.js
import tatiyeNet,{
    replace,
    setSplit,
    ngoreiDomElement,
    routerNavigasi,
    mainrouteReturn,
    explode,
    hashtagPage,
    mainrouteReturnHasPage,
    navigasiRequere,
    switchPage,
    tabDOM,
    internalDOM,
    userAgent
}  from "../es6.js"; 
let count = 0; 
import {spinner,spinnerModal} from "./spinner.js";
export class readRouter {
    constructor(storage,navigasi='') {
        this.storage = storage
        this.navigasi = navigasi
    }
    hashtag (main){
            var phat=main.autoload.split('=');
           
            let setSplitPage=setSplit(1,'row/',main.autoload)

            if (phat[1]) {
                var titelKey=phat[1];
            } else {
                var titelKey='index';
            }
          
          if (setSplitPage) {
            var pageY=replace(setSplitPage,'#','/');
          } else {
            var pageY=phat[0];
          }
            $.ajax({
                   type: 'POST',
                    url: tatiyeNet.route,
                    data: 'package='+pageY+'&key='+titelKey,
                    beforeSend: function() {
                      $("#"+main.mainroute).html(spinner(main.spinner));  
                   },
                   success: function(data) {
                      $("#"+main.mainroute).html(data);
                  }
              });
    }
 
    hasPage(main,segmenGet,segmenMeta='',eleDom='',navigasi=''){
         let setSegmenGet=explode(segmenGet,'=',0)

         $.ajax({
                type: 'POST',
                 url: tatiyeNet.route,
                 data: 'package='+setSegmenGet+'&key='+segmenMeta+'&base='+setSegmenGet,
                 beforeSend: function() {
                  // $("#"+main.page).html(spinner(main.spinner));    
                },
                success: function(data) {
                   $("#"+main.page).html(data);
                   if (eleDom) {
                    ngoreiDomElement(eleDom,segmenGet)
                   }
                    mainrouteReturn("#"+main.page+" a",main);
                    $("#"+main.page+" a").click(function(evt){
                          
                           if ($(this).attr("class")=='external') {
                             } else  if ($(this).attr("tab")) {
                                 tabDOM(this)
                           } else  if ($(this).attr("internal")) {
                                internalDOM(this)  
                              $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("lightbox")) {
                              } else  if ($(this).attr("modal")) {
                                 } else  if ($(this).attr("download")) {
                           } else  if ($(this).attr("href")=='#') {
                                $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("href")=='') {
                                $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("id")=='back') {
                              $(this).attr("onclick","goBack()")
                           } else {
                            if ($(this).attr("load")) {
                               var mainroute=$(this).attr("load");

                            } else {
                               var mainroute=main.mainroute;
                            }  
                              if ($(this).attr('scroll')) {
                                    let styleScl=explode($(this).attr('scroll'),'|',1)
                                    if (styleScl) {
                                       $('html, body').animate({scrollTop:styleScl}, 'slow');
                                    } else {
                                        if ($(this).attr('scroll')=='top') {
                                              $(this).removeAttr('href')
                                             $('html, body').animate({scrollTop:0}, 'slow'); 
                                        } else {
                                             $('html, body').animate({scrollTop:$(document).height()}, 'slow');
                                        }
                                        
                                    } 
                              }


                                evt.preventDefault();
                                 if ($(this).attr("href") !=="javascript:void(0);") {
                                    history.pushState({}, "", $(this).attr("href"))
                                   var pageZ=$(this).attr("href").split("/").splice(tatiyeNet.patroutes,10).join("/");
                                   var segmenMeta=$(this).attr("href").split("/").splice(tatiyeNet.patroutes,1).join("/");
                                         let subSegmenGet=explode(pageZ,'=',1)
                                          let subPage=replace(pageZ,'#','/')
                                          let getsubPage=replace(subPage,'=','/')
                                          if (subSegmenGet) {
                                          let getsubPage=replace(subPage,'=','/')

                                          } else {
                                          let getsubPage=pageZ

                                          }



 
                                   routerNavigasi({
                                         'spinner'     :main.spinner,
                                         'mainroute'   :mainroute,
                                         'selector'    :main.selector,
                                         'page'        :main.page,
                                         'path'        :getsubPage,
                                         'segmenMeta'  :segmenMeta,
                                         'eleDom'      :eleDom
                                   })
                               }
                                }
                    });
            
               }
           });
    } 



    template(main,segmenGet,segmenMeta='',eleDom='',navigasi=''){
       if (segmenGet=='users/logout') {
         window.location.href =tatiyeNet.routes+"/"+segmenGet;
       } else {
        if (main.useragent) {
           let dataset=userAgent({
                'indexOn':'Desktop'
            })
         }
         let setSegmenGet=explode(segmenGet,'=',0)
         let hasPage=replace(setSegmenGet,'#','/')
         let hashtag=explode(setSegmenGet,'#',1)
         let page=explode(setSegmenGet,'#',0)
         let subSegmenGet=explode(segmenGet,'=',1)
        if (subSegmenGet) {
          // var subhasPage=page+'/'+hashtag+'/'+subSegmenGet;
        } 
         $.ajax({
                type: 'POST',
                 url: tatiyeNet.route,
                 data: 'package='+page+'&key='+segmenGet+'&base='+segmenMeta+'&path='+page,
                 beforeSend: function() {
                     $("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                   $("#"+main.mainroute).html(data);
                  
                  
                   if (eleDom) {
                       ngoreiDomElement(eleDom,segmenGet)
                   }
                   if (hashtag) {
                        hashtagPage(main,hasPage,segmenMeta,eleDom,navigasi)
                   }

                   if (subSegmenGet) {

                      //switchPage(main,subhasPage,segmenMeta,eleDom,navigasi)
                   }   
                 
                    mainrouteReturn("#"+main.mainroute+" a",main);
                    $("#"+main.mainroute+" a").click(function(evt){

              
 
                           if ($(this).attr("class")=='external') {
                           } else  if ($(this).attr("tab")) {
                                   tabDOM(this)  
                           } else  if ($(this).attr("internal")) {
                                   $(this).attr('href',"javascript:void(0);")
                                   internalDOM(this)  
                           } else  if ($(this).attr("data-slide")) {
                           } else  if ($(this).attr("lightbox")) {
                           } else  if ($(this).attr("modal")) {
                           } else  if ($(this).attr("download")) {
                             $(this).attr('href',"javascript:void(0);")
                           } else  if ($(this).attr("dropdown")) {
                           } else  if ($(this).attr("href")=='#') {
                                $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("href")=='') {
                                $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("id")=='back') {
                              $(this).attr("onclick","goBack()")
                           } else {
                            if ($(this).attr("load")) {
                               var mainroute=$(this).attr("load");
                               var mainroute=$(this).attr("load");
                               var mainpage=getsubPage;


                            } else {
                               var mainroute=main.mainroute;
                               var mainpage=main.page;
                            }    
                
                              if ($(this).attr('scroll')) {
                                    let styleScl=explode($(this).attr('scroll'),'|',1)
                                    if (styleScl) {
                                       $('html, body').animate({scrollTop:styleScl}, 'slow');
                                    } else {
                                        if ($(this).attr('scroll')=='top') {
                                             $(this).removeAttr('href')
                                             $('html, body').animate({scrollTop:0}, 'slow'); 
                                        } else {
                                             $('html, body').animate({scrollTop:$(document).height()}, 'slow');
                                        }
                                        
                                    } 
                              }
                                evt.preventDefault();
                                if ($(this).attr("href")!=='javascript:void(0);') {
                                    if ($(this).attr("href")) {
                                   if ($(this).attr("data-slide")) {} else {
                                     history.pushState({}, "", $(this).attr("href"))
                                   }
                                   var pageZ=$(this).attr("href").split("/").splice(tatiyeNet.patroutes,10).join("/");
                                   var segmenMeta=$(this).attr("href").split("/").splice(tatiyeNet.patroutes,1).join("/");
                                  
                                           routerNavigasi({ 
                                                 'useragent'   :main.useragent,
                                                 'spinner'     :main.spinner,
                                                 'mainroute'   :mainroute,
                                                 'selector'    :main.selector,
                                                 'page'        :main.page,
                                                 'path'        :pageZ,
                                                 'segmenMeta'  :segmenMeta,
                                                 'eleDom'      :eleDom
                                           })
                                   }
                                   }
                          }
                    });
  
               }
           });
     }
       
    } 
templateModal(main,segmenGet,segmenMeta='',eleDom='',navigasi=''){
       if (segmenGet=='users/logout') {
         window.location.href =tatiyeNet.routes+"/"+segmenGet;
       } else {
        if (main.useragent) {
           let dataset=userAgent({
                'indexOn':'Desktop'
            })
         }
         let setSegmenGet=explode(segmenGet,'=',0)
         let hasPage=replace(setSegmenGet,'#','/')
         let hashtag=explode(setSegmenGet,'#',1)
         let page=explode(setSegmenGet,'#',0)
         let subSegmenGet=explode(segmenGet,'=',1)
        if (subSegmenGet) {
          // var subhasPage=page+'/'+hashtag+'/'+subSegmenGet;
        } 
         $.ajax({
                type: 'POST',
                 url: tatiyeNet.route,
                 data: 'package='+page+'&key='+segmenGet+'&base='+segmenMeta+'&path='+page,
                 beforeSend: function() {
                     $("#"+main.mainroute).html(spinnerModal(main.spinner));    
                },
                success: function(data) {
                   $("#"+main.mainroute).html(data);
                  
                  
                   if (eleDom) {
                       ngoreiDomElement(eleDom,segmenGet)
                   }
                   if (hashtag) {
                        hashtagPage(main,hasPage,segmenMeta,eleDom,navigasi)
                   }

                   if (subSegmenGet) {

                      //switchPage(main,subhasPage,segmenMeta,eleDom,navigasi)
                   }   
                 
                    mainrouteReturn("#"+main.mainroute+" a",main);
                    $("#"+main.mainroute+" a").click(function(evt){

              
 
                           if ($(this).attr("class")=='external') {
                           } else  if ($(this).attr("tab")) {
                                   tabDOM(this)  
                           } else  if ($(this).attr("internal")) {
                                   $(this).attr('href',"javascript:void(0);")
                                   internalDOM(this)  
                           } else  if ($(this).attr("data-slide")) {
                           } else  if ($(this).attr("lightbox")) {
                           } else  if ($(this).attr("modal")) {
                           } else  if ($(this).attr("download")) {
                             $(this).attr('href',"javascript:void(0);")
                           } else  if ($(this).attr("dropdown")) {
                           } else  if ($(this).attr("href")=='#') {
                                $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("href")=='') {
                                $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("id")=='back') {
                              $(this).attr("onclick","goBack()")
                           } else {
                            if ($(this).attr("load")) {
                               var mainroute=$(this).attr("load");
                               var mainroute=$(this).attr("load");
                               var mainpage=getsubPage;


                            } else {
                               var mainroute=main.mainroute;
                               var mainpage=main.page;
                            }    
                
                              if ($(this).attr('scroll')) {
                                    let styleScl=explode($(this).attr('scroll'),'|',1)
                                    if (styleScl) {
                                       $('html, body').animate({scrollTop:styleScl}, 'slow');
                                    } else {
                                        if ($(this).attr('scroll')=='top') {
                                             $(this).removeAttr('href')
                                             $('html, body').animate({scrollTop:0}, 'slow'); 
                                        } else {
                                             $('html, body').animate({scrollTop:$(document).height()}, 'slow');
                                        }
                                        
                                    } 
                              }
                                evt.preventDefault();
                                if ($(this).attr("href")!=='javascript:void(0);') {
                                    if ($(this).attr("href")) {
                                   if ($(this).attr("data-slide")) {} else {
                                     history.pushState({}, "", $(this).attr("href"))
                                   }
                                   var pageZ=$(this).attr("href").split("/").splice(tatiyeNet.patroutes,10).join("/");
                                   var segmenMeta=$(this).attr("href").split("/").splice(tatiyeNet.patroutes,1).join("/");
                                  
                                           routerNavigasi({ 
                                                 'useragent'   :main.useragent,
                                                 'spinner'     :main.spinner,
                                                 'mainroute'   :mainroute,
                                                 'selector'    :main.selector,
                                                 'page'        :main.page,
                                                 'path'        :pageZ,
                                                 'segmenMeta'  :segmenMeta,
                                                 'eleDom'      :eleDom
                                           })
                                   }
                                   }
                          }
                    });
  
               }
           });
     }
       
    } 

    switchsubPage(main,segmenGet,segmenMeta='',eleDom='',navigasi=''){
        // console.log(segmenGet)

        $.ajax({
                   type: 'POST',
                    url: tatiyeNet.route,
                    data: 'package='+segmenGet+'&key=0',
                    beforeSend: function() {
                      $("#"+main.switch).html(spinner(main.switch));  
                   },
                   success: function(data) {
                      $("#"+main.switch).html(data);
                       if (eleDom) {
                        ngoreiDomElement(eleDom,segmenGet)
                       }
                       mainrouteReturn("#"+main.switch+" a",main);
                      $("#"+main.switch+" a").click(function(evt){
                           if ($(this).attr("class")=='external') {
                           } else  if ($(this).attr("tab")) {
                                tabDOM(this)
                           } else  if ($(this).attr("internal")) {
                               internalDOM(this)  
                          } else  if ($(this).attr("lightbox")) {
                           } else  if ($(this).attr("href")=='#') {
                                $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("modal")) {
                             } else  if ($(this).attr("download")) {
                           } else  if ($(this).attr("href")=='') {
                                $(this).attr("href","javascript:void(0);")
                           } else  if ($(this).attr("id")=='back') {
                              $(this).attr("onclick","goBack()")
                           } else {
                            if ($(this).attr("load")) {
                               var mainroute=$(this).attr("load");
                               var mainroute=$(this).attr("load");
                               var mainpage=getsubPage;

                            } else {
                               var mainroute=main.switch;
                               var mainpage=main.page;
                            }    
                
                              if ($(this).attr('scroll')) {
                                    let styleScl=explode($(this).attr('scroll'),'|',1)
                                    if (styleScl) {
                                       $('html, body').animate({scrollTop:styleScl}, 'slow');
                                    } else {
                                        if ($(this).attr('scroll')=='top') {
                                             $(this).removeAttr('href')
                                             $('html, body').animate({scrollTop:0}, 'slow'); 
                                        } else {
                                             $('html, body').animate({scrollTop:$(document).height()}, 'slow');
                                        }
                                        
                                    } 
                              }
                                evt.preventDefault();
                                if ($(this).attr("href")!=='javascript:void(0);') {
                                   history.pushState({}, "", $(this).attr("href"))
                                   var pageZ=$(this).attr("href").split("/").splice(tatiyeNet.patroutes,10).join("/");
                                   var segmenMeta=$(this).attr("href").split("/").splice(tatiyeNet.patroutes,1).join("/");
                                   routerNavigasi({
                                         'spinner'     :main.spinner,
                                         'mainroute'   :mainroute,
                                         'selector'    :main.selector,
                                         'page'        :main.page,
                                         'path'        :pageZ,
                                         'segmenMeta'  :segmenMeta,
                                         'eleDom'      :eleDom
                                   })
                                       }
                                }
                    });
  
               }
           });
       
    } 
    theme(main){

            $.ajax({
                   type: 'POST',
                    url: tatiyeNet.route,
                    data: 'package='+main.autoload+'&key=0',
                    beforeSend: function() {
                      $("#"+main.mainroute).html(spinner(main.spinner));  
                   },
                   success: function(data) {
                      $("#"+main.mainroute).html(data);
                  }
              });
    }

    navRequere(main,segmenGet,segmenMeta='',eleDom='',navigasi=''){
    
        let page=replace(segmenGet,'#','/')
        
         $.ajax({
                type: 'POST',
                 url: tatiyeNet.route,
                 data: 'package='+page+'&key=0'+'&base='+segmenMeta,
                 beforeSend: function() {
                   $("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                   $("#"+main.mainroute).html(data);
            
                   if (eleDom) {
                    ngoreiDomElement(eleDom,segmenGet)
                   }
                     mainrouteReturn("#"+main.mainroute+" a",main);
                   // if (hashtag) {
                   //      hashtagPage(main,hasPage,segmenMeta,eleDom,navigasi)
                   // }
                  
                   //  mainrouteReturn("#"+main.mainroute+" a",main);
               }
           });
         
    }  
    performance(main,segmenGet,segmenMeta='',eleDom='',navigasi=''){
         let setSegmenGet=explode(segmenGet,'=',0)
         
        
         let hasPage=replace(setSegmenGet,'#','/')
         let hashtag=explode(setSegmenGet,'#',1)
         // console.log(setSegmenGet)
         let page=explode(setSegmenGet,'#',0)

          let subSegmenGet=explode(segmenGet,'=',1)
     
        if (subSegmenGet) {
       
          var subhasPage=page+'/'+hashtag+'/'+subSegmenGet;
        } 
          

         $.ajax({
                type: 'POST',
                 url: tatiyeNet.route,
                 data: 'package='+page+'&key='+segmenGet+'&base='+segmenMeta,
                 beforeSend: function() {
                     $("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                    $("#"+main.mainroute).html(data);
                   if (eleDom) {
                    ngoreiDomElement(eleDom,segmenGet)
                   }

                   if (hashtag) {
                        hashtagPage(main,hasPage,segmenMeta,eleDom,navigasi)
                   }

                     //  if (subSegmenGet) {
                     //    console.log(subhasPage)
                     // //      switchPage(main,subhasPage,segmenMeta,eleDom,navigasi)
                     //  }   
                  
                   mainrouteReturn("#"+main.mainroute+" a",main);
                }
            });
    }  


    package (main,segmenGet='',key='',addTabel=''){
        // console.log(key)
            let hashtag=explode(segmenGet,'/',1)
            if (hashtag) {
               $("#"+hashtag).addClass("active").siblings().removeClass("active"); 
               $("#nav"+hashtag).addClass("active").siblings().removeClass("active");
            } else {
            $("#home").addClass("active").siblings().removeClass("active"); 
            $("#navhome").addClass("active").siblings().removeClass("active");
            }
         $.ajax({
                type: 'POST',
                 url: tatiyeNet.autoload,
                 data: 'package='+segmenGet+'&key='+key+'&tabel='+addTabel,
                 beforeSend: function() {
                   $("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                   $("#"+main.mainroute).html(data);
                   mainrouteReturn("#"+main.mainroute+" a",main);
                   history.replaceState(null, null, '');
               }
           });
    }    
    
}
