// tatiye/Router/index.js
import tatiyeNet,{replace,setSplit}  from "../es6.js";  
import {spinner} from "./spinner.js";
export class readRouter {
    constructor(storage) {
        this.storage = storage
    }
    hashtag (main){
            var phat=main.autoload.split('=');
           
            let setSplitPage=setSplit(1,'row/',main.autoload)

            if (phat[1]) {
                var titelKey=phat[1];
            } else {
                var titelKey='index';
            }
          
          if (setSplitPage) {
            var pageY=replace(setSplitPage,'#','/');
          } else {
            var pageY=phat[0];
          }
            $.ajax({
                   type: 'POST',
                    url: tatiyeNet.route,
                    data: 'package='+pageY+'&key='+titelKey,
                    beforeSend: function() {
                      $("#"+main.mainroute).html(spinner(main.spinner));  
                   },
                   success: function(data) {
                      $("#"+main.mainroute).html(data);
                  }
              });
    }
    template(main,segmenGet,segmenMeta=''){
         let page=replace(segmenGet,'#','/')
         $.ajax({
                type: 'POST',
                 url: tatiyeNet.route,
                 data: 'package='+page+'&key=0'+'&base='+segmenMeta,
                 beforeSend: function() {
                   $("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                   $("#"+main.mainroute).html(data);
                   history.replaceState(null, null, '');
               }
           });
         
    }  
    theme(main){

            $.ajax({
                   type: 'POST',
                    url: tatiyeNet.route,
                    data: 'package='+main.autoload+'&key=0',
                    beforeSend: function() {
                      $("#"+main.mainroute).html(spinner(main.spinner));  
                   },
                   success: function(data) {
                      $("#"+main.mainroute).html(data);
                  }
              });
    }
    package (main,segmenGet='',key=''){
         $.ajax({
                type: 'POST',
                 url: tatiyeNet.autoload,
                 data: 'package='+segmenGet+'&key='+key,
                 beforeSend: function() {
                   $("#"+main.mainroute).html(spinner(main.spinner));    
                },
                success: function(data) {
                   $("#"+main.mainroute).html(data);
                   history.replaceState(null, null, '');
               }
           });
    }    
    
}
