// tatiye/Router/routerLink.js
import tatiyeNet,{setModal,setFrom,itemSpinner}  from "../es6.js";  
import {spinner} from "./spinner.js";
 let count = 0;
export function routerLink(row,main='') {
 count++;
 var expoModel=main['titel'].replace(/[^A-Za-z0-9]/g, '')+count
 if (row[2]) {
   var routes=tatiyeNet.routes+'/resquire/pec/'+row[1];  
 } else {
   var routes=tatiyeNet.routes+'/resquire/theme/'+row[1];  
 }
 if (row[0]=='modal') {
     var element='get';
 } else {
     var element='get';

 }
 // console.log(routes)
if (row[0]=='modal') {
      setModal({
           'titel'   :main['titel'],
           'width'   :main['width'],
           'key'     :main['key'],
           'model'   :expoModel,
           'route'   :main['package'],
           'tabel'   :main['tabel'],
           'content' :'fromAction',
           'element' :element,
           'footer'  :'public', //public,privacy 
           'page'    :1,
           'setdata' :'public',
       });
      $.ajax({
            type: 'POST',
             url: routes,
             data: 'key='+main['key']+'&tabel='+main['tabel']+'&titel='+main['titel'],
             beforeSend: function() {
              $("#body"+expoModel).html(itemSpinner('modal'));    
            },
            success: function(data) {
               $("#body"+expoModel).html(data);
           }
       }); 
} else {

}
   //console.log(row) 
}
