// tatiye/Assets/elementsDOM.js
import tatiyeNet,{
modalEntri,
mainRouterId,
modalRouteFrom,
useHandel,
useTabPages,
readmore,
components,
getItemStorage,
setSplit,
replace,
explode,
countWords,
setItemStorage,
packageIndex,
useHandelIDTabel,
setFrom,
setModal,
elDOM,
pathRoute,
queryElements,
itemSpinner,
objectRoute,
romawi,
splitLeft,
assetsDOM,
elsDOM,
rtsDOM,
textContentDOM
}  from "../es6.js"; 
import '../pagination.min.js';
let count = 0;
export function elementsDOM(query,key,content,router='') {
          var keyname=key;
          var elementcontent =content;
          var setID="#"+router.mainroute+" "+elementcontent;   
          $(elementcontent).hide(); 
          // console.log(content+'_'+key)
          var useElement =content+'_'+key; 
          var myTemplate='';
         $.each(query, function(i, row) {
          $(setID+" a").each(function( index ) {
                    if ($(this).attr(keyname)) {
                        var atrNm=$(this).attr(keyname);
                        $(this).attr('id',row.id)
                         if ($(this).attr('link')) {
                            var nmLink=tatiyeNet.routes+"/"+$(this).attr('link');
                            $(this).attr("href", nmLink+"/"+row[atrNm]);
                        } else {
                            $(this).attr("href", row[atrNm]);
                        }
                }
                   if ($(this).attr('onclick')) {
                       $(this).attr('id',row.id)
                       $(this).attr('content',item.tabel)
                       $(this).attr("href", 'javascript:void(0);');
                    }
                    // console.log($(this).attr('modal'))
                    if ($(this).attr('modal')) {
                        $(this).attr('id',row.id)
                        $(this).attr("onclick","return false;")
                        $(this).attr("href", 'javascript:void(0);');
                    }
             }); 

             $(setID+" button").each(function( index ) {
                  $(this).attr('id',row.id)
             });

             $(setID+" h1").each(function( index ) {
                 if ($(this).attr(keyname)) {
                      $(this).attr('id',"h1"+$(this).attr(keyname)+index+row.id)
                      $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
             });  
              $(setID+" h2").each(function( index ) {
                 if ($(this).attr(keyname)) {
                     $(this).attr('id',"h2"+$(this).attr(keyname)+index+row.id)
                      $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
             });  
              $(setID+" h3").each(function( index ) {
                 if ($(this).attr(keyname)) {
                      $(this).attr('id',"h3"+$(this).attr(keyname)+index+row.id)
                      $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
             }); 
              $(setID+" h4").each(function( index ) {
                 if ($(this).attr(keyname)) {
                      $(this).attr('id',"h4"+$(this).attr(keyname)+index+row.id)
                      $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
             }); 
              $(setID+" h5").each(function( index ) {
                 if ($(this).attr(keyname)) {
                      $(this).attr('id',"h5"+$(this).attr(keyname)+index+row.id)
                      $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
             }); 
              $(setID+" h6").each(function( index ) {
                 if ($(this).attr(keyname)) {
                      $(this).attr('id',"h6"+$(this).attr(keyname)+index+row.id)
                      $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
             }); 
             $(setID+" p").each(function( index ) {
                 if ($(this).attr(keyname)) {
                      $(this).attr('id',"p"+$(this).attr(keyname)+index+row.id)
                      $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
             });  
             $(setID+" div").each(function( index ) {
                 if ($(this).attr(keyname)) {
                  $(this).attr('id',"div"+$(this).attr(keyname)+index+row.id)
                  $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
                 if ($(this).attr('primarykey')) {
                  $(this).attr('primarykey',keyname)
                  $(this).attr('id',keyname+row.id).nextAll().remove();
                 }
             });  
             $(setID+" li").each(function( index ) {
                 if ($(this).attr(keyname)) {
                  $(this).attr('id',"li"+$(this).attr(keyname)+index+row.id)
                  $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
                 if ($(this).attr('primarykey')) {
                  $(this).attr('primarykey',keyname)
                  $(this).attr('id',keyname+row.id).nextAll().remove();
                 }
             });  

            $(setID+" ul").each(function( index ) {
                 if ($(this).attr(keyname)) {
                  $(this).attr('id',"ul"+$(this).attr(keyname)+index+row.id)
                  $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
                 if ($(this).attr('primarykey')) {
                  $(this).attr('primarykey',keyname)
                  $(this).attr('id',keyname+row.id).nextAll().remove();
                 }
             });  

            $(setID+" textView").each(function( index ) {
                 if ($(this).attr(keyname)) {
                  $(this).attr('id',"textView"+$(this).attr(keyname)+index+row.id)
                  $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
            });
            $(setID+" small").each(function( index ) {
                 if ($(this).attr(keyname)) {
                  $(this).attr('id',"small"+$(this).attr(keyname)+index+row.id)
                  $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
            });

            $(setID+" span").each(function( index ) {
                 if ($(this).attr(keyname)) {
                  $(this).attr('id',"span"+$(this).attr(keyname)+index+row.id)
                  $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
            }); 
            $(setID+" dd").each(function( index ) {
                 if ($(this).attr(keyname)) {
                  $(this).attr('id',"dd"+$(this).attr(keyname)+index+row.id)
                  $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
            }); 

            $(setID+" dt").each(function( index ) {
                 if ($(this).attr(keyname)) {
                  $(this).attr('id',"dt"+$(this).attr(keyname)+index+row.id)
                  $(this).addClass(keyname+'-'+$(this).attr(keyname));
                 }
            }); 
          $(setID+" img").each(function( index ) {
                if ($(this).attr(keyname)) {
                  $(this).attr('id',"img"+$(this).attr(keyname)+index+row.id)
                }
               if ($(this).attr('size')) {
                   var size=$(this).attr('size');
                   var css=size.split('x');
                   var sizeId=$(this).attr('id');
                      $("#"+sizeId).css({
                           'height': css[1]+'px',
                           'width':  css[0]+'px'
                   });
               }
            });
              // BATAS // 
               myTemplate+=$(setID).html();
         })
        $(useElement).html(myTemplate)
        $.each(query, function(i, set) {
                        $(setID+"  img").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="img"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                               if ($(this).attr('link')) {
                                 if (set[myIdname]==null) {
                                    var nySrc=tatiyeNet.routes+'/'+'images/img/carousel-3.svg'
                                 } else {
                                    var nySrc=tatiyeNet.routes+'/'+set[myIdname]
                                 }
                                 $(useElement +" #"+myId).attr("src", nySrc);
                               } else {
                                  if (set[myIdname]) {
                                     var nySrc=set[myIdname]
                                  } else {
                                       var nySrc=tatiyeNet.routes+'/'+'images/img/carousel-3.svg'
                                  }
                                  $(useElement +" #"+myId).attr("src", nySrc);
                               }
                           }
                        });

                      $(setID+"  h1").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="h1"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                        $(setID+"  h2").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="h2"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                              }
                     
                        });
                        $(setID+"  textView").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="textView"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
     
                        $(setID+"  h3").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="h3"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                        $(setID+"  h4").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="h4"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                        $(setID+"  h5").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="h5"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                        $(setID+"  h6").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="h6"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                         $(setID+"  span").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="span"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                         $(setID+" p").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="p"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                         $(setID+" div").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="div"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                         $(setID+"  li").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="li"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                        $(setID+"  dt").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="dt"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                            $(setID+"  dd").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="dd"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                        $(setID+"  b").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="b"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                         $(setID+"  strong").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="strong"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                         $(setID+"  small").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="small"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                      $(setID+"  class").each(function( index ) {
                           if ($(this).attr(keyname)) {
                              var myId="class"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                         $(setID+" i").each(function( index ) {
                           if ($(this).attr(keyname)) {
                               var myId="i"+$(this).attr(keyname)+index+set.id;
                              var myIdname=$(this).attr(keyname);
                                var origText=$(useElement +" #"+myId).html();
                                var setorigText=replace(origText,set[myIdname],'')+' '+set[myIdname]
                                var titelText=replace(setorigText,'undefined','')
                                $(useElement +" #"+myId).html(titelText);
                           }   
                        });
                  
                      

           })

 var assets=document.querySelectorAll(setID+" div,a,ul,li,i,img,span,nav,dd,dt,small,h1,h2,h3,h4,h5,h6,button,input,p");

  $(assets).each(function(index) {
    

        if ($(this).attr('modal')) {
            $(this).click(function(){
                if ($(this).attr('onclick')) {
                    console.log($(this).attr('id'))
                     var myId=$(this).attr('modal');
                     var Id=$(this).attr('id');
                     var keyID=$(this).attr('fetchKey');
                     var el=$(this).attr('el');
                     var setitem=$(this).attr('setitem');
                     var setitemtitel=$(this).attr('titel');
                     var setswitch=$(this).attr('switch');
                     var privacy=$(this).attr('privacy');
                     var path=$(this).attr('path');

                     var setoption=$(this).attr('option');
                     var setfooter=$(this).attr('footer');
                     var setbackdrop=$(this).attr('backdrop');
                     var setsizes=$(this).attr('sizes');
                     var seteffect=$(this).attr('effect');
                     var setdraggable=$(this).attr('draggable');
                     var bodyheader=$(this).attr('body-header');
                     var bodyfooter=$(this).attr('body-footer');

                     // console.log(privacy)
                     if (privacy) {
                        var dataprivacy='privacy|'+privacy
                        $("button #privacy").show();

                     } else {
                        var dataprivacy='public'
                     }



                     let row=packageIndex(setitem)
                     setModal({
                           'titel':"header",
                           'width':'300px',
                           'key'  :'add',
                           'model':'header',
                           'route':'path',
                           'tabel':'tabel',
                           'content':'element',
                           'page'   :'1',
                           'setdata'   :'public',
                           'data'   :row,
                          });
                    $.ajax({
                             type: 'POST',
                             url:  tatiyeNet.route,
                             data: 'package=elements/'+myId+'&key=0',
                             beforeSend: function() {
                            },
                            success: function(data) {
                                $("#bodyModal").html(data);  
                                var eltDOM=elsDOM('modal');
                                const section = document.querySelectorAll(eltDOM['selector'][3]);
                                var rowsetIData =useHandelIDTabel(row.storage.tabel+','+Id);
                           
                                $(eltDOM['selector'][2]).attr('id','bodyFrom'+setitem);
                                $(eltDOM['selector'][2]).addClass('row row-xs');
                                $(eltDOM['selector'][2]).css({"padding-top":"0rem","padding-bottom":"0rem"});
                                let hasCalss=replace(eltDOM['selector'][2],'.','')
                    
                                $(section[0]).attr('onclick',"closeModal()");
                                $(section[1]).attr('onclick',"closeModal()");
                                var myClass=$(section[2]).attr('class');
                                var setEl=elDOM('el','spinner');
                              $('#modal').modal({
                                backdrop: 'static'
                              }); 

                             $(eltDOM['selector'][2]).before(`<div class="${hasCalss}-header"></div>`);
                             $(eltDOM['selector'][2]).after(`<div class="${hasCalss}-footer"></div>`);
                             $(eltDOM['selector'][2]+'-header').css({
                                "padding-left":"1rem",
                                "padding-right":"1rem",
                                "padding-top":"0.3rem",
                                "padding-bottom":"0.3rem"
                            });
                             $(eltDOM['selector'][2]+'-footer').css({
                                "padding-left":"1rem",
                                "padding-right":"1rem",
                                "padding-top":"0.3rem",
                                "padding-bottom":"0.3rem"
                            });
                           

                           if (bodyheader) {
                             textContentDOM(".modal-body-header",bodyheader); 
                           }
                           if (bodyfooter) {
                              textContentDOM(".modal-body-footer",bodyfooter); 
                           }


                             if (setfooter) {
                                $(eltDOM['selector'][4]).html('');
                             }
                             if (!setbackdrop) {
                               $(eltDOM['selector'][5]).removeClass('show')
                               $(eltDOM['selector'][0]).addClass('shadow-lg');
                             }
                             if (setsizes) {
                               $(eltDOM['selector'][0]).css({'width':setsizes})
                             }
                             if (seteffect) {
                                $(eltDOM['selector'][6]).addClass(seteffect);
                             }

                             if (!setdraggable) {
                              $(eltDOM['selector'][6]).draggable({
                                  handle: eltDOM['selector'][7],
                                  scroll: false
                               });
                             }








                            // setswitch Data Modal
                            if (setswitch=='delete') {
                                 var tedID=$(eltDOM['selector'][2]).html();
                                 $(eltDOM['selector'][2]).html(tedID+' <b>'+rowsetIData[setitemtitel]+'</b>');
                                 $(section[2]).attr('onclick',"onPressKey([`"+row.storage.segment+"`,'delete',`"+Id+"`,`"+keyID+"`,`"+keyID+""+Id+"`])");
                                 $(section[2]).attr('id','spinner'+Id);
                                 $("#spinner"+Id).after(setEl.template);
                                 $(setEl.data[0]).attr('class',myClass);
                            } else {
                            // Form action
                              if (myId=='route') {
                                $(eltDOM['selector'][1]).text(rowsetIData[setitemtitel]);
                                 objectRoute(path,Id,rowsetIData,'bodyFrom'+setitem)
                              } else {
                              $(section[2]).attr('id',"sendEntri"+setitem);
                              $("#sendEntri"+setitem).after(setEl.template);
                              $(setEl.data[0]).attr('id','endEntri'+setitem);
                              $("#endEntri"+setitem).attr('class',myClass);


                               $(eltDOM['selector'][1]).text(rowsetIData[setitemtitel]);
                               setFrom({
                                "storage"      :{
                                     'label'       :row.storage[0],
                                     'tabel'       :row.storage.tabel,
                                     'segment'     :row.storage.segment,
                                     'az'          :row.storage.variabel[2],
                                     'key'         :Id,
                                     'model'       :setitem,
                                     'route'       :row.storage.route,
                                     'content'     :row.storage.content,
                                     'page'        :1,
                                     'footer'      :false,
                                     'setdata'     :dataprivacy,
                                     'setItem'     :false,
                                   },
                                   "action"      :row.action,
                                });
                               }
                           } 
                           }
                       });    
               }
             });
         }


         if ($("#"+$(this).attr('id')).attr('textLeft')) {
             if ($(this).attr('textRight')) {
              $("#"+$(this).attr('id')).html($(this).attr('textLeft')+" "+$(this).text()+" "+$(this).attr('textRight'));
             } else {
              $("#"+$(this).attr('id')).html($(this).attr('textLeft')+" "+$(this).text());

             }

         }
          /*
          |--------------------------------------------------------------------------
          | Initializes  text
          |--------------------------------------------------------------------------
          | Develover Tatiye.Net 2020
          | @Date 12/13/2023 10:08:39 PM
          */

         // if ($(this).attr('text')) {
         //    var data=$(this).html();
         //    var datasplit=$(this).attr('split');
         //    if ($(this).attr('text')=='romawi') {
         //     var netRomawi=this.textContent.replace(splitLeft(data,datasplit),romawi(splitLeft(data,datasplit)));
         //     $(this).html(netRomawi);
         //    }
         // }
       /*
          |--------------------------------------------------------------------------
          | Initializes  width
          |--------------------------------------------------------------------------
          | Develover Tatiye.Net 2020
          | @Date 12/13/2023 10:08:39 PM
          */
         if ($(this).attr('wt')) {
            var ht=$(this).attr('wt');
            var sizeId=$(this).attr('id');
            $("#"+sizeId).css({'width': ht});
           }

          // border-radius 
          if ($(this).attr('br')) {
            var padding=$(this).attr('br');
            $(this).css({'border-radius': padding});
           } 
          /*
          |--------------------------------------------------------------------------
          | Initializes  Background
          |--------------------------------------------------------------------------
          | Develover Tatiye.Net 2020
          | @Date 12/13/2023 10:08:39 PM
          */
         if ($(this).attr('bg')) {
            var bg=$(this).attr('bg');
            var sizeId=$(this).attr('id');
            $("#"+sizeId).css({'background-color': bg});
           }
          /*
          |--------------------------------------------------------------------------
          | Initializes  padding
          |--------------------------------------------------------------------------
          | Develover Tatiye.Net 2020
          | @Date 12/13/2023 10:08:39 PM
          */ 
          if ($(this).attr('pd')) {
            var padding=$(this).attr('pd');
            $(this).css({'padding': padding});
           }
          /*
          |--------------------------------------------------------------------------
          | Initializes  padding-top
          |--------------------------------------------------------------------------
          | Develover Tatiye.Net 2020
          | @Date 12/13/2023 10:08:39 PM
          */ 
          if ($(this).attr('pt')) {
            var padding=$(this).attr('pt');
            $(this).css({'padding-top': padding});
           }
          /*
          |--------------------------------------------------------------------------
          | Initializes  padding-right
          |--------------------------------------------------------------------------
          | Develover Tatiye.Net 2020
          | @Date 12/13/2023 10:08:39 PM
          */
          if ($(this).attr('pt')) {
            var padding=$(this).attr('pt');
            $(this).css({'padding-right': padding});
           }
          /*
          |--------------------------------------------------------------------------
          | Initializes  padding-bottom
          |--------------------------------------------------------------------------
          | Develover Tatiye.Net 2020
          | @Date 12/13/2023 10:08:39 PM
          */
          if ($(this).attr('pb')) {
            var padding=$(this).attr('pb');
            $(this).css({'padding-bottom': padding});
           }
          /*
          |--------------------------------------------------------------------------
          | Initializes  padding-left
          |--------------------------------------------------------------------------
          | Develover Tatiye.Net 2020
          | @Date 12/13/2023 10:08:39 PM
          */
          if ($(this).attr('pl')) {
            var padding=$(this).attr('pl');
            $(this).css({'padding-left': padding});
           }
           // Margins
          if ($(this).attr('mg')) {
            var padding=$(this).attr('mg');
            $(this).css({'margin': padding});
           }
           // padding-top
          if ($(this).attr('mt')) {
            var padding=$(this).attr('mt');
            var sizeId=$(this).attr('id');
            $("#"+sizeId).css({'margin-top': padding});
           }
           // padding-right
          if ($(this).attr('mr')) {
            var padding=$(this).attr('mr');
            var sizeId=$(this).attr('id');
            $(this).css({'margin-right': padding});
           }
           // padding-bottom
          if ($(this).attr('mb')) {
            var padding=$(this).attr('mb');
            $(this).css({'margin-bottom': padding});
           }
           // padding-left
          if ($(this).attr('ml')) {
            var padding=$(this).attr('ml');
            $(this).css({'margin-left': padding});
           }

           // Font Size
          if ($(this).attr('fs')) {
            var tx=$(this).attr('fs');
            $(this).css({'font-size': tx});
           }


           if ($(this).attr('scrollbar')) {
             $(this).addClass('setScrol');
              var tx=$(this).attr('scrollbar');
            $(this).css({'max-height': tx});
           }



           if ($(this).attr('textMore')) {

               var textValue=readmore($(this).text(),$(this).attr('textMore'));
               $(this).html(textValue);
            }
           // Typography
          if ($(this).attr('tx')) {
            var tx=$(this).attr('tx');
            $(this).css({'color': tx});
           }




       if ($(this).attr('refresh')) {
              var myIda=$(this).attr('refresh');
              const getAtr = document.querySelector("#"+myIda);
              var gmyId=$(getAtr).attr('content');
              var myId=gmyId+'_'+myIda;
             

              var desain=`
               <div style="display:none;" id="endReload${myId}">
                    <div class="text-center ${rtsDOM('spinner')}">
                      <div class="spinner-border"></div>
                    </div>
               </div>`;
               $("#"+myId).before(desain);
                  $("a,button").click(function(){
                     $("#"+myId).hide();
                     $("#endReload"+myId).show();
                     $("#endReload"+myId).show();  
                    setTimeout(function(){
                         $("#"+myId).show();
                         $("#endReload"+myId).hide();
                        reload(); 
                    },300);    

                     
                  });
           }



       /*
       |--------------------------------------------------------------------------
       | Initializes  Icon Images
       |--------------------------------------------------------------------------
       | Develover Tatiye.Net 2020
       | @Date 12/13/2023 10:08:39 PM
       */

        if ($(this).attr('icon')) {
            var nmIcon=$(this).attr('icon');
            var size=$(this).attr('size');
            var css=size.split('x');
                   $(this).css({
                      'display': 'inline-block',
                       'background-image': 'url(' + nmIcon + ')',
                         'background-repeat': 'no-repeat',
                         'background-size': '100% 100%'
                      });


           }
  

         })
         $(setID+" img").each(function( index ) {
                  if ($(this).attr('resizer')) {
                       var size=$(this).attr('resizer');
                       var filesrc=$(this).attr('src');
                       var watermark=$(this).attr('watermark');
                       var cropImage=$(this).attr('crop');
                       var filealt=$(this).attr('alt');
                       let imgFile=setSplit(1,'images',filesrc)

            // console.log(filealt)
            var css=size.split('x');
             if(filealt == 'wt') {
              var style=css[2]+'x'+css[1];
              var width=css[2];
              var height=css[1];
              var sizecastom=css[2];
             } else if (filealt == 'ht'){
              var style=css[0]+'x'+css[2];
              var width=css[0];
              var height=css[2];
              var sizecastom=css[2];
             } else if (filealt == 'black'){
              var style=css[0]+'x'+css[1];
              var width=css[0];
              var height=css[1];
              var sizecastom='';
             } else {
              var style=css[0]+'x'+css[1];
              var width=css[0];
              var height=css[1];
              var sizecastom='';
             }


            let mydata = [];
            if (imgFile) {
              var file=tatiyeNet.routes+'/images/'+style+''+imgFile;  
            var fileresizer=style+''+imgFile;
            } else {
              var file=tatiyeNet.routes+'/images/'+style+'/'+filesrc;  
              var fileresizer=style+'/'+filesrc;

            }
                  $(this).attr('src',file);
                  var sizeId=$(this).attr('id');
                  $("#"+sizeId).css({
                      'display': 'inline-block',
                       'background-image': 'url(' +mydata.resizer+ ')',
                         'background-repeat': 'no-repeat',
                         'background-size': '100% 100%',
                         'width':  width+'px',
                         'height': height+'px'
                 });

                if (watermark) {
                    var water=watermark;
                
                } else {
                    var water='';
                }

                if (cropImage) {
                    var crop=cropImage;
                    var cropFile=size+'/'+filesrc;
                } else {
                    var crop='';
                    var cropFile='';

                }
                var settings = {
                       "size"    :fileresizer,
                       "file"    :filesrc,
                       "resizer" :style,
                       "sizecastom":sizecastom,
                       "label" :filealt,
                       "watermark" :water,
                       "cropimg" :crop
                    };
                $.ajax({
                    url: tatiyeNet.api+"/v1/select/imageresizer",
                    method: 'POST',
                    data: JSON.stringify(settings),
                    datatype: "json",
                    async: false,
                    success: function(data){
                        var add=JSON.stringify(data, null, 10);
                        mydata = JSON.parse(add); 
                                 
                    }
                  });
                  }

               }); 
 
};
