// tatiye/Assets/index.js
import tatiyeNet,{
  modalEntri,
  useHandelIDTabel,
  openSSLKey,
  setModal,
  setFrom,
  useHandel
}  from "../es6.js"; 
import {setTab} from "./tab.js";
import {setDropdown} from "./dropdown.js";
import {setDropdownTabel} from "./dropdownTabel.js";
import {setfrom} from "./from.js";
import {setElement} from "./element.js";
import {setMyModal} from "./modal.js";
import {setModalStandar} from "./modalStandar.js";
import {setKeywords} from "./keywords.js";
import {setTabel} from "./tabel.js";
import {setElementFrom} from "./elementFrom.js";
import {setModalBootstrap} from "./modalBootstrap.js";
import {setDatatables} from "./datatables.js";
import {setOffice} from "./Office.js";
import {setdatatablesXlsx} from "./datatablesXlsx.js";
import {setuseDevices} from "./devicesOffice.js";
export class Assets {
  constructor() {
    this.count        =0; 
    this._viewText ='';
  }

    bootstrap(row) {
      setModalBootstrap(row);
    } 
    tab(row) {
      setTab(row);
    } 
    dropdown(row) {
      setDropdown(row);
    } 
    useDevices(row) {
      setuseDevices(row);
    } 

    datatables(row) {
      setDatatables(row);
    } 
    officeDatatables(row) {
       setOffice(row);
    } 
    datatablesXlsx(row) {
        setdatatablesXlsx(row);
    } 

    tabel(row) {
      setTabel(row);
    } 
    modal(row){
      setMyModal(row);
    }
    modalStandar(row){
      setModalStandar(row);
    }
    keywords(row){
      setKeywords(row);
    }
    dropdownTabel(row){
      setDropdownTabel(row);
    }
    from(row){

      if (row.storage[3]) {
        var azName=row.storage[3];
      } else {
        var azName='z';
      }
      // console.log(row.storage[0])
      let storage={
         'label'    :row.storage[0],
         'tabel'    :row.storage[1],
         'segment'  :false,
         'az'       :azName,
         'key'      :row.storage[2],
         'model'    :false,
         'route'    :false,
         'content'  :false, //default,from,upload
         'page'     :'1',
       };
       
      setfrom(row,storage);
    }

    useHandel(row){
       this.count ++;
        let attribute=row.app+'-tatiye';
         const getAtr = document.querySelector(attribute); 
         let element =row.app+'-'+getAtr.getAttribute('element');
        var dabatabse=useHandel({
           "token"          :row.token,
            "storage"       :row.storage
          })
        if (row.console==true) {
            console.log(dabatabse)
        }
            /*
            |--------------------------------------------------------------------------
            | Initializes getAttribute 
            |--------------------------------------------------------------------------
            | Develover Tatiye.Net 2018
            | @Date 
            */
            /**
             * @param array  options the display options .
             * @param mixed  Block to generate a customized inside  content.
             */
             
             let setID='#'+getAtr.getAttribute('id');
             // $(setID).hide();
             var myTemplate='';
             dabatabse.storage.forEach(set => {
                 $(setID+" div").each(function( index ) {
                    if ($( this ).attr('name')) {
                     $(this).attr('id',$( this ).attr('name')+index+set.no)
                    }
                });
                $(setID+" p").each(function( index ) {
                    if ($( this ).attr('name')) {
                     $(this).attr('id',$( this ).attr('name')+index+set.no)
                    }
                });

                $(setID+" a").each(function(index) {
                    if ($( this ).attr('name')) {
                       $(this).attr('name',index+set.no)
                       $(this).attr('id',set.id)
                    }
                });
                $(setID+" dropdown").each(function(index) {
                    if ($(this).attr('name')) {
                        $(this).attr('id','down'+set.no)
                        $(this).attr('name',set.id)
                    }
                });

             // }
               myTemplate +=$(setID).html();
            });
             $(element).html(myTemplate) 
             //console.log(myTemplate)

            /*
            |--------------------------------------------------------------------------
            | Initializes Database 
            |--------------------------------------------------------------------------
            | Develover Tatiye.Net 2018
            | @Date 
            */
            /**
             * @param array  options the display options .
             * @param mixed  Block to generate a customized inside  content.
             */
             dabatabse.storage.forEach(set => {
                 
                 $(setID+"  p").each(function( index ) {
                   if ($( this ).attr('name')) {
                      var myId=$(this).attr('name')+index+set.no;
                      var myIdname=$(this).attr('name');
                      var myText=$(element +" #"+myId).html();
                      if ($(this).attr('textRight')) {
                         var textRight=$(this).attr('textRight');
                      } else {
                          var textRight='';
                      }
                      if ($(this).attr('textClass')) {
                         var textValue='<tx class="'+$(this).attr('textClass')+'">'+set[myIdname]+'</tx>';
                      } else {
                          var textValue=set[myIdname];
                      }  
                      $(element +" #"+myId).html(myText+' '+textValue+' '+textRight);
                   }
                });
                 $(setID+"  div").each(function( index ) {
                   if ($( this ).attr('name')) {
                      var myId=$(this).attr('name')+index+set.no;
                      var myIdname=$(this).attr('name');
                      var myText=$(element +" #"+myId).html();
                      if ($(this).attr('textRight')) {
                         var textRight=$(this).attr('textRight');
                      } else {
                          var textRight='';
                      }
                      if ($(this).attr('textClass')) {
                         var textValue='<tx class="'+$(this).attr('textClass')+'">'+set[myIdname]+'</tx>';
                      } else {
                          var textValue=set[myIdname];
                      }  
                      $(element +" #"+myId).html(myText+' '+textValue+' '+textRight);
                   }
                });
              
            });
           
    // return row;


    }
    element(row,element){
        setElement(row,element); 
    }

    myfromModal(from,setId='',elementFrom=''){
         setElementFrom(from); 
        // console.log(from)
     }
   
}
