// tatiye/Tables/index.js
import tatiyeNet,{setDecode,readmore}  from "../es6.js";  


export class Tables {
    constructor(token) {
        this.token = token
        this.content =''
    }
    headTabel(e){
      // console.log(e)
      var tableObj='#'+e.table;
      var dataTable='#'+e.dataTable;
      var appData='';
      if (e.myfrom) {

          var myFrom        ='/helpers/Form/myfrom';
          var actionFrom    =JSON.stringify(e.myfrom);
         $.post(tatiyeNet.etc,{
                 etc          :myFrom,
                 action      :actionFrom,
                })
               .done(function (data) {
                $("#fromAction_"+e.table).html('<div class="row row-xs">'+data+'</div>');
          });
        }
        appData=`
            <div id="fromAction_${e.table}"></div>
            <table id="${e.dataTable}" class="table" style="width:100%"><thead><tr>`;
            $.each(e.tableHead,function(key,value){
                if (value=='No') {
                   var sty='style="width: 40px;"';
                } else {
                   var sty='';
                }

                //var desainTabel1=`<th ${sty} id="${value}">${value}</th>`;
                var desainTabel=`<th ${sty} id="${value}">${value}</th>`;
                appData+=desainTabel;
              });
            if (e.tabelFooter==true) {
                appData +=`</tr></thead><tfoot align="right"><tr>`;
                $.each(e.tableHead,function(key,value){
                    var desainTabel=`<td class="text-left"></td>`;
                     appData+=desainTabel;
                 });  
            }
            appData +=`</tr></tfoot></table>`;
            $(tableObj).before(appData);
             this.content=e.table+'Body';
             // var idTabel='#dropdown';
             // $(idTabel).text('Helllo');
             // console.log(idTabel)
     // console.log(e)
    }
    extend(e){
        if(e.extend ==true) {
           var my_array = [];
           $.each(e.sheetData,function(key, set){
              my_array.push({
                    extend: 'colvisGroup',
                    text: '<span class="tdText"><i class="picons-180"></i> <span id="tText">'+key+'</span></span>',
                    show: set['show'],
                    hide: set['hide'],
                   });
           });
          var sheetData=my_array;
      // console.log(sheetData)
        } else {
        var sheetData=[];
     
        }
        if(e.refresh ==true) {
          var roled={
                   'text': '<span class="tdText"><i class="picons-143"></i> <span id="tText">Refresh</span></span>',
                    action: function ( e, dt, node, config ) {
                       $(".table").DataTable().ajax.reload( null, false);
                    }
                 }
        } else {
           var roled=[]
        }
      //console.log(Array.isArray(e.sheetData));
      return [{   
                     "extend": 'pdfHtml5',
                     "filename": e.filename,
                      "orientation": 'landscape',
                      "pageSize": 'LEGAL',
                     "footer": true, 
                     "exportOptions": {
                         "modifier": {
                             "page": 'current'
                         },
                         
                     }
                 },
                 {   
                     "extend": 'csvHtml5',
                     "filename": e.filename,
                     "footer": true, 
                     "exportOptions": {
                         "modifier": {
                             "page": 'current'
                         }
                     }
                 },
                 {   
                     "extend": 'excelHtml5',
                     "filename": e.filename,
                     "footer": true, 
                     "exportOptions": {
                         "modifier": {
                             "page": 'current'
                         }
                     }
                 },


                 roled,
                 sheetData,
            ]
    }
    sheetData(data){
      return data
    }
    fromDropdown(e){
      
      // console.log(e.dropdown)
      // console.log(e)



               var setdropdown='';
                 $.each(e.dropdown,function(keydwn,dwn){
 
                    if (dwn[0]=='download') {

                      setdropdown +=`<a  href="${typeDevices}" class="pt-5px dropdown-item"target="_blank">
                       <i class="icon-feather-${dwn[2]} fs-14px mr-5px"></i> ${keydwn}</a>`;
                    } else {
                      setdropdown +=`<a  onclick="useOnclick(['${dwn[0]}','${e.id}','${dwn[1]}','${dwn[3]}']);" href="javascript:void(0);" class="pt-5px dropdown-item ">
                       <i class="feat feat-${dwn[2]} fs-14px mr-5px"></i> ${keydwn}</a>`;
                    }
    
                  });
                var desainTabel=`
                 <a href="javascript:void(0);" class="dropdown-link" data-toggle="dropdown"><i class="feat feat-more-vertical"></i></a>
                 <div class="dropdown-menu">
                     ${setdropdown}
                 </div>
                `;
                return desainTabel;


               // $("#"+e).html('e');



    }
    storage(){
      return tatiyeNet.api+'/v2/'+this.token
    }
    authorization(){
        // console.log(tatiyeNet.token)
       return tatiyeNet.token
    }


}
